<div class="container">

<table style="width: 100%;"><tr>
<td>iC10</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>A copy number and expression-based classfier for breast cancers</h2>

<h3>Description</h3>

<p>iC10 implements the classifier described in the paper 'Genome-driven integrated classification of breast cancer validated in over 7,500 samples' (Ali HR et al., Genome Biology 2014). It uses copy number and/or expression form breast cancer data, trains a pamr classifier (Tibshirani et al.) with the features available and predicts the iC10 group.
</p>


<h3>Usage</h3>

<pre><code class="language-R">iC10(x, seed=25435)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>An object with class <code>iC10features</code>: A list with elements 'train.CN', 'train.Exp', 'train.iC10',
'CN', 'Exp', 'map.cn', 'map.exp'</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seed</code></td>
<td>
<p>seed to initialize random number generator. It is passed to <code>set.seed()</code>.
See details.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function trains a <code>pamr</code> classifier and predicts the set of
samples. The shrinkage parameter is obtained with crossvalidation,
therefore different runs can give different results (unless a seed is specified).
</p>


<h3>Value</h3>

<p>An object of class <code>iC10</code>. A list with the following elements:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>class</code></td>
<td>
<p>Prediction classes for the samples</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>posterior</code></td>
<td>
<p>Probablitites for each sample to belong to each of
the 10 groups</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>centroids</code></td>
<td>
<p>Shrunken Centroids for each of the 10 groups.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fitted</code></td>
<td>
<p>Normalized features for the samples classified.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>map.cn</code></td>
<td>
<p>Annotation data for the copy number features</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>map.exp</code></td>
<td>
<p>Annotation data for the expression features</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Oscar M. Rueda
</p>


<h3>References</h3>

<p>Ali HR et al. Genome-driven integrated classification of breast cancer
validated in over 7,500 samples. Genome Biology 2014; 15:431.
Curtis et al. The genomic and transcriptomic architecture of 2,000
breast tumours reveals novel subgroups. Nature 2012; 486:346-352.
Tibshirani et al. Diagnosis of multiple cancer types by shrunken
centroids of gene expression. PNAS 2002; 99(10):6567-6572.
</p>


<h3>See Also</h3>

<p>See <code>pamr.train</code>, <code>pamr.cv</code> and <code>pamr.predict</code> in package <code>pamr</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">require(iC10TrainingData)
data(train.CN)
data(train.Exp)
features &lt;- matchFeatures(Exp=train.Exp, Exp.by.feat="probe")
features &lt;- normalizeFeatures(features, "scale")
res &lt;- iC10(features)

</code></pre>


</div>