<div class="container">

<table style="width: 100%;"><tr>
<td>getExpfeatures</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Internal function for matching expression features.
</h2>

<h3>Description</h3>

<p>Internal function for matching expression features.
</p>


<h3>Usage</h3>

<pre><code class="language-R">getExpfeatures(Exp, Probes, Synonyms, by.feat)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Exp</code></td>
<td>

<p>Matrix of expression features
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Probes</code></td>
<td>

<p>Vector of probes to match
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Synonyms</code></td>
<td>

<p>vector of synonyms fo gene names
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>by.feat</code></td>
<td>

<p>either "probe" or "gene"
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A matrix with the <code>Probes</code> in Exp.
</p>


<h3>Note</h3>

<p>This function is not supposed to be called directly. use <code>matchFeatures</code> instead.
</p>


<h3>Author(s)</h3>

<p>Oscar M Rueda
</p>


<h3>References</h3>

<p>Curtis et al. The genomic and transcriptomic architecture of 2,000
breast tumours reveals novel subgroups. Nature 2012; 486:346-352.
</p>


<h3>See Also</h3>

<p><code>matchFeatures</code>
</p>


</div>