<div class="container">

<table style="width: 100%;"><tr>
<td>ReadMutectVCFs</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Read Mutect VCF files.</h2>

<h3>Description</h3>

<p>Read Mutect VCF files.
</p>


<h3>Usage</h3>

<pre><code class="language-R">ReadMutectVCFs(files, names.of.VCFs = NULL, tumor.col.names = NA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>files</code></td>
<td>
<p>Character vector of file paths to the VCF files.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>names.of.VCFs</code></td>
<td>
<p>Character vector of names of the VCF files. The order of
names in <code>names.of.VCFs</code> should match the order of VCF file paths in
<code>files</code>. If <code>NULL</code>(default), this function will remove all of the
path up to and including the last path separator (if any) in <code>files</code>
and file paths without extensions (and the leading dot) will be used as the
names of the VCF files.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tumor.col.names</code></td>
<td>
<p>Character vector of column names in VCFs which contain
the tumor sample information. The order of names in <code>tumor.col.names</code>
should match the order of VCFs specified in <code>files</code>. If
<code>tumor.col.names</code> is equal to <code>NA</code>(default), this function will
use the 10th column in all the VCFs to calculate VAFs.
See <code>GetMutectVAF</code> for more details.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list of data frames which store data lines of VCF files
with two additional columns added which contain the VAF(variant allele
frequency) and read depth information.
</p>


</div>