<div class="container">

<table style="width: 100%;"><tr>
<td>TranscriptRanges</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Transcript ranges data</h2>

<h3>Description</h3>

<p>Transcript ranges and strand information for a particular reference genome.
</p>


<h3>Usage</h3>

<pre><code class="language-R">trans.ranges.GRCh37

trans.ranges.GRCh38

trans.ranges.GRCm38
</code></pre>


<h3>Format</h3>

<p>A <code>data.table</code> which contains transcript
range and strand information for a particular reference genome.
<code>colname</code>s are <code>chrom</code>, <code>start</code>, <code>end</code>, <code>strand</code>,
<code>Ensembl.gene.ID</code>, <code>gene.symbol</code>. It uses one-based coordinates.
</p>
<p>An object of class <code>data.table</code> (inherits from <code>data.frame</code>) with 19083 rows and 6 columns.
</p>
<p>An object of class <code>data.table</code> (inherits from <code>data.frame</code>) with 19096 rows and 6 columns.
</p>
<p>An object of class <code>data.table</code> (inherits from <code>data.frame</code>) with 20325 rows and 6 columns.
</p>


<h3>Details</h3>

<p>This information is needed to generate catalogs that
depend on transcriptional
strand information, for example catalogs of 
class <code>SBS192Catalog</code>. 
</p>
<p><code>trans.ranges.GRCh37</code>:  <strong>Human</strong> GRCh37. 
</p>
<p><code>trans.ranges.GRCh38</code>:  <strong>Human</strong> GRCh38. 
</p>
<p><code>trans.ranges.GRCm38</code>:  <strong>Mouse</strong> GRCm38. 
</p>
<p>For these two tables, only genes that are associated with a CCDS ID are kept for transcriptional
strand bias analysis. 
</p>
<p>This information is needed for <code>StrelkaSBSVCFFilesToCatalog</code>, <br><code>StrelkaSBSVCFFilesToCatalogAndPlotToPdf</code>, 
<code>MutectVCFFilesToCatalog</code>, <br><code>MutectVCFFilesToCatalogAndPlotToPdf</code>,
<code>VCFsToSBSCatalogs</code> and <code>VCFsToDBSCatalogs</code>.
</p>


<h3>Source</h3>

<p><a href="ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_30/GRCh37_mapping/gencode.v30lift37.annotation.gff3.gz">ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_30/GRCh37_mapping/gencode.v30lift37.annotation.gff3.gz</a>
</p>
<p><a href="ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_30/gencode.v30.annotation.gff3.gz">ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_30/gencode.v30.annotation.gff3.gz</a>
</p>
<p><a href="ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_mouse/release_M21/gencode.vM21.annotation.gff3.gz">ftp://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_mouse/release_M21/gencode.vM21.annotation.gff3.gz</a>
</p>


<h3>Examples</h3>

<pre><code class="language-R">trans.ranges.GRCh37
# chrom    start      end strand Ensembl.gene.ID  gene.symbol
#     1    65419    71585      + ENSG00000186092        OR4F5
#     1   367640   368634      + ENSG00000235249       OR4F29
#     1   621059   622053      - ENSG00000284662       OR4F16
#     1   859308   879961      + ENSG00000187634       SAMD11
#     1   879583   894689      - ENSG00000188976        NOC2L
#   ...      ...      ...    ...             ...          ... 
</code></pre>


</div>