<div class="container">

<table style="width: 100%;"><tr>
<td>AnnotateIDVCF</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Add sequence context to an in-memory ID (insertion/deletion) VCF, and
confirm that they match the given reference genome</h2>

<h3>Description</h3>

<p>Add sequence context to an in-memory ID (insertion/deletion) VCF, and
confirm that they match the given reference genome
</p>


<h3>Usage</h3>

<pre><code class="language-R">AnnotateIDVCF(
  ID.vcf,
  ref.genome,
  flag.mismatches = 0,
  name.of.VCF = NULL,
  suppress.discarded.variants.warnings = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ID.vcf</code></td>
<td>
<p>An in-memory ID (insertion/deletion) VCF as a
<code>data.frame</code>. This function expects that there is a "context base" to
the left, for example REF = ACG, ALT = A (deletion of CG) or REF = A, ALT =
ACC (insertion of CC).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ref.genome</code></td>
<td>
<p>A <code>ref.genome</code> argument as described in
<code>ICAMS</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>flag.mismatches</code></td>
<td>
<p>Deprecated. If there are ID variants whose <code>REF</code>
do not match the extracted sequence from <code>ref.genome</code>, the function
will automatically discard these variants. See element
<code>discarded.variants</code> in the return value for more details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>name.of.VCF</code></td>
<td>
<p>Name of the VCF file.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>suppress.discarded.variants.warnings</code></td>
<td>
<p>Logical. Whether to suppress
warning messages showing information about the discarded variants. Default
is TRUE.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list of elements:
</p>

<ul>
<li> <p><code>annotated.vcf</code>: The original VCF data
frame with two new columns added to the input data frame:
</p>

<ul>
<li> <p><code>seq.context</code>: The sequence embedding the variant.
</p>
</li>
<li> <p><code>seq.context.width</code>: The width of <code>seq.context</code> to the left.
</p>
</li>
</ul>
</li>
<li> <p><code>discarded.variants</code>: <strong>Non-NULL only if</strong> there are variants
that were excluded from the analysis. See the added extra column
<code>discarded.reason</code> for more details.
</p>
</li>
</ul>
<h3>Examples</h3>

<pre><code class="language-R">file &lt;- c(system.file("extdata/Strelka-ID-vcf/",
                      "Strelka.ID.GRCh37.s1.vcf",
                      package = "ICAMS"))
ID.vcf &lt;- ReadStrelkaIDVCFs(file)[[1]]
if (requireNamespace("BSgenome.Hsapiens.1000genomes.hs37d5", quietly = TRUE)) {
  list &lt;- AnnotateIDVCF(ID.vcf, ref.genome = "hg19")
  annotated.ID.vcf &lt;- list$annotated.vcf}
</code></pre>


</div>