<div class="container">

<table style="width: 100%;"><tr>
<td>GetVAF</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Extract the VAFs (variant allele frequencies) and read depth information from
a VCF file</h2>

<h3>Description</h3>

<p>Extract the VAFs (variant allele frequencies) and read depth information from
a VCF file
</p>


<h3>Usage</h3>

<pre><code class="language-R">GetStrelkaVAF(vcf, name.of.VCF = NULL)

GetMutectVAF(vcf, name.of.VCF = NULL, tumor.col.name = NA)

GetFreebayesVAF(vcf, name.of.VCF = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>vcf</code></td>
<td>
<p>Said VCF as a data.frame.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>name.of.VCF</code></td>
<td>
<p>Name of the VCF file.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tumor.col.name</code></td>
<td>
<p>Optional. Only applicable to <strong>Mutect</strong> VCF. Name
of the column in <strong>Mutect</strong> VCF which contains the tumor sample
information. It <strong>must</strong> have quotation marks. If
<code>tumor.col.name</code> is equal to <code>NA</code>(default), this function will
use the 10th column to calculate VAFs.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>The original <code>vcf</code> with two additional columns added which
contain the VAF(variant allele frequency) and read depth information.
</p>


<h3>Examples</h3>

<pre><code class="language-R">file &lt;- c(system.file("extdata/Strelka-SBS-vcf",
                      "Strelka.SBS.GRCh37.s1.vcf",
                      package = "ICAMS"))
MakeDataFrameFromVCF &lt;- getFromNamespace("MakeDataFrameFromVCF", "ICAMS")
df &lt;- MakeDataFrameFromVCF(file)
df1 &lt;- GetStrelkaVAF(df)
</code></pre>


</div>