<div class="container">

<table style="width: 100%;"><tr>
<td>StrelkaIDVCFFilesToCatalogAndPlotToPdf</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Create ID (small insertion and deletion) catalog from Strelka ID VCF files
and plot them to PDF</h2>

<h3>Description</h3>

<p>Create ID (small insertion and deletion) catalog from the Strelka ID VCFs
specified by <code>files</code> and plot them to PDF
</p>


<h3>Usage</h3>

<pre><code class="language-R">StrelkaIDVCFFilesToCatalogAndPlotToPdf(
  files,
  ref.genome,
  region = "unknown",
  names.of.VCFs = NULL,
  output.file = "",
  flag.mismatches = 0,
  return.annotated.vcfs = FALSE,
  suppress.discarded.variants.warnings = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>files</code></td>
<td>
<p>Character vector of file paths to the Strelka ID VCF files.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ref.genome</code></td>
<td>
<p>A <code>ref.genome</code> argument as described in
<code>ICAMS</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>region</code></td>
<td>
<p>A character string designating a genomic region;
see <code>as.catalog</code> and <code>ICAMS</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>names.of.VCFs</code></td>
<td>
<p>Optional. Character vector of names of the VCF files.
The order of names in <code>names.of.VCFs</code> should match the order of VCF
file paths in <code>files</code>. If <code>NULL</code>(default), this function will
remove all of the path up to and including the last path separator (if any)
in <code>files</code> and file paths without extensions (and the leading dot)
will be used as the names of the VCF files.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>output.file</code></td>
<td>
<p>Optional. The base name of the PDF file to be produced;
the file is ending in <code>catID.pdf</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>flag.mismatches</code></td>
<td>
<p>Deprecated. If there are ID variants whose <code>REF</code>
do not match the extracted sequence from <code>ref.genome</code>, the function
will automatically discard these variants and an element
<code>discarded.variants</code> will appear in the return value. See
<code>AnnotateIDVCF</code> for more details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>return.annotated.vcfs</code></td>
<td>
<p>Logical. Whether to return the annotated VCFs
with additional columns showing mutation class for each variant. Default is
FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>suppress.discarded.variants.warnings</code></td>
<td>
<p>Logical. Whether to suppress
warning messages showing information about the discarded variants. Default
is TRUE.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function calls <code>StrelkaIDVCFFilesToCatalog</code> and
<code>PlotCatalogToPdf</code>
</p>


<h3>Value</h3>

<p>A <strong>list</strong> of elements:
</p>

<ul>
<li> <p><code>catalog</code>: The ID (small insertion and deletion) catalog with
attributes added. See <code>as.catalog</code> for more details.
</p>
</li>
<li> <p><code>discarded.variants</code>: <strong>Non-NULL only if</strong> there are variants
that were excluded from the analysis. See the added extra column
<code>discarded.reason</code> for more details.
</p>
</li>
<li> <p><code>annotated.vcfs</code>:
<strong>Non-NULL only if</strong> <code>return.annotated.vcfs</code> = TRUE. A list of
data frames which contain the original VCF's ID mutation rows with three
additional columns <code>seq.context.width</code>, <code>seq.context</code> and
<code>ID.class</code> added. The category assignment of each ID mutation in VCF can
be obtained from <code>ID.class</code> column.
</p>
</li>
</ul>
<h3>ID classification</h3>

<p>See <a href="https://github.com/steverozen/ICAMS/blob/master/data-raw/PCAWG7_indel_classification_2021_09_03.xlsx">https://github.com/steverozen/ICAMS/blob/master/data-raw/PCAWG7_indel_classification_2021_09_03.xlsx</a>
for additional information on ID (small insertion and deletion) mutation
classification.
</p>
<p>See the documentation for <code>Canonicalize1Del</code> which first handles
deletions in homopolymers, then handles deletions in simple repeats with
longer repeat units, (e.g. <code>CACACACA</code>, see
<code>FindMaxRepeatDel</code>), and if the deletion is not in a simple
repeat, looks for microhomology (see <code>FindDelMH</code>).
</p>
<p>See the code for unexported function <code>CanonicalizeID</code>
and the functions it calls for handling of insertions.
</p>


<h3>Note</h3>

<p>In ID (small insertion and deletion) catalogs, deletion repeat sizes range
from 0 to 5+, but for plotting and end-user documentation deletion repeat
sizes range from 1 to 6+.
</p>


<h3>Examples</h3>

<pre><code class="language-R">file &lt;- c(system.file("extdata/Strelka-ID-vcf",
                      "Strelka.ID.GRCh37.s1.vcf",
                      package = "ICAMS"))
if (requireNamespace("BSgenome.Hsapiens.1000genomes.hs37d5", quietly = TRUE)) {
  catID &lt;-
    StrelkaIDVCFFilesToCatalogAndPlotToPdf(file, ref.genome = "hg19",
                                           region = "genome",
                                           output.file =
                                           file.path(tempdir(), "StrelkaID"))}

</code></pre>


</div>