<div class="container">

<table style="width: 100%;"><tr>
<td>ReadVCF</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Read in the data lines of a Variant Call Format (VCF) file</h2>

<h3>Description</h3>

<p>Read in the data lines of a Variant Call Format (VCF) file
</p>


<h3>Usage</h3>

<pre><code class="language-R">ReadVCF(
  file,
  variant.caller = "unknown",
  name.of.VCF = NULL,
  tumor.col.name = NA,
  filter.status = NULL,
  get.vaf.function = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p>The name/path of the VCF file, or a complete URL.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>variant.caller</code></td>
<td>
<p>Name of the variant caller that produces the VCF, can
be either <code>"strelka"</code>, <code>"mutect"</code>, <code>"freebayes"</code> or
<code>"unknown"</code>. This information is needed to calculate the VAFs (variant
allele frequencies). If <code>"unknown"</code>(default) and
<code>get.vaf.function</code> is NULL, then VAF and read depth will be NAs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>name.of.VCF</code></td>
<td>
<p>Name of the VCF file. If <code>NULL</code>(default), this
function will remove all of the path up to and including the last path
separator (if any) in <code>file</code> and file path without extensions (and the
leading dot) will be used as the name of the VCF file.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tumor.col.name</code></td>
<td>
<p>Optional. Only applicable to <strong>Mutect</strong> VCF. Name
of the column in <strong>Mutect</strong> VCF which contains the tumor sample
information. It <strong>must</strong> have quotation marks. If
<code>tumor.col.name</code> is equal to <code>NA</code>(default), this function will
use the 10th column to calculate VAFs. See <code>GetMutectVAF</code> for
more details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filter.status</code></td>
<td>
<p>The status indicating a variant has passed all filters.
An example would be <code>"PASS"</code>. Variants which don't have the specified
<code>filter.status</code> in the <code>FILTER</code> column in VCF will be removed. If
<code>NULL</code>(default), no variants will be removed from the original VCF.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>get.vaf.function</code></td>
<td>
<p>Optional. Only applicable when <code>variant.caller</code> is
<strong>"unknown"</strong>. Function to calculate VAF(variant allele frequency) and read
depth information from original VCF. See <code>GetMutectVAF</code> as an example.
If <code>NULL</code>(default) and <code>variant.caller</code> is "unknown", then VAF
and read depth will be NAs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Optional arguments to <code>get.vaf.function</code>.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A data frame storing data lines of the VCF file with two additional
columns added which contain the VAF(variant allele frequency) and read
depth information.
</p>


</div>