<div class="container">

<table style="width: 100%;"><tr>
<td>CheckAndFixChrNamesForTransRanges</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Check and, if possible, correct the chromosome names in a trans.ranges <code>data.table</code>
</h2>

<h3>Description</h3>

<p>Check and, if possible, correct the chromosome names in a trans.ranges <code>data.table</code>
</p>


<h3>Usage</h3>

<pre><code class="language-R">CheckAndFixChrNamesForTransRanges(
  trans.ranges,
  vcf.df,
  ref.genome,
  name.of.VCF = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>trans.ranges</code></td>
<td>
<p>A <code>data.table</code> which contains
transcript range and strand information. Please refer to
<code>TranscriptRanges</code> for more details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>vcf.df</code></td>
<td>
<p>A VCF as a <code>data.frame</code>. Check the names in column
<code>CHROM</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ref.genome</code></td>
<td>
<p>The reference genome with the chromosome names to check
<code>vcf.df$CHROM</code> against; must be a Bioconductor
<code>BSgenome</code>, e.g.
<code>BSgenome.Hsapiens.UCSC.hg38</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>name.of.VCF</code></td>
<td>
<p>Name of the VCF file.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>If the <code>vcf.df$CHROM</code> values are correct or can be corrected,
then a vector of chromosome names that can be used as a replacement for
<code>trans.ranges$chrom</code>. If the names in <code>vcf.df$CHROM</code> cannot be
made to be consistent with the chromosome names in
<code>trans.ranges$chrom</code>, then <code>stop</code>.
</p>


</div>