<div class="container">

<table style="width: 100%;"><tr>
<td>CreateOneColIDMatrix</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Create one column of the matrix for an indel catalog from *one* in-memory VCF.</h2>

<h3>Description</h3>

<p>Create one column of the matrix for an indel catalog from *one* in-memory VCF.
</p>


<h3>Usage</h3>

<pre><code class="language-R">CreateOneColIDMatrix(
  ID.vcf,
  SBS.vcf = NULL,
  sample.id = "count",
  return.annotated.vcf = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ID.vcf</code></td>
<td>
<p>An in-memory VCF as a data.frame annotated by the
<code>AnnotateIDVCF</code> function. It must only
contain indels and must <strong>not</strong> contain SBSs
(single base substitutions), DBSs, or triplet
base substitutions, etc.
</p>
<p>One design decision for variant callers is the representation of "complex
indels", e.g. mutations e.g. CAT &gt; GC. Some callers represent this as C&gt;G,
A&gt;C, and T&gt;_. Others might represent it as CAT &gt; CG. Multiple issues can
arise. In PCAWG, overlapping indel/SBS calls from different callers were
included in the indel VCFs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>SBS.vcf</code></td>
<td>
<p>This argument defaults to <code>NULL</code> and
is not used. Ideally this should be an in-memory SBS VCF 
as a data frame. The rational is that for some data,
complex indels might be represented as an indel with adjoining
SBSs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sample.id</code></td>
<td>
<p>Usually the sample id, but defaults to "count".</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list of a 1-column ID matrix containing the mutation catalog
information and the annotated VCF with ID categories information added. If
some ID variants were excluded in the analysis, an additional element
<code>discarded.variants</code> will appear in the return list.
</p>


</div>