<div class="container">

<table style="width: 100%;"><tr>
<td>FindDelMH</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Return the length of microhomology at a deletion</h2>

<h3>Description</h3>

<p>Return the length of microhomology at a deletion
</p>


<h3>Usage</h3>

<pre><code class="language-R">FindDelMH(context, deleted.seq, pos, trace = 0, warn.cryptic = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>context</code></td>
<td>
<p>The deleted sequence plus ample surrounding
sequence on each side (at least as long as <code>del.sequence</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>deleted.seq</code></td>
<td>
<p>The deleted sequence in <code>context</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pos</code></td>
<td>
<p>The position of <code>del.sequence</code> in <code>context</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trace</code></td>
<td>
<p>If &gt; 0, then generate various 
messages showing how the computation is carried out.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>warn.cryptic</code></td>
<td>
<p>if <code>TRUE</code> generating a warning
if there is a cryptic repeat (see the example).</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function is primarily for internal use, but we export it
to document the underlying logic.
</p>
<p>Example:
</p>
<p><code>GGCTAGTT</code> aligned to <code>GGCTAGAACTAGTT</code> with
a deletion represented as:
</p>
<pre>

GGCTAGAACTAGTT
GG------CTAGTT GGCTAGTT GG[CTAGAA]CTAGTT
                           ----   ----
</pre>
<p>Presumed repair mechanism leading to this:
</p>
<pre>
  ....
GGCTAGAACTAGTT
CCGATCTTGATCAA

=&gt;

  ....
GGCTAG      TT
CC      GATCAA
        ....

=&gt;

GGCTAGTT
CCGATCAA

</pre>
<p>Variant-caller software can represent the
same deletion in several
different, but completely equivalent, ways.
</p>
<pre>

GGC------TAGTT GGCTAGTT GGC[TAGAAC]TAGTT
                          * ---  * ---

GGCT------AGTT GGCTAGTT GGCT[AGAACT]AGTT
                          ** --  ** --

GGCTA------GTT GGCTAGTT GGCTA[GAACTA]GTT
                          *** -  *** -

GGCTAG------TT GGCTAGTT GGCTAG[AACTAG]TT
                          ****   ****
</pre>
<p>This function finds:
</p>

<ol>
<li>
<p> The maximum match of undeleted sequence to the left
of the deletion that is
identical to the right end of the deleted sequence, and
</p>
</li>
<li>
<p> The maximum match of undeleted sequence to the right
of the deletion that
is identical to the left end of the deleted sequence.
</p>
</li>
</ol>
<p>The microhomology sequence is the concatenation of items
(1) and (2).
</p>
<p><strong>Warning</strong><br>
A deletion in a <em>repeat</em> can also be represented
in several different ways. A deletion in a repeat
is abstractly equivalent to a deletion with microhomology that
spans the entire deleted sequence. For example;
</p>
<pre>
GACTAGCTAGTT
GACTA----GTT GACTAGTT GACTA[GCTA]GTT
                        *** -*** -
</pre>
<p>is really a repeat
</p>
<pre>
GACTAG----TT GACTAGTT GACTAG[CTAG]TT
                        **** ----

GACT----AGTT GACTAGTT GACT[AGCT]AGTT
                        ** --** --
</pre>
<p><strong>This function only flags these
"cryptic repeats" with a -1 return; it does not figure
out the repeat extent.</strong>
</p>


<h3>Value</h3>

<p>The length of the maximum microhomology of <code>del.sequence</code>
in <code>context</code>.
</p>


<h3>ID classification</h3>

<p>See <a href="https://github.com/steverozen/ICAMS/blob/master/data-raw/PCAWG7_indel_classification_2021_09_03.xlsx">https://github.com/steverozen/ICAMS/blob/master/data-raw/PCAWG7_indel_classification_2021_09_03.xlsx</a>
for additional information on ID (small insertion and deletion) mutation
classification.
</p>
<p>See the documentation for <code>Canonicalize1Del</code> which first handles
deletions in homopolymers, then handles deletions in simple repeats with
longer repeat units, (e.g. <code>CACACACA</code>, see
<code>FindMaxRepeatDel</code>), and if the deletion is not in a simple
repeat, looks for microhomology (see <code>FindDelMH</code>).
</p>
<p>See the code for unexported function <code>CanonicalizeID</code>
and the functions it calls for handling of insertions.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># GAGAGG[CTAGAA]CTAGTT
#        ----   ----
FindDelMH("GGAGAGGCTAGAACTAGTTAAAAA", "CTAGAA", 8, trace = 0)  # 4

# A cryptic repeat
# 
# TAAATTATTTATTAATTTATTG
# TAAATTA----TTAATTTATTG = TAAATTATTAATTTATTG
# 
# equivalent to
#
# TAAATTATTTATTAATTTATTG
# TAAAT----TATTAATTTATTG = TAAATTATTAATTTATTG 
# 
# and
#
# TAAATTATTTATTAATTTATTG
# TAAA----TTATTAATTTATTG = TAAATTATTAATTTATTG  

FindDelMH("TAAATTATTTATTAATTTATTG", "TTTA", 8, warn.cryptic = FALSE) # -1
</code></pre>


</div>