<div class="container">

<table style="width: 100%;"><tr>
<td>CreateOneColDBSMatrix</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Create the matrix a DBS catalog for *one* sample from an in-memory VCF.</h2>

<h3>Description</h3>

<p>Create the matrix a DBS catalog for *one* sample from an in-memory VCF.
</p>


<h3>Usage</h3>

<pre><code class="language-R">CreateOneColDBSMatrix(vcf, sample.id = "count", return.annotated.vcf = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>vcf</code></td>
<td>
<p>An in-memory VCF file annotated with sequence context and
transcript information by function <code>AnnotateDBSVCF</code>. It must
*not* contain indels and must *not* contain SBS (single base
substitutions), or triplet base substitutions etc.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sample.id</code></td>
<td>
<p>Usually the sample id, but defaults to "count".</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list of three 1-column matrices with the names <code>catDBS78</code>,
<code>catDBS136</code>, and <code>catDBS144</code>. If trans.ranges is NULL,
<code>catDBS144</code> is not generated. Do not rely on the order of elements in
the list. If <code>return.annotated.vcf</code> = TRUE, another element
<code>annotated.vcf</code> will appear in the list. If there are DBS variants
whose tetranucleotide context contains "N", they will be excluded in the
analysis and an additional element <code>discarded.variants</code> will appear in
the return list.
</p>


<h3>Note</h3>

<p>DBS 144 catalog only contains mutations in transcribed regions.
</p>


</div>