<div class="container">

<table style="width: 100%;"><tr>
<td>AddAndCheckSBSClassInVCF</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Add and check SBS class in an annotated VCF with the corresponding SBS
mutation matrix</h2>

<h3>Description</h3>

<p>Add and check SBS class in an annotated VCF with the corresponding SBS
mutation matrix
</p>


<h3>Usage</h3>

<pre><code class="language-R">AddAndCheckSBSClassInVCF(vcf, mat96, mat1536, mat192 = NULL, sample.id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>vcf</code></td>
<td>
<p>An in-memory VCF file annotated with sequence context and
transcript information by function <code>AnnotateSBSVCF</code>. It must
*not* contain indels and must *not* contain DBS (double base
substitutions), or triplet base substitutions etc., even if encoded as
neighboring SBS.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mat96</code></td>
<td>
<p>The SBS96 mutation count matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mat1536</code></td>
<td>
<p>The SBS1536 mutation count matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mat192</code></td>
<td>
<p>The SBS192 mutation count matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sample.id</code></td>
<td>
<p>Usually the sample id, but defaults to "count".</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>The original <code>vcf</code> with three additional columns
<code>SBS96.class</code>, <code>SBS192.class</code> and <code>SBS1536.class</code> added.
</p>


</div>