<div class="container">

<table style="width: 100%;"><tr>
<td>VCFsToDBSCatalogs</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Create DBS catalogs from VCFs</h2>

<h3>Description</h3>

<p>Create a list of 3 catalogs (one each for DBS78, DBS144 and DBS136)
out of the contents in list.of.DBS.vcfs. The VCFs must not contain
any type of mutation other then DBSs.
</p>


<h3>Usage</h3>

<pre><code class="language-R">VCFsToDBSCatalogs(
  list.of.DBS.vcfs,
  ref.genome,
  num.of.cores = 1,
  trans.ranges = NULL,
  region = "unknown",
  return.annotated.vcfs = FALSE,
  suppress.discarded.variants.warnings = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>list.of.DBS.vcfs</code></td>
<td>
<p>List of in-memory data frames of pure DBS mutations
â€“ no SBS or 3+BS mutations. The list names will be the sample ids in the
output catalog.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ref.genome</code></td>
<td>
<p>A <code>ref.genome</code> argument as described in
<code>ICAMS</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>num.of.cores</code></td>
<td>
<p>The number of cores to use. Not available on Windows
unless <code>num.of.cores = 1</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trans.ranges</code></td>
<td>
<p>Optional. If <code>ref.genome</code> specifies one of the
<code>BSgenome</code> object
</p>

<ol>
<li> <p><code>BSgenome.Hsapiens.1000genomes.hs37d5</code>
</p>
</li>
<li> <p><code>BSgenome.Hsapiens.UCSC.hg38</code>
</p>
</li>
<li> <p><code>BSgenome.Mmusculus.UCSC.mm10</code>
</p>
</li>
</ol>
<p>then the function will infer <code>trans.ranges</code> automatically. Otherwise,
user will need to provide the necessary <code>trans.ranges</code>. Please refer to
<code>TranscriptRanges</code> for more details.
If <code>is.null(trans.ranges)</code> do not add transcript range
information.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>region</code></td>
<td>
<p>A character string designating a genomic region;
see <code>as.catalog</code> and <code>ICAMS</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>return.annotated.vcfs</code></td>
<td>
<p>Logical. Whether to return the annotated VCFs
with additional columns showing mutation class for each variant. Default is
FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>suppress.discarded.variants.warnings</code></td>
<td>
<p>Logical. Whether to suppress
warning messages showing information about the discarded variants. Default
is TRUE.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list containing the following objects:
</p>

<ul>
<li> <p><code>catDBS78</code>, <code>catDBS136</code>, <code>catDBS144</code>: Matrix of
3 DBS catalogs (one each for 78, 136, and 144).
</p>
</li>
<li> <p><code>discarded.variants</code>: <strong>Non-NULL only if</strong> there are variants
that were excluded from the analysis. See the added extra column
<code>discarded.reason</code> for more details.
</p>
</li>
<li> <p><code>annotated.vcfs</code>: <strong>Non-NULL only if</strong>
<code>return.annotated.vcfs</code> = TRUE. DBS VCF annotated by
<code>AnnotateDBSVCF</code> with three new columns <code>DBS78.class</code>,
<code>DBS136.class</code> and <code>DBS144.class</code> showing the mutation class for
each DBS variant.
</p>
</li>
</ul>
<p>If <code>trans.ranges</code> is not provided by user and cannot be inferred by
ICAMS, DBS 144 catalog will not be generated. Each catalog has
attributes added. See <code>as.catalog</code> for more details.
</p>


<h3>Comments</h3>

<p>To add or change attributes of the catalog, you can use function
<code>attr</code>. <br> For example, <code>attr(catalog, "abundance")
&lt;- custom.abundance</code>.
</p>


<h3>Note</h3>

<p>DBS 144 catalog only contains mutations in transcribed regions.
</p>


<h3>Examples</h3>

<pre><code class="language-R">file &lt;- c(system.file("extdata/Mutect-vcf",
                      "Mutect.GRCh37.s1.vcf",
                      package = "ICAMS"))
list.of.DBS.vcfs &lt;- ReadAndSplitMutectVCFs(file)$DBS
if (requireNamespace("BSgenome.Hsapiens.1000genomes.hs37d5", quietly = TRUE)) {
  catalogs.DBS &lt;- VCFsToDBSCatalogs(list.of.DBS.vcfs, ref.genome = "hg19",
                                    trans.ranges = trans.ranges.GRCh37,
                                    region = "genome")}
</code></pre>


</div>