<div class="container">

<table style="width: 100%;"><tr>
<td>GetStrandedKmerCounts</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Generate stranded k-mer abundance from a given genome and gene annotation file</h2>

<h3>Description</h3>

<p>Generate stranded k-mer abundance from a given genome and gene annotation file
</p>


<h3>Usage</h3>

<pre><code class="language-R">GetStrandedKmerCounts(
  k,
  ref.genome,
  stranded.ranges,
  filter.path,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>k</code></td>
<td>
<p>Length of k-mers (k&gt;=2)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ref.genome</code></td>
<td>
<p>A <code>ref.genome</code> argument as described in
<code>ICAMS</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>stranded.ranges</code></td>
<td>
<p>A keyed data table which has stranded ranges
information. It has four columns: chrom, start, end and strand. It should
use one-based coordinate system.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filter.path</code></td>
<td>
<p>If given, homopolymers will be masked from
genome(sequence). Only simple repeat masking is accepted now.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>If <code>TRUE</code> generate progress messages.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Matrix of the counts of each stranded k-mer across the <code>ref.genome</code>
</p>


</div>