<div class="container">

<table style="width: 100%;"><tr>
<td>CreateOneColSBSMatrix</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Create the matrix an SBS catalog for *one* sample from an in-memory VCF.</h2>

<h3>Description</h3>

<p>Create the matrix an SBS catalog for *one* sample from an in-memory VCF.
</p>


<h3>Usage</h3>

<pre><code class="language-R">CreateOneColSBSMatrix(vcf, sample.id = "count", return.annotated.vcf = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>vcf</code></td>
<td>
<p>An in-memory VCF file annotated with sequence context and
transcript information by function <code>AnnotateSBSVCF</code>. It must
*not* contain indels and must *not* contain DBS (double base
substitutions), or triplet base substitutions etc., even if encoded as
neighboring SBS.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sample.id</code></td>
<td>
<p>Usually the sample id, but defaults to "count".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>return.annotated.vcf</code></td>
<td>
<p>Whether to return the annotated VCF with
additional columns showing the mutation class for each variant. Default is
FALSE.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list of three 1-column matrices with the names
<code>catSBS96</code>, <code>catSBS192</code>, <code>catSBS1536</code>. If transcript
information is not available in <code>vcf</code>, <code>catSBS192</code> is not
generated. Do not rely on the order of elements in the list. If
<code>return.annotated.vcf</code> = TRUE, another element <code>annotated.vcf</code>
will appear in the list. If there are SBS variants whose pentanucleotide
context contains "N", they will be excluded in the analysis and an
additional element <code>discarded.variants</code> will appear in the return
list.
</p>


<h3>Note</h3>

<p>catSBS192 only contains mutations in transcribed regions.
</p>


</div>