<div class="container">

<table style="width: 100%;"><tr>
<td>all.abundance</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>K-mer abundances</h2>

<h3>Description</h3>

<p>An R list with one element each for 
<code>BSgenome.Hsapiens.1000genomes.hs37d5</code>, <br><code>BSgenome.Hsapiens.UCSC.hg38</code> and <code>BSgenome.Mmusculus.UCSC.mm10</code>.
Each element is in turn a sub-list keyed by 
<code>exome</code>, <code>transcript</code>, 
and <code>genome</code>. Each element of the sub list
is keyed by the number of rows in the catalog class (as a string, e.g. 
<code>"78"</code>, not <code>78</code>). The keys are:
78 (<code>DBS78Catalog</code>), 96 (<code>SBS96Catalog</code>), 136 (<code>DBS136Catalog</code>),
144 (<code>DBS144Catalog</code>), 192 (<code>SBS192Catalog</code>),
and 1536 (<code>SBS1536Catalog</code>). So, for example to get the exome
abundances for SBS96 catalogs for <code>BSgenome.Hsapiens.UCSC.hg38</code> exomes
one would reference <br><code>all.abundance[["BSgenome.Hsapiens.UCSC.hg38"]][["exome"]][["96"]]</code> <br>
or <code>all.abundance$BSgenome.Hsapiens.UCSC.hg38$exome$"96"</code>.
The value of the abundance is an integer vector with the K-mers
as names and each value being the count of that K-mer.
</p>


<h3>Usage</h3>

<pre><code class="language-R">all.abundance
</code></pre>


<h3>Format</h3>

<p>See Description.
</p>


<h3>Examples</h3>

<pre><code class="language-R">all.abundance$BSgenome.Hsapiens.UCSC.hg38$transcript$`144` 
#        AA        AC        AG        AT        CA        CC ... 
#  90769160  57156295  85738416  87552737  83479655  63267896 ...
# There are 90769160 AAs on the sense strands of transcripts in
# this genome.
</code></pre>


</div>