<div class="container">

<table style="width: 100%;"><tr>
<td>AddTranscript</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Add transcript information to a data frame with mutation records</h2>

<h3>Description</h3>

<p>Add transcript information to a data frame with mutation records
</p>


<h3>Usage</h3>

<pre><code class="language-R">AddTranscript(df, trans.ranges = NULL, ref.genome, name.of.VCF = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>df</code></td>
<td>
<p>A data frame storing mutation records of a VCF file.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trans.ranges</code></td>
<td>
<p>A <code>data.table</code> which contains
transcript range and strand information. Please refer to
<code>TranscriptRanges</code> for more details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ref.genome</code></td>
<td>
<p>A <code>ref.genome</code> argument as described in
<code>ICAMS</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>name.of.VCF</code></td>
<td>
<p>Name of the VCF file.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A data frame with new columns added to the input data frame,
which contain the mutated gene's name, range and strand information.
</p>


</div>