<div class="container">

<table style="width: 100%;"><tr>
<td>ReadMutectVCF</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Read in the data lines of a Variant Call Format (VCF) file created by Mutect</h2>

<h3>Description</h3>

<p>Read in the data lines of a Variant Call Format (VCF) file created by Mutect
</p>


<h3>Usage</h3>

<pre><code class="language-R">ReadMutectVCF(file, name.of.VCF = NULL, tumor.col.name = NA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p>The name/path of the VCF file, or a complete URL.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>name.of.VCF</code></td>
<td>
<p>Name of the VCF file. If <code>NULL</code>(default), this
function will remove all of the path up to and including the last path
separator (if any) in <code>file</code> and file path without extensions (and the
leading dot) will be used as the name of the VCF file.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tumor.col.name</code></td>
<td>
<p>Name of the column in VCF which contains the tumor
sample information. It <strong>must</strong> have quotation marks. If
<code>tumor.col.name</code> is equal to <code>NA</code>(default), this function will
use the 10th column to calculate VAFs. See <code>GetMutectVAF</code> for
more details.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A data frame storing data lines of a VCF file with two
additional columns added which contain the VAF(variant allele frequency)
and read depth information.
</p>


</div>