<div class="container">

<table style="width: 100%;"><tr>
<td>norm.data</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Normalize data</h2>

<h3>Description</h3>

<p>This function takes an object of class iCellR and normalized the data based on "global.glsf", "ranked.glsf" or "spike.in" methods.
</p>


<h3>Usage</h3>

<pre><code class="language-R">norm.data(
  x = NULL,
  norm.method = "ranked.glsf",
  top.rank = 500,
  spike.in.factors = NULL,
  rpm.factor = 1000,
  rounding.digits = 3,
  round.num = TRUE,
  ATAC.data = FALSE,
  ATAC.filter = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>An object of class iCellR.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>norm.method</code></td>
<td>
<p>Choose a normalization method, there are three option currently.
Choose from "global.glsf", "ranked.glsf","spike.in" or no.norm, default = "ranked.glsf".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>top.rank</code></td>
<td>
<p>If the method is set to "ranked.glsf", you need to set top number of genes sorted based on global base mean, default = 500.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>spike.in.factors</code></td>
<td>
<p>A numeric vector of spike-in values with the same cell id order as the main data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rpm.factor</code></td>
<td>
<p>If the norm.method is set to "rpm" the library sizes would be divided by this number, default = 1000 (higher numbers recomanded for bulk RNA-Seq).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rounding.digits</code></td>
<td>
<p>integer indicating the number of decimal places (round) or significant digits (signif) to be used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>round.num</code></td>
<td>
<p>Rounding of Numbers, default = FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ATAC.data</code></td>
<td>
<p>If TURE, it would normalize ATAC-Seq data and not RNA-Seq, default = FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ATAC.filter</code></td>
<td>
<p>If TURE, all the cells filtered in RNA-Seq will be filtered in ATAC-Seq. This needs to be done for both data to match,  default = TRUE.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>An object of class iCellR.
</p>


</div>