<div class="container">

<table style="width: 100%;"><tr>
<td>run.impute</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Impute the main data</h2>

<h3>Description</h3>

<p>This function takes an object of class iCellR and runs imputation on the main data.
</p>


<h3>Usage</h3>

<pre><code class="language-R">run.impute(
  x = NULL,
  imp.method = "iCellR.imp",
  dims = 1:10,
  nn = 10,
  ATAC.data = FALSE,
  rounding.digits = 4,
  round.num = TRUE,
  data.type = "pca",
  genes = "all_genes",
  k = 10,
  alpha = 15,
  t = "auto",
  npca = 100,
  init = NULL,
  t.max = 20,
  knn.dist.method = "euclidean",
  verbose = 1,
  n.jobs = 1,
  seed = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>An object of class iCellR.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>imp.method</code></td>
<td>
<p>Choose between "iCellR.imp" and "magic", defualt = "iCellR.imp".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dims</code></td>
<td>
<p>PC dimentions to be used for the analysis, default = 10.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nn</code></td>
<td>
<p>Number of neighboring cells to find, default = 10.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ATAC.data</code></td>
<td>
<p>If TURE, it would normalize ATAC-Seq data and not RNA-Seq, default = FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rounding.digits</code></td>
<td>
<p>integer indicating the number of decimal places (round) or significant digits (signif) to be used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>round.num</code></td>
<td>
<p>Rounding of Numbers, default = FALSE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>data.type</code></td>
<td>
<p>Choose between "tsne", "pca", "umap", "diffusion", "knetl", default = "pca".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genes</code></td>
<td>
<p>character or integer vector, default: NULL vector of column names or column indices for which to return smoothed data If 'all_genes' or NULL, the entire smoothed matrix is returned</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>k</code></td>
<td>
<p>if imp.method is magic; int, optional, default: 10 number of nearest neighbors on which to build kernel</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p>if imp.method is magic; int, optional, default: 15 sets decay rate of kernel tails. If NULL, alpha decaying kernel is not used</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>t</code></td>
<td>
<p>if imp.method is magic; int, optional, default: 'auto' power to which the diffusion operator is powered sets the level of diffusion. If 'auto', t is selected according to the Procrustes disparity of the diffused data.'</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>npca</code></td>
<td>
<p>number of PCA components that should be used; default: 100.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>init</code></td>
<td>
<p>magic object, optional object to use for initialization. Avoids recomputing intermediate steps if parameters are the same.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>t.max</code></td>
<td>
<p>if imp.method is magic; int, optional, default: 20 Maximum value of t to test for automatic t selection.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>knn.dist.method</code></td>
<td>
<p>string, optional, default: 'euclidean'. recommended values: 'euclidean', 'cosine' Any metric from 'scipy.spatial.distance' can be used distance metric for building kNN graph.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>'int' or 'boolean', optional (default : 1) If 'TRUE' or '&gt; 0', message verbose updates.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.jobs</code></td>
<td>
<p>'int', optional (default: 1) The number of jobs to use for the computation. If -1 all CPUs are used. If 1 is given, no parallel computing code is used at all, which is useful for debugging. For n_jobs below -1, (n.cpus + 1 + n.jobs) are used. Thus for n_jobs = -2, all CPUs but one are used</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seed</code></td>
<td>
<p>int or 'NULL', random state (default: 'NULL')</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>An object of class iCellR.
</p>


</div>