<div class="container">

<table style="width: 100%;"><tr>
<td>ICOMP</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Informational Complexity</h2>

<h3>Description</h3>

<p>These functions calculates Informational Complexity
(ICOMP) variants for "lm" and "glm" objects.
</p>


<h3>Usage</h3>

<pre><code class="language-R">ICOMP(model, type = "IFIM", C = "C1")

ICOMP_IFIM_CF(model)

ICOMP_IFIM_C1(model)

ICOMP_IFIM_C1F(model)

ICOMP_IFIM_C1R(model)

ICOMP_PEU_CF(model)

ICOMP_PEU_C1(model)

ICOMP_PEU_C1F(model)

ICOMP_PEU_C1R(model)

ICOMP_PEU_LN_CF(model)

ICOMP_PEU_LN_C1(model)

ICOMP_PEU_LN_C1F(model)

ICOMP_PEU_LN_C1R(model)

CICOMP_CF(model)

CICOMP_C1(model)

CICOMP_C1F(model)

CICOMP_C1R(model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>model</code></td>
<td>
<p>a "lm" or "glm" object</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>
<p>type of ICOMP. Available types are "IFIM", "PEU", "PEU_LN" and "CICOMP". Default is "IFIM".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>C</code></td>
<td>
<p>type of complexity. Available types are "CF", "C1", "C1F" and "C1R". Default is "C1".</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>ICOMP(IFIM) (Bozdogan, 2003) is calculated as
</p>
<p style="text-align: center;"><code class="reqn">-2LL(theta) + 2C(F^{-1})</code>
</p>

<p>ICOMP(IFIM-peu) (Koc and Bozdogan, 2015) as
</p>
<p style="text-align: center;"><code class="reqn">-2LL(theta) + k + 2C(F^{-1})</code>
</p>

<p>ICOMP(IFIM-peuln) (Bozdogan, 2010) as
</p>
<p style="text-align: center;"><code class="reqn">-2LL(theta) + k + 2log(n)C(F^{-1})</code>
</p>

<p>and CICOMP (Pamukcu et al., 2015) as
</p>
<p style="text-align: center;"><code class="reqn">-2LL(theta) + k(log(n) + 1) + 2C(F^{-1})</code>
</p>

<p><code class="reqn">F</code> is the fisher information matrix. <code class="reqn">F^{-1}</code> is the
reverse Fisher information matrix.
<code class="reqn">C</code> is the complexity measure. Four variants are available:
</p>
<p><code class="reqn">C_1</code> (Bozdogan, 2010) is
</p>
<p style="text-align: center;"><code class="reqn">C_1(F^{-1}) = s/2*log(lambda_a / lambda_g)</code>
</p>

<p><code class="reqn">C_F</code> (Bozdogan, 2010) is
</p>
<p style="text-align: center;"><code class="reqn">C_F(F^{-1}) = 1/s*sum_i^s(lambda_i - lambda_a)</code>
</p>

<p><code class="reqn">C_1F</code> (Bozdogan, 2010) is
</p>
<p style="text-align: center;"><code class="reqn">C_1F(F^{-1}) = 1/(4lambda_a^2)*sum_i^s(lambda_i - lambda_a)</code>
</p>

<p><code class="reqn">C_1R</code> (Bozdogan, 2000) is
</p>
<p style="text-align: center;"><code class="reqn">C_1R(F^{-1}) = 1/2*log(|R|)</code>
</p>

<p>Here, <code class="reqn">R</code> is the correlation matrix of the model, <code class="reqn">lambda_1, ..., lambda_s</code>
are eigenvalues of <code class="reqn">F</code>, <code class="reqn">lambda_a</code> and <code class="reqn">lambda_g</code> are arithmetic and
geometric mean of eigenvalues of <code class="reqn">F</code>, respectively. <code class="reqn">s</code> is the dimension
of <code class="reqn">F</code>.
While calculating the Fisher information matrix (<code class="reqn">F</code>), we used
the joint parameters (<code class="reqn">beta,sigma^2</code>) of the models. In <code class="reqn">C1R(.)</code> function,
we utilized the usual variance-covariance matrix <code class="reqn">Cov(beta)</code> of the
models. beta is the vector of regression coefficients.
</p>


<h3>Value</h3>

<p>Informational Complexity measurement of the model
</p>


<h3>References</h3>

<p>Bozdogan, H. (2003). Intelligent Statistical Data Mining with Information Complexity and Genetic Algorithms Hamparsum Bozdogan University of Tennessee, Knoxville, USA. In Statistical data mining and knowledge discovery (pp. 47-88). Chapman and Hall/CRC.
</p>
<p>Koc, E. K., &amp; Bozdogan, H. (2015). Model selection in multivariate adaptive regression splines (MARS) using information complexity as the fitness function. Machine Learning, 101(1), 35-58.
</p>
<p>Bozdogan, H. (2010). A new class of information complexity (ICOMP) criteria with an application to customer profiling and segmentation. İstanbul Üniversitesi İşletme Fakültesi Dergisi, 39(2), 370-398.
</p>
<p>Pamukçu, E., Bozdogan, H., &amp; Çalık, S. (2015). A novel hybrid dimension reduction technique for undersized high dimensional gene expression data sets using information complexity criterion for cancer classification. Computational and mathematical methods in medicine, 2015.
</p>
<p>Bozdogan, H. (2000). Akaike's information criterion and recent developments in information complexity. Journal of mathematical psychology, 44(1), 62-91.
</p>


<h3>Examples</h3>

<pre><code class="language-R">x1 &lt;- rnorm(100, 3, 2)
x2 &lt;- rnorm(100, 5, 3)
x3 &lt;- rnorm(100, 67, 5)
err &lt;- rnorm(100, 0, 4)

## round so we can use it for Poisson regression
y &lt;- round(3 + 2*x1 - 5*x2 + 8*x3 + err)

m1 &lt;- lm(y~x1 + x2 + x3)
m2 &lt;- glm(y~x1 + x2 + x3, family = "gaussian")
m3 &lt;- glm(y~x1 + x2 + x3, family = "poisson")

ICOMP_IFIM_CF(m1)
ICOMP_IFIM_CF(m2)
ICOMP_IFIM_CF(m3)
CICOMP_C1(m1)
CICOMP_C1(m2)
CICOMP_C1(m3)
ICOMP(m1, type = "PEU", C = "C1R")


</code></pre>


</div>