<div class="container">

<table style="width: 100%;"><tr>
<td>plotIdiograms</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot idiograms of karyotypes/genomes with and without centromere</h2>

<h3>Description</h3>

<p>This function reads a data.frame passed to <code>dfChrSize</code> with columns: <code>chrName</code> (mono/holo) and
<code>shortArmSize</code> and <code>longArmSize</code> for monocentrics or a column <code>chrSize</code> for holocentrics and produces a plot of
idiograms. If more
than one species, a column named <code>OTU</code> is needed.
</p>
<p>Optionally, it reads another data.frame passed to <code>dfMarkPos</code> with the position of
marks (sites). Examples: <code>markposDFs</code>. Another data.frame for mark characteristics
can be used <code>dfMarkColor</code> or a character vector passed to <code>mycolors</code>
</p>


<h3>Usage</h3>

<pre><code class="language-R"> plotIdiograms( dfChrSize, defaultFontFamily, revOTUs = FALSE,
 karHeight = 2, karHeiSpace = 2.5, karSepar = TRUE, amoSepar = 10,
 addMissingOTUAfter = NA, addMissingOTUBefore = NA, missOTUspacings = 0,
 moveKarHor = "", moveAllKarValueHor = 0, moveAllKarValueY = 0,
 karAnchorLeft = "", karAnchorRight = "", anchor = FALSE, anchorLineLty = 1,
 anchorText = "", anchorTextMParental, anchorTextMoveX = 0.5,
 anchorTextMoveY = 1, anchorTextMoveParenX = 0, anchorTextMoveParenY = 0,
 anchorVsizeF = 0.5, anchorHsizeF = 1, pchAnchor = 23, moveAnchorV = 0,
 moveAnchorH = 0, mkhValue = 0.5, n = 50, markN = 25, notes, leftNotes,
 leftNotesUp, notesTextSize = 1, leftNotesTextSize = 1, leftNotesUpTextSize = 1,
 notesLeft, notesPosX = 0.5, notesPosY = 0, leftNotesPosX = 0.5,
 leftNotesPosY = 0, leftNotesUpPosX = 0.5, leftNotesUpPosY = 0, noteFont = 1,
 leftNoteFont = 1, leftNoteFontUp = 1, parseTypes = TRUE, parseStr2lang = FALSE,
 propWidth = FALSE, MbThreshold = 10000, threshold = 35, MbUnit = "Mb",
 yTitle = "Âµm", specialyTitle = "cM", specialOTUNames = "", addOTUName = TRUE,
 OTUTextSize = 1, OTUfont, OTUfamily = "", OTUasNote = FALSE,
 OTUasLeftNote = FALSE, orderChr = "size", chrId = "original",
 classMbName = "Pm.", classcMName = "L.G.", classChrName = "Chr.",
 classChrNameUp = "Type", classGroupName = "", chrNameUp = FALSE,
 chrIdPatternRem, indexIdTextSize = 1, distTextChr = 1, groupUp = FALSE,
 groupName = TRUE, groupSepar = 0.5, chromatids = TRUE, arrowsBothChrt = TRUE,
 holocenNotAsChromatids = FALSE, excHoloFrArrToSide = FALSE, xModifier = 12,
 xModMonoHoloRate = 1, chrWidth = 0.5, chrSpacing = 0.5, specialChrWidth = 0.3,
 specialChrSpacing = 0.7, chrColor = "gray", chrBorderColor, centromereSize = 0,
 autoCenSize = TRUE, cenColor, fixCenBorder = NULL, gishCenBorder = FALSE,
 hideCenLines = 1.75, roundedCen, cenFormat = "rounded", cenFactor = 1,
 squareness = 4, lwd.chr = 0.5, lwd.cM, lwd.marks = 99, dfMarkPos,
 defaultStyleMark = "square", markDistType = "beg", protruding = 0.2,
 startPos = 0, pMarkFac = 0.25, origin = "b", efZero = 1e-05,
 cMBeginCenter = FALSE, arrowhead = 0.3, shrinkArrow = 0.3333,
 arrowheadWidthShrink = 0.1, arrowsToSide = TRUE, useOneDot = FALSE,
 dotsAsOval = FALSE, dfMarkColor, mycolors, alpha_val = 1,
 borderOfWhiteMarks = TRUE, colorBorderMark = "", lwd.mimicCen, defCenStyleCol,
 pattern = "", legend = "aside", remSimiMarkLeg = TRUE, bannedMarkName,
 bMarkNameAside = FALSE, forbiddenMark, legendWidth = 1.7, legendHeight = NA,
 markLabelSize = 1, markLabelSpacer = 1, legendYcoord = 0, markNewLine = NA,
 mylheight = 0.7, chrSize = FALSE, nsmall = 1, chrSizeMbp = FALSE, markPer = "",
 showMarkPos = FALSE, bToRemove = "", perAsFraction = FALSE, chrIndex = "both",
 morpho = "both", nameChrIndexPos = 2, karIndex = TRUE, karIndexPos = 0.5,
 ruler = TRUE, useMinorTicks = FALSE, miniTickFactor = 10, rulerPos = 0,
 ruler.tck = -0.02, rulerNumberPos = 0.5, rulerNumberSize = 1,
 collapseCen = TRUE, rulerInterval = 0, rulerIntervalcM = 0,
 rulerIntervalMb = 0, ceilingFactor = 0, xPosRulerTitle = 2.6,
 yPosRulerTitle = 0, rulerTitleSize = 1, xlimLeftMod = 1, xlimRightMod = 2,
 ylimBotMod = 0.2, ylimTopMod = 0.2, callPlot = TRUE, asp = 1,
 circularPlot = FALSE, verticalPlot = TRUE, karSpaceHor = 0, shrinkFactor = 0.9,
 separFactor = 1.5, labelSpacing = 0.7, labelOutwards = FALSE,
 chrLabelSpacing = 0.5, radius = 0.5, rotation = 0.5, circleCenter = 1,
 circleCenterY = 1, OTUlabelSpacing = 0.3, OTUsrt = 0, OTUplacing = "first",
 OTULabelSpacerx = 0, OTULabelSpacery = 0, OTUcentered = TRUE, OTUjustif = 0,
 OTUlegendHeight = NA, roundness, ... 
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>dfChrSize</code></td>
<td>
<p>mandatory data.frame, with columns: <code>OTU</code> (optional), <code>chrName</code> (mandatory),
<code>shortArmSize</code>, <code>longArmSize</code> for monocen. or <code>chrSize</code> for holocen.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>defaultFontFamily</code></td>
<td>
<p>character. use this as the font family. No default value.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>revOTUs</code></td>
<td>
<p>boolean, The order of species is the one in the main
data.frame, use <code>TRUE</code> to reverse</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>karHeight</code></td>
<td>
<p>numeric, vertical size of karyotypes. See also  <code>karHeiSpace</code>. Defaults to <code>2</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>karHeiSpace</code></td>
<td>
<p>numeric, vertical size of karyotypes including spacing. Use with <code>karSepar=FALSE</code>. Proportional to
<code>karHeight</code>, if overlap, increase. Defautl value <code>2.5</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>karSepar</code></td>
<td>
<p>boolean, reduce distance among karyotypes <code>FALSE</code> = equally
sized karyotypes or <code>TRUE</code> = equally spaced karyotypes. Incompatible with <code>addMissingOTUAfter</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>amoSepar</code></td>
<td>
<p>numeric, depends on <code>karSepar=TRUE</code>, if zero your
karyotypes will have no distance among them, if overlap,
increase this and <code>karHeiSpace</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>addMissingOTUAfter</code></td>
<td>
<p>character, when you want to add space (ghost OTUs) after one or several OTUs, pass the names of OTUs
preceding the desired space in a character vector i.e. <code>c("species one",</code>  <code>"species five")</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>addMissingOTUBefore</code></td>
<td>
<p>character, when you want to add space (ghost OTUs) before one or several OTUs, pass the names of OTUs after
the desired space in a character vector i.e. <code>c("species one",</code>  <code>"species five")</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>missOTUspacings</code></td>
<td>
<p>numeric, when you use <code>addMissingOTUAfter</code> this numeric vector should have the same length and corresponds
to the number of free spaces (ghost OTUs) to add after each OTU respectively</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>moveKarHor</code></td>
<td>
<p>character, OTUs' names of karyotypes that should be moved horizontally. See <code>mkhValue</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>moveAllKarValueHor</code></td>
<td>
<p>numeric, similar to <code>mkhValue</code>, but affects all karyotypes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>moveAllKarValueY</code></td>
<td>
<p>numeric, similar to <code>moveAllKarValueHor</code>, but affects y axis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>karAnchorLeft</code></td>
<td>
<p>character, OTUs' add anchor to the left of this OTU names of karyotypes. For <code>verticalPlot=FALSE</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>karAnchorRight</code></td>
<td>
<p>character, OTUs' add anchor to the right of this OTU names of karyotypes. For <code>verticalPlot=FALSE</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>anchor</code></td>
<td>
<p>boolean, when <code>TRUE</code>, plots a parent progeny structure in karyotypes in <code>moveKarHor</code>. Or a horizontal anchor
to the left/right
of <code>karAnchorLeft, karAnchorRight</code> when <code>verticalPlot=FALSE</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>anchorLineLty</code></td>
<td>
<p>numeric, type of line in <code>anchor</code>, corresponds to <code>lty</code>. Defaults to <code>1</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>anchorText</code></td>
<td>
<p>character, text to add to <code>anchor</code> structure near symbol. See <code>anchor</code>. Defaults to <code>""</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>anchorTextMParental</code></td>
<td>
<p>character, designed to fill with a character object the space left of a missing parental in the
<code>anchor</code> structure.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>anchorTextMoveX</code></td>
<td>
<p>numeric, for vertical plots with <code>anchorText</code> move text in X axis. Defaults to <code>0.5</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>anchorTextMoveY</code></td>
<td>
<p>numeric, for horizontal plots with <code>anchorText</code> move text in Y axis. Defaults to <code>1</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>anchorTextMoveParenX</code></td>
<td>
<p>numeric, for plots with <code>anchorTextMParental</code> move text in X axis. Defaults to <code>0</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>anchorTextMoveParenY</code></td>
<td>
<p>numeric, for plots with <code>anchorTextMParental</code> move text in Y axis. Defaults to <code>0</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>anchorVsizeF</code></td>
<td>
<p>numeric, factor to modify vertical size of anchor <code>0.5</code> (default). Size itself is equal to <code>karHeiSpace</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>anchorHsizeF</code></td>
<td>
<p>numeric, factor to modify horizontal size of anchor <code>1</code> (default).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pchAnchor</code></td>
<td>
<p>numeric, symbol for anchor, see <code>?points</code> and <code>anchor</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>moveAnchorV</code></td>
<td>
<p>numeric, displace anchor vertical portion to right or left. See <code>anchor</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>moveAnchorH</code></td>
<td>
<p>numeric, displace anchor horizontal portion to right or left. See <code>anchor</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mkhValue</code></td>
<td>
<p>numeric, value to move kar. hor. See <code>moveKarHor</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n, </code></td>
<td>
<p>numeric vertices number for round corners</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>markN, </code></td>
<td>
<p>numeric vertices number for round corners of marks</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>notes, </code></td>
<td>
<p>data.frame, or csv file name in quotes, (shown to the right of kar.), with columns <code>OTU</code> and <code>note</code> for adding
notes to each OTU, they appear to the right of chromosomes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>leftNotes, </code></td>
<td>
<p>data.frame, or csv file name in quotes (shown to the left), with columns <code>OTU</code> and <code>note</code> for adding notes
to each OTU, they appear to the left of chromosomes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>leftNotesUp, </code></td>
<td>
<p>data.frame, or csv file name in quotes, (shown to the left-up), similar to <code>leftNotes</code>, but intended for
placement over chr.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>notesTextSize</code></td>
<td>
<p>numeric, font size of notes, see <code>notes</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>leftNotesTextSize</code></td>
<td>
<p>numeric, font size of notes, see <code>leftNotes</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>leftNotesUpTextSize</code></td>
<td>
<p>numeric, font size of notes, see <code>leftNotesUp</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>notesLeft</code></td>
<td>
<p>deprecated, use a data.frame for <code>leftNotes</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>notesPosX</code></td>
<td>
<p>numeric, move right notes to the right or left (x axis)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>notesPosY</code></td>
<td>
<p>numeric, move right notes down or up (y axis)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>leftNotesPosX</code></td>
<td>
<p>numeric, move left notes to the right or left (x axis)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>leftNotesPosY</code></td>
<td>
<p>numeric, move left notes (<code>leftNotes</code>) down or up (y axis)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>leftNotesUpPosX</code></td>
<td>
<p>numeric, move up left notes to the right or left (x axis)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>leftNotesUpPosY</code></td>
<td>
<p>numeric, move up left notes (<code>leftNotesUp</code>) down or up (y axis)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>noteFont, </code></td>
<td>
<p>numeric  <code>1</code> for normal,  <code>2</code> for bold,  <code>3</code> for italics,  <code>4</code> for bold-italics. See <code>notes</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>leftNoteFont, </code></td>
<td>
<p>numeric  <code>1</code> for normal,  <code>2</code> for bold,  <code>3</code> for italics,  <code>4</code> for bold-italics. See
<code>leftNotes</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>leftNoteFontUp, </code></td>
<td>
<p>numeric  <code>1</code> for normal,  <code>2</code> for bold,  <code>3</code> for italics,  <code>4</code> for bold-italics. See
<code>leftNotesUp</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>parseTypes, </code></td>
<td>
<p>boolean, parse in <code>notes</code> the <em>Citrus</em> chr. types names. Creates subindex pos. for FL. Defaults to
<code>TRUE</code>. Incompatible with <code>parseStr2lang</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>parseStr2lang, </code></td>
<td>
<p>bolean, parse string in <code>notes</code> with function <code>str2lang(paste0("paste(",note,")") )</code> for ex:
<code>"italic('C. sinensis'), ' Author'"</code>. See <code>notes</code>, <code>leftNotes</code>,<code>leftNotesUp</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>propWidth, </code></td>
<td>
<p>boolean, defaults to <code>FALSE</code>. Diminishes chr. width with increasing number of OTUs</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>MbThreshold, </code></td>
<td>
<p>numeric, if greater than this number (defaults to <code>10000</code>), <code>MbUnit</code> will apply and <code>specialyTitle</code>
will not.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>threshold, </code></td>
<td>
<p>this is the max. value allowed for the main two significative digits, otherwise scale will shrink. For example,
after 35 <code class="reqn">\mu m</code> (Default), apparent size will be 3.5 and scale interval will change. See also <code>ceilingFactor</code>, you may have to
use <code>-1</code>. Introduced in 1.13</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>MbUnit, </code></td>
<td>
<p>character, text of units of title when <code>MbThreshold</code> met and OTU not in <code>specialOTUNames</code>. See
<code>specialyTitle</code>
Defaults to <code>"Mb"</code>, but anything can be used. Introduced in 1.13. See <code>specialyTitle</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>yTitle</code></td>
<td>
<p>character, units for common title. Defaults to <code class="reqn">\mu m</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>specialyTitle, </code></td>
<td>
<p>character, title of ruler if OTU is in <code>specialOTUNames</code>. Will not apply if <code>MbThreshold</code> met. In that
case use <code>MbUnit</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>specialOTUNames</code></td>
<td>
<p>character vector, normally title of ruler is micrometer or Mb (big numbers). Use this param. to be able to put
a different unit in ruler title. See <code>"specialyTitle"</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>addOTUName</code></td>
<td>
<p>boolean, when <code>TRUE</code> adds OTU (species) name to karyotype</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>OTUTextSize</code></td>
<td>
<p>numeric, font size of OTU name (species). Defaults to <code>1</code>. When <code>OTUasNote</code> is <code>TRUE</code>, use
<code>notesTextSize</code> instead</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>OTUfont</code></td>
<td>
<p>numeric, <code>1</code> for normal,  <code>2</code> for bold,  <code>3</code> for italics,  <code>4</code> for bold-italics</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>OTUfamily</code></td>
<td>
<p>character, font family for OTU name.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>OTUasNote</code></td>
<td>
<p>boolean, when <code>TRUE</code> adds OTU (species) name to the right, see <code>notes</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>OTUasLeftNote</code></td>
<td>
<p>boolean, when <code>TRUE</code> adds OTU (species) name to the left-up, see <code>leftNotesUp</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>orderChr</code></td>
<td>
<p>character, when <code>"size"</code>, sorts chromosomes by total
length from the largest to the smallest. <code>"original"</code>: preserves d.f. order. <code>"name"</code>: sorts alphabetically;
<code>"group"</code>: sorts by group name; <code>"chrNameUp"</code>: sorts according to column <code>chrNameUp</code>. See <code>chrNameUp</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chrId</code></td>
<td>
<p>character, print name of chromosome, <code>"original"</code> uses the original
name in OTU column of dfChrSize, <code>"simple"</code> (just 1 to ...) or <code>"none"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>classMbName</code></td>
<td>
<p>character, name of "chromosome" when in Mbp. Defaults to <code>"Pm"</code>. See <code>MbUnit</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>classcMName</code></td>
<td>
<p>character, name of "chromosome" when OTU in <code>specialOTUNames</code>. Defaults to <code>"L.G."</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>classChrName</code></td>
<td>
<p>character, name of "chromosome" when in micrometers (apparently). Defaults to <code>"Chr."</code>. See
<code>specialOTUnames</code>, <code>classMbName</code>, <code>classcMName</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>classChrNameUp</code></td>
<td>
<p>character, name of "chromosome" for col. <code>"chrNameUp"</code>. Defaults to <code>"Type"</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>classGroupName</code></td>
<td>
<p>character, name of groups. Defaults to <code>""</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chrNameUp</code></td>
<td>
<p>boolean, when <code>TRUE</code> adds secondary chromosome name from col. <code>chrNameUp</code> over chrs. Defaults to <code>FALSE</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chrIdPatternRem</code></td>
<td>
<p>character, regex pattern to remove from chr. names</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>indexIdTextSize</code></td>
<td>
<p>numeric, font size of chr. and kar. indices and
chromosome name. Defaults to <code>1</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>distTextChr</code></td>
<td>
<p>numeric, distance from name of chromosome to chromosome,
also affects vertical separation of indices. Defaults to <code>1</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>groupUp</code></td>
<td>
<p>boolean, when <code>TRUE</code> when groups present, they appear over the chr. name. Defaults to <code>FALSE</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>groupName</code></td>
<td>
<p>boolean, when <code>TRUE</code> (default), shows group names. When <code>FALSE</code> only line</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>groupSepar</code></td>
<td>
<p>numeric, factor for affecting chr. spacing <code>chrSpacing</code> among groups. Defaults to <code>0.5</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chromatids</code></td>
<td>
<p>boolean, when <code>TRUE</code> shows separated chromatids. Defaults to <code>TRUE</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>arrowsBothChrt</code></td>
<td>
<p>boolean, when <code>TRUE</code> (default) (for <code>chromatids=TRUE</code>) shows <code>upArrow,</code>  <code> downArrow</code>
styles of marks in both chromatids when <code>arrowsToSide=TRUE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>holocenNotAsChromatids</code></td>
<td>
<p>boolean, when <code>TRUE</code> and <code>chromatids=TRUE</code> does not plot holocen kar. with chromatids.
Defaults to <code>FALSE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>excHoloFrArrToSide</code></td>
<td>
<p>boolean, when <code>arrowsToSide=TRUE</code>, excludes holocen. from this behaviour, plotting a centered arrow only.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xModifier</code></td>
<td>
<p>numeric, for <code>chromatids=TRUE</code>, separation among chromatids. Quotient for <code>chrWidth</code>. Defaults to <code>12
: chrWidth/12</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xModMonoHoloRate</code></td>
<td>
<p>numeric, factor to shrink chromatid separ. for holocen. 5 means 5 times smaller (quotient).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chrWidth</code></td>
<td>
<p>numeric, relative chromosome width. Defaults to <code>0.5</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chrSpacing</code></td>
<td>
<p>numeric, horizontal spacing among chromosomes, see also  <code>chrWidth</code>. Defaults to <code>0.5</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>specialChrWidth</code></td>
<td>
<p>numeric, relative chromosome width. Defaults to <code>0.5</code> for OTUs in <code>specialOTUNames</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>specialChrSpacing</code></td>
<td>
<p>numeric, horizontal spacing among chromosomes for OTUs in <code>specialOTUNames</code>, see also  <code>chrWidth</code>.
Defaults to <code>0.5</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chrColor</code></td>
<td>
<p>character, main color for chromosomes. Defaults to <code>"gray"</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chrBorderColor</code></td>
<td>
<p>character, color for border of chromosomes, defaults to <code>chrColor</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>centromereSize</code></td>
<td>
<p>numeric, optional, this establishes the apparent size of cen. in the plot in <code class="reqn">\mu</code>m. See
<code>autoCenSize=TRUE</code>. Default: <code>0</code>. Use with <code>autoCenSize=FALSE</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>autoCenSize</code></td>
<td>
<p>boolean, when <code>TRUE</code> ignores <code>centromereSize</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cenColor</code></td>
<td>
<p>character, color for centromeres, if GISH use <code>NULL</code> or <code>NA</code>. Defaults to <code>chrColor</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fixCenBorder</code></td>
<td>
<p>boolean, when <code>TRUE</code> uses <code>chrColor</code> as centromere (and cen. mark) border color. See also <code>cenColor</code>,
<code>chrColor</code>, <code>colorBorderMark</code>, <code>borderOfWhiteMarks</code>. No default value. When <code>chrColor</code> is <code>"white"</code> this
turns into <code>"black"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gishCenBorder</code></td>
<td>
<p>boolean, when <code>TRUE</code>, cen. mark border color is the same as mark color, ignoring <code>colorBorderMark</code>. No
default.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hideCenLines</code></td>
<td>
<p>numeric, factor to multiply line width (lwd) used for covering cen. border, when <code>chrColor</code> is <code>white</code>
or when <code>gishCenBorder=TRUE</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>roundedCen</code></td>
<td>
<p>deprecated, see cenFormat</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cenFormat</code></td>
<td>
<p>boolean, when <code>"triangle"</code>, cen. has triangular aspect. When <code>"rounded"</code>, it has rounded aspect (Default).
<code>"inProtein"</code> for using the mark with style of same name.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cenFactor</code></td>
<td>
<p>numeric, modifies any cen. mark and cen. size. Defaults to <code>1</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>squareness</code></td>
<td>
<p>numeric, shape of vertices of chromosomes and square marks,
higher values more squared. Defaults to <code>4</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lwd.chr</code></td>
<td>
<p>thickness of border of chr., some marks and ruler. Thick of <code>cM</code> marks when <code>lwd.cM</code> absent and other marks
when <code>lwd.marks</code> absent. Defaults to <code>0.5</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lwd.cM</code></td>
<td>
<p>thickness of cM marks. Defaults to <code>lwd.chr</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lwd.marks</code></td>
<td>
<p>thickness of most marks. Except <code>cM</code> marks and centr. related marks. See <code>lwd.chr</code>, <code>lwd.cM</code>. Defaults
to <code>lwd.chr</code> value when <code>99</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dfMarkPos</code></td>
<td>
<p>data.frame of marks (sites): columns: <code>OTU</code> (opt), <code>chrName</code>,
<code>markName</code> (name of site), <code>chrRegion</code> (for monocen. and opt for whole arm (w) in holocen.), <code>markDistCen</code> (for
monocen.),
<code>markPos</code> (for holocen.), <code>markSize</code>; column <code>chrRegion</code>:
use <code>p</code> for short arm, <code>q</code> for long arm, <code>cen</code> for centromeric mark and <code>w</code> for whole chr. mark; column
<code>markDistCen</code>: use distance from
centromere to mark, not necessary for cen. marks (cen), w, p, q (when whole arm). See also param. <code>markDistType</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>defaultStyleMark</code></td>
<td>
<p>character, default style of mark, only used when <code>style</code> column of <code>dfMarkColor</code> data.frame is
missing or in absence of this data.frame. Use <code>"square"</code> (default), <code>"squareLeft"</code>, <code>"dots"</code>, <code>"cM"</code>,
<code>"cMLeft"</code>,<code>"cenStyle"</code>, <code>"upArrow"</code>, <code>"downArrow"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>markDistType</code></td>
<td>
<p>character, if <code>"cen"</code> = the distance you provided in data.frame (<code>dfMarkPos</code>) column <code>markDistCen</code>
or <code>markPos</code>  is to the center of the mark, if <code>"beg"</code> = the distance you provided is to the
beginning of the mark (Default)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>protruding</code></td>
<td>
<p>numeric, when style of mark is <code>"cM"</code>, fraction of chrWidth to stretch marker. Defaults to <code>0.2</code>.
Introduced in 1.13</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>startPos</code></td>
<td>
<p>numeric, factor to increase separation of <code>exProtein</code> marks to chromosome. Defaults to <code>0</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pMarkFac</code></td>
<td>
<p>numeric, fraction of chr. size for <code>exProtein</code> style marks. Defaults to <code>0.25</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>origin, </code></td>
<td>
<p>For non-monocentric chr. (for holocentrics only) Use <code>"b"</code> (default) if distance to mark in (<code>"markPos"</code>
column in <code>"dfMarkPos"</code>) data.frame measured from bottom of chromosome, use <code>"t"</code> for distance to mark from top of chr.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>efZero, </code></td>
<td>
<p>numeric, numbers below this one will be considered as zero, for comparison purposes. Defaults to <code>1e-5</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cMBeginCenter, </code></td>
<td>
<p>boolean, start position of <code>cM</code> and <code>cMLeft</code> marks. If <code>TRUE</code>, starts in the center (width) of
chr. . Defaults to <code>FALSE</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>arrowhead</code></td>
<td>
<p>numeric, proportion of head of arrow (mark styles: <code>upArrow,</code>  <code>downArrow</code>). Defaults to <code>0.3</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>shrinkArrow</code></td>
<td>
<p>numeric, proportion, shrinks body of arrow. Defaults to <code>0.3333</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>arrowheadWidthShrink</code></td>
<td>
<p>numeric, proportion, shrinks head of arrow. Defaults to <code>0.1</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>arrowsToSide</code></td>
<td>
<p>boolean, when <code>FALSE</code> use a centered arrow, instead of an arrow next to chr. margins (<code>TRUE</code>, default).
See <code>arrowsBothChrt</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>useOneDot</code></td>
<td>
<p>boolean, use one dot instead of two in style of marks <code>dots</code>. Defaults to <code>FALSE</code>. Not useful for
<code>chromatids=TRUE</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dotsAsOval</code></td>
<td>
<p>boolean, use oval instead of two dots in style of marks <code>dots</code>. Defaults to <code>FALSE</code>. See <code>useOneDot</code>.
Not useful for <code>chromatids=TRUE</code> or <code>circularPlot=TRUE</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dfMarkColor</code></td>
<td>
<p>data.frame, optional, specifying colors and style for marks (sites);
columns: <code>markName</code>, <code>markColor</code>, <code>style</code>. <code>style</code> accepts: <code>square</code>, <code>squareLeft</code>, <code>dots</code>,
<code>cM</code>,
<code>"cMLeft"</code>, <code>"cenStyle"</code>, <code>"upArrow"</code>, <code>"downArrow"</code>, <code>"exProtein"</code>.
(if column <code>style</code> missing all (except 5S) are plotted as in param. <code>defaultStyleMark</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mycolors</code></td>
<td>
<p>character vector, optional, i.e. <code>c("blue",</code> <code>"red",</code> <code>"green")</code> for specifying color of marks in order
of appearance. if diverges with number of marks will be recycled if <code>dfMarkColor</code> present, mycolors will be ignored. To know the
order of your marks use something like: <code>unique(dfMarkPos$markName)</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha_val</code></td>
<td>
<p>numeric vector, make marks transparent, accepts values from 0 to 1, @see <code>scales::alpha</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>borderOfWhiteMarks</code></td>
<td>
<p>boolean, if <code>TRUE</code> (Default) uses black border for white marks. See <code>dfMarkColor</code>. Does not apply
to marks with style <code>cenStyle</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>colorBorderMark</code></td>
<td>
<p>character, without default, pass a name of a color to use as border of marks. See <code>borderOfWhiteMarks</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lwd.mimicCen</code></td>
<td>
<p>thickness of lines of <code>cenStyle</code> marks; affects only lateral borders. Defaults to <code>lwd.chr</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>defCenStyleCol</code></td>
<td>
<p>character, color of outer part of <code>cenStyle</code> marks. Defaults to <code>white</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pattern</code></td>
<td>
<p>REGEX pattern to remove from names of marks</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>legend</code></td>
<td>
<p>character, <code>"none"</code> for no legend; <code>"inline"</code> prints labels near
chromosomes; <code>"aside"</code> prints legend to the right of karyotypes (default). See <code>markLabelSpacer</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>remSimiMarkLeg</code></td>
<td>
<p>boolean, when <code>legend="aside"</code>, if you use <code>pattern</code>, you can have several marks with same name. When
<code>TRUE</code> this remove this pseudoduplicates from legend. Be sure that this pseudoduplicates have the same color, otherwise you should
use <code>FALSE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bannedMarkName</code></td>
<td>
<p>character, character string or vector with mark names to be removed from plot. Not the marks but the labels.
Except when <code>bMarkNameAside</code> is used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bMarkNameAside</code></td>
<td>
<p>boolean, when <code>TRUE</code> and <code>legend="inline"</code>, shows marks in <code>bannedMarkName</code> as
<code>legend="aside"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>forbiddenMark, </code></td>
<td>
<p>character, character string or vector with mark names to be removed from plot. Not the marks but the labels.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>legendWidth</code></td>
<td>
<p>numeric, factor to increase width of squares and of legend. Defaults to <code>1.7</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>legendHeight</code></td>
<td>
<p>numeric, factor to increase height of squares and dots of legend. Automatic.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>markLabelSize</code></td>
<td>
<p>numeric, only if legend != (not) "", size of the font of
labels of marks (legend). Defaults to <code>1</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>markLabelSpacer</code></td>
<td>
<p>numeric, only if <code>legend="aside"</code>, space from the
rightmost chr. to legend. Defaults to <code>1</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>legendYcoord</code></td>
<td>
<p>numeric, modify Y position of legend when <code>legend="aside"</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>markNewLine, </code></td>
<td>
<p>character, character to split mark Names into different lines. Applies to <code>square</code> marks. Defaults to <code>NA</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mylheight, </code></td>
<td>
<p>numeric, for <code>markNewLine!=NA</code>; is equivalent to <code>lheight</code> of <code>par</code>: "The line height multiplier. The
height of a line of text (used to vertically space multi-line text) is found by multiplying the character height both by the current
character expansion and by the line height multiplier." Defaults to <code>0.7</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chrSize</code></td>
<td>
<p>boolean, when <code>TRUE</code> adds total chr size under each chr. Defaults to <code>FALSE</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nsmall</code></td>
<td>
<p>numeric, rounding decimals for <code>chrSize</code> parameter. Defaults to <code>1</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chrSizeMbp</code></td>
<td>
<p>boolean, when <code>TRUE</code> adds total Mbp chr. size to each chr. provided, there is a <code>Mbp</code> column in
<code>dfChrSize</code> data.frame. Defaults to <code>FALSE</code>. If data in columns <code>shortArmSize</code>, or col. <code>chrSize</code> is in millions
("Mbp"). Use <code>chrSize=TRUE</code> not this one (not column <code>Mbp</code>, you don't need this).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>markPer</code></td>
<td>
<p>character vector, name of mark(s) to calculate % of mark in chr. and add it to plot. See <code>perAsFraction</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>showMarkPos</code></td>
<td>
<p>boolean, adds position of marks under karyotype (fraction 0-1) when <code>TRUE</code>. Defaults to <code>FALSE</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bToRemove, </code></td>
<td>
<p>character vector, bands to remove from calc. of pos., when <code>showMarkPos = TRUE</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>perAsFraction</code></td>
<td>
<p>boolean, when <code>TRUE</code> % is shown as fraction. Defaults to <code>FALSE</code>. See <code>markPer</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chrIndex</code></td>
<td>
<p>character, add arm ratio with <code>"AR"</code> and centromeric index with <code>"CI"</code>, or <code>"both"</code> (Default), or
<code>"none"</code> for none</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>morpho</code></td>
<td>
<p>character, when <code>"both"</code> (default) prints the Guerra and Levan classif of cen. position, use also <code>"Guerra"</code>
or  <code>"Levan"</code> or <code>"none"</code> for none. See also <code>?armRatioCI</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nameChrIndexPos</code></td>
<td>
<p>numeric, modify position of name of chr. indices</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>karIndex</code></td>
<td>
<p>logical, add karyotype indices A (intrachromosomal -
centromere pos.) and A2 (interchromosomal asymmetry, variation among
chromosome sizes)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>karIndexPos</code></td>
<td>
<p>numeric, move karyotype index. Defaults to <code>0.5</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ruler</code></td>
<td>
<p>boolean, display ruler to the left of karyotype, when <code>FALSE</code> no ruler</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>useMinorTicks</code></td>
<td>
<p>boolean, display minor ticks between labeled ticks in ruler. See <code>miniTickFactor</code>. Defaults to <code>FALSE</code>.
(ticks without label)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>miniTickFactor</code></td>
<td>
<p>numeric, number of minor ticks for each labeled tick. See <code>useMinorTicks</code>. Defaults to <code>10</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rulerPos</code></td>
<td>
<p>numeric, absolute position of ruler, corresponds to <code>pos</code>
argument of <code>axis</code> R plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ruler.tck</code></td>
<td>
<p>numeric, tick size of ruler, corresponds to <code>tck</code> argument of
<code>axis</code> R plot. Defaults to <code>-0.02</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rulerNumberPos</code></td>
<td>
<p>numeric, modify position of numbers of ruler. Defaults to <code>0.5</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rulerNumberSize</code></td>
<td>
<p>numeric, size of number's font in ruler. Defaults to <code>1</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>collapseCen</code></td>
<td>
<p>boolean, avoid spacing in ruler between short arm and long arm.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rulerInterval</code></td>
<td>
<p>numeric, intervals in ruler. No default, automatic.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rulerIntervalcM</code></td>
<td>
<p>numeric, intervals in ruler of OTU in <code>specialOTUNames</code>. No default. Introduced in 1.13</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rulerIntervalMb</code></td>
<td>
<p>numeric, intervals in ruler of OTU with data in Mb (&gt;<code>MbThreshold</code>) and absent from
<code>specialOTUNames</code>. No default. Usa data in millions</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ceilingFactor</code></td>
<td>
<p>numeric, affects number of decimals for ceiling. Affects max. value of ruler. Defaults to <code>0</code>. When
<code>threshold</code> is greater than <code>35</code> this may have to be negative. Introduced in 1.13</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xPosRulerTitle, </code></td>
<td>
<p>numeric, modify position of ruler title. See <code>yTitle,</code>  <code> specialyTitle,</code>  <code>
MbUnit</code>. Defaults to <code>2.6</code>. A value of <code>2.6</code> means <code>2.6</code> times the value of <code>chrSpacing</code> to the left, from the first
chr.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>yPosRulerTitle, </code></td>
<td>
<p>numeric, affects vertical position of ruler title. Defaults to <code>0</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rulerTitleSize, </code></td>
<td>
<p>numeric font size of units of ruler. See also <code>xPosRulerTitle</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xlimLeftMod</code></td>
<td>
<p>numeric, modifies <code>xlim</code> left argument of plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xlimRightMod</code></td>
<td>
<p>numeric, <code>xlim</code> right side modification by adding space to the right
of idiograms. Defaults to <code>2</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ylimBotMod</code></td>
<td>
<p>numeric, modify <code>ylim</code> bottom argument of plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ylimTopMod</code></td>
<td>
<p>numeric, modify <code>ylim</code> top argument of plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>callPlot</code></td>
<td>
<p>boolean, create new plot in your device. Defaults to <code>TRUE</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>asp, </code></td>
<td>
<p>numeric, y x aspect of plot. Defaults to <code>1</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>circularPlot</code></td>
<td>
<p>boolean, if <code>TRUE</code> chromosomes/karyotypes are plotted in concentric circles. Defaults to <code>FALSE</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verticalPlot</code></td>
<td>
<p>boolean, when <code>TRUE</code> karyotypes are plotted vertically, otherwise, horizontally Defaults to <code>TRUE</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>karSpaceHor</code></td>
<td>
<p>numeric, separation among horizontal karyotypes. When <code>verticalPlot=FALSE</code>. Defaults to <code>0</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>shrinkFactor</code></td>
<td>
<p>numeric, for <code>circularPlot=TRUE</code> percentage of usage of circle. Defaults to <code>0.9</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>separFactor</code></td>
<td>
<p>numeric, for <code>circularPlot=TRUE</code> modify separation of concentric karyotypes. Defaults to <code>1.5</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>labelSpacing</code></td>
<td>
<p>numeric, for <code>circularPlot=TRUE</code>. Spacing of mark labels. Defaults to <code>0.7</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>labelOutwards</code></td>
<td>
<p>boolean, inline labels projected outwards</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chrLabelSpacing</code></td>
<td>
<p>numeric, for <code>circularPlot=TRUE</code>. Spacing of chr. labels. Defaults to <code>0.5</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>radius</code></td>
<td>
<p>numeric, for <code>circularPlot=TRUE</code>. Affects radius of karyotypes. Defaults to <code>0.5</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rotation</code></td>
<td>
<p>numeric, anti-clockwise rotation, defaults to <code>0.5</code> which rotates chr. from top to -90 degrees. (-0.5*<code class="reqn">\pi</code> )</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>circleCenter</code></td>
<td>
<p>numeric, for <code>circularPlot=TRUE</code>. Coordinate X of center of circles. Affects <code>legend="aside"</code> position.
Defaults to <code>1</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>circleCenterY</code></td>
<td>
<p>numeric, for <code>circularPlot=TRUE</code>. Coordinate Y of center of circles. Affects <code>legend="aside"</code> position.
Defaults to <code>1</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>OTUlabelSpacing</code></td>
<td>
<p>numeric, for <code>circularPlot=TRUE</code>. Spacing for OTU names. Defaults to <code>0.3</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>OTUsrt</code></td>
<td>
<p>numeric, for <code>circularPlot=TRUE</code>. Angle to use for OTU names. Defaults to <code>0</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>OTUplacing</code></td>
<td>
<p>character, for <code>circularPlot=TRUE</code>. location of OTU name. Defaults to <code>"first"</code> plots name near
first chr. <code>"number"</code> places number near 1st chr. and index and name to the right or center.
<code>"simple"</code> place name to the right or center without numbering. See also <code>OTUcentered</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>OTULabelSpacerx</code></td>
<td>
<p>numeric, for <code>circularPlot=TRUE</code> and <code>OTUplacing="number" or "simple"</code>. Modifies x names position</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>OTULabelSpacery</code></td>
<td>
<p>numeric, for <code>circularPlot=TRUE</code> and <code>OTUplacing="number" or "simple"</code>. Modifies y names position</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>OTUcentered</code></td>
<td>
<p>boolean, for <code>circularPlot=TRUE</code> and <code>OTUplacing="number" or "simple"</code>. OTU name in center of circle
when <code>TRUE</code>, otherwise, to the right.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>OTUjustif</code></td>
<td>
<p>numeric, for <code>circularPlot=TRUE</code> and <code>OTUplacing="number" or "simple"</code>. Justification of OTU name. <code>0</code>
= left
(Default); use <code>0.5</code> for centered. See <code>?text</code> -&gt; <code>adj</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>OTUlegendHeight</code></td>
<td>
<p>numeric, for <code>circularPlot=TRUE</code> and <code>OTUplacing="number" or "simple"</code>. Modifies y names separation</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>roundness</code></td>
<td>
<p>deprecated, use <code>squareness</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>accepts other arguments for the plot, see, <code>?plot</code></p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>plot
</p>


<h3>See Also</h3>

<p><code>asymmetry</code>
</p>
<p><code>armRatioCI</code>
</p>
<p><code>chrbasicdatamono</code>
</p>
<p><code>chrbasicdataHolo</code>
</p>
<p><code>markposDFs</code>
</p>
<p><code>markdataholo</code>
</p>
<p><code>dfMarkColor</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(dfOfChrSize)
plotIdiograms(dfOfChrSize, ylimBotMod = .75, rulerPos = -.5)
plotIdiograms(dfOfChrSize, circularPlot = TRUE, chrLabelSpacing = 1)
plotIdiograms(dfChrSizeHolo, rulerPos = -.5)
</code></pre>


</div>