<div class="container">

<table style="width: 100%;"><tr>
<td>namesToColumns</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>FUNCTION that modifies marks' names into columns</h2>

<h3>Description</h3>

<p>Reads a data.frame with marks' of styles
<code>downArrow,upArrow,cM,cMLeft</code>
positions. It separates names in columns, avoiding overlap when multiple
close names
</p>
<p>Exceptionally this function requires the column style in the
data.frame of marks'
positions.
</p>
<p>Returns a data.frame
</p>


<h3>Usage</h3>

<pre><code class="language-R"> namesToColumns( marksDf, dfChrSize, markType = c("downArrow", "upArrow",
 "cMLeft", "cM"), amountofSpaces = 13, colNumber = 2, protruding = 0.2,
 protrudingInt = 0.5, circularPlot = TRUE, rotation = 0.5,
 defaultStyleMark = "square", orderChr = "size", halfModDown = NA,
 halfModUp = NA, rotatMod = 0 
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>marksDf</code></td>
<td>
<p>data.frame with columns: <code>markName,style,markPos</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dfChrSize</code></td>
<td>
<p>data.frame, size of chr. Same of plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>markType</code></td>
<td>
<p>character, use
<code>c("downArrow","upArrow","cM","cMLeft")</code> or a subset</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>amountofSpaces</code></td>
<td>
<p>numeric, number of spaces for each column</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>colNumber</code></td>
<td>
<p>numeric, number of columns</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>protruding</code></td>
<td>
<p>numeric, same as plot, minimal protruding for arrow
marks, equivalent to cM
protruding</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>protrudingInt</code></td>
<td>
<p>numeric, spacing of columns in terms of width of chr.
percent 1 = 100%.
Defaults to <code>0.5</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>circularPlot</code></td>
<td>
<p>boolean, use <code>TRUE</code> for circular plots. Use
<code>FALSE</code> otherwise</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rotation</code></td>
<td>
<p>numeric, same as plot, anti-clockwise rotation, defaults to
<code>0.5</code> which
rotates chr. from top to -90 degrees. (-0.5*<code class="reqn">\pi</code> )</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>defaultStyleMark</code></td>
<td>
<p>character, if some data in column style missing
fill with this one.
Defaults to <code>"square"</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>orderChr</code></td>
<td>
<p>character, replaces <code>orderBySize - deprecated</code> when <code>"size"</code>, sorts chromosomes by total
length from the largest to the smallest. <code>"original"</code>: preserves d.f. order. <code>"name"</code>:
sorts alphabetically; <code>"group"</code>: sorts by group name</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>halfModDown</code></td>
<td>
<p>numeric, for circ. plots, when plotting several
chromosomes in a circular
plot, using a small value <code>0.05</code> corrects for alignment problems of
<code>downArrows,
cMLeft</code> labels. Defaults to <code>NA</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>halfModUp</code></td>
<td>
<p>numeric, for circ. plots, when plotting several
chromosomes in a circular
plot, using a small value <code>0.05</code> corrects for alignment problems of
<code>upArrows, cM</code>
labels. Defaults to <code>NA</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rotatMod</code></td>
<td>
<p>numeric, for circ. plots, when rotation != 0 (diff.),
corrects alignment of
labels. Defaults to <code>0</code></p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>data.frame
</p>


</div>