<div class="container">

<table style="width: 100%;"><tr>
<td>GetGDEscore</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>GetGDEscore</h2>

<h3>Description</h3>

<p>Function "GetGDEscore" is used to calculate gene differential expression levels.
</p>


<h3>Usage</h3>

<pre><code class="language-R">GetGDEscore(ExpData,Label)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ExpData</code></td>
<td>
<p>A gene expression profile of interest (rows are genes, columns are samples).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Label</code></td>
<td>
<p>A character vector consists of "0" and "1" which represent sample class in the gene expression profile. "0" means normal sample and "1" means disease sample.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A matrix with one column of GDEscore.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Obtain the example data
GEP&lt;-GetData_Mirna("GEP")
label&lt;-GetData_Mirna("label")
# Run the function
GDEscore&lt;-GetGDEscore(GEP,label)
</code></pre>


</div>