<div class="container">

<table style="width: 100%;"><tr>
<td>infer_ancestry</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Infer ancestry of GWAS dataset by matching against 1000 genomes allele frequencies</h2>

<h3>Description</h3>

<p>Uses ~20k SNPs selected for common frequency across 5 major super populations
</p>


<h3>Usage</h3>

<pre><code class="language-R">infer_ancestry(d, snpinfo = NULL, opengwas_jwt = get_opengwas_jwt())
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>d</code></td>
<td>
<p>Data frame containing at least <code>rsid</code> and <code>eaf</code> columns.
e.g. output from associations</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snpinfo</code></td>
<td>
<p>Output from <code>afl2_list</code>, <code>afl2_rsid</code> or <code>afl2_chrpos</code>.
If <code>NULL</code> then <code>afl2_list()</code> is used by default</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>opengwas_jwt</code></td>
<td>
<p>Used to authenticate protected endpoints. Login to <a href="https://api.opengwas.io">https://api.opengwas.io</a> to obtain a jwt. Provide the jwt string here, or store in .Renviron under the keyname OPENGWAS_JWT.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>data frame ordered by most likely ancestry, each row represents a super population and cor column represents the correlation between the GWAS dataset and the 1000 genomes super population allele frequencies
</p>


</div>