<div class="container">

<table style="width: 100%;"><tr>
<td>afl2_list</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Retrieve a allele frequency and LD scores for pre-defined lists of variants</h2>

<h3>Description</h3>

<p>Data frame includes 1000 genomes metadata including sample sizes,
allele frequency and LD score, separated by 5 super populations
(EUR = European, AFR = African, EAS = East Asian, AMR = Admixed American,
SAS = South Asian)
</p>


<h3>Usage</h3>

<pre><code class="language-R">afl2_list(
  variantlist = c("reduced", "hapmap3")[1],
  opengwas_jwt = get_opengwas_jwt()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>variantlist</code></td>
<td>
<p>Choose pre-defined list. reduced = ~20k SNPs that are
common in all super populations (default). hapmap3 = ~1.3 million hm3 SNPs</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>opengwas_jwt</code></td>
<td>
<p>Used to authenticate protected endpoints. Login to <a href="https://api.opengwas.io">https://api.opengwas.io</a> to obtain a jwt. Provide the jwt string here, or store in .Renviron under the keyname OPENGWAS_JWT.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Data frame containing ancestry specific LD scores and allele frequencies for each variant
</p>


</div>