<div class="container">

<table style="width: 100%;"><tr>
<td>afl2_chrpos</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Look up allele frequencies and LD scores for 1000 genomes populations by chrpos</h2>

<h3>Description</h3>

<p>Look up allele frequencies and LD scores for 1000 genomes populations by chrpos
</p>


<h3>Usage</h3>

<pre><code class="language-R">afl2_chrpos(chrpos, reference = "1000g", opengwas_jwt = get_opengwas_jwt())
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>chrpos</code></td>
<td>
<p>list of <code style="white-space: pre;">⁠&lt;chr&gt;:&lt;pos&gt;⁠</code> in build 37, e.g. <code>c("3:46414943", "3:122991235")</code>.
Also allows ranges e.g <code>"7:105561135-105563135"</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>reference</code></td>
<td>
<p>Default=<code>"1000g"</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>opengwas_jwt</code></td>
<td>
<p>Used to authenticate protected endpoints. Login to <a href="https://api.opengwas.io">https://api.opengwas.io</a> to obtain a jwt. Provide the jwt string here, or store in .Renviron under the keyname OPENGWAS_JWT.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Data frame containing ancestry specific LD scores and allele frequencies for each variant
</p>


</div>