<div class="container">

<table style="width: 100%;"><tr>
<td>ld_matrix</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get LD matrix for list of SNPs</h2>

<h3>Description</h3>

<p>This function takes a list of SNPs and searches for them in a specified
super-population in the 1000 Genomes phase 3 reference panel.
It then creates an LD matrix of r values (signed, and not squared).
All LD values are with respect to the major alleles in the 1000G dataset.
You can specify whether the allele names are displayed.
</p>


<h3>Usage</h3>

<pre><code class="language-R">ld_matrix(
  variants,
  with_alleles = TRUE,
  pop = "EUR",
  opengwas_jwt = get_opengwas_jwt(),
  bfile = NULL,
  plink_bin = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>variants</code></td>
<td>
<p>List of variants (rsids)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>with_alleles</code></td>
<td>
<p>Whether to append the allele names to the SNP names. Default: <code>TRUE</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pop</code></td>
<td>
<p>Super-population to use as reference panel. Default = <code>"EUR"</code>.
Options are <code>"EUR"</code>, <code>"SAS"</code>, <code>"EAS"</code>, <code>"AFR"</code>, <code>"AMR"</code>.
<code>'legacy'</code> also available - which is a previously used verison of the EUR
panel with a slightly different set of markers</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>opengwas_jwt</code></td>
<td>
<p>Used to authenticate protected endpoints. Login to <a href="https://api.opengwas.io">https://api.opengwas.io</a> to obtain a jwt. Provide the jwt string here, or store in .Renviron under the keyname OPENGWAS_JWT.#' @param bfile If this is provided then will use the API. Default = <code>NULL</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bfile</code></td>
<td>
<p>If this is provided then will use the API. Default = <code>NULL</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plink_bin</code></td>
<td>
<p>If <code>NULL</code> and bfile is not <code>NULL</code> then will detect packaged
plink binary for specific OS. Otherwise specify path to plink binary. Default = <code>NULL</code></p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The data used for generating the LD matrix includes only bi-allelic SNPs
with MAF &gt; 0.01, so it's quite possible that a variant you want to include
will be absent. If it is absent, it will be automatically excluded from the results.
</p>
<p>You can check if your variants are present in the LD reference panel using
<code>ld_reflookup()</code>
</p>
<p>This function does put load on the OpenGWAS servers, which makes life more
difficult for other users, and has been limited to analyse only up to 500
variants at a time. We have implemented a method and made available the LD
reference panels to perform the operation locally, see <code>ld_matrix()</code> and
related vignettes for details.
</p>


<h3>Value</h3>

<p>Matrix of LD r values
</p>


</div>