<div class="container">

<table style="width: 100%;"><tr>
<td>associations</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Query specific variants from specific GWAS</h2>

<h3>Description</h3>

<p>Every rsid is searched for against each requested GWAS id. To get a list of
available GWAS ids, or to find their meta data, use <code>gwasinfo</code>.
Can request LD proxies for instances when the requested rsid is not present
in a particular GWAS dataset. This currently only uses an LD reference panel
composed of Europeans in 1000 genomes version 3.
It is also restricted to biallelic single nucleotide polymorphisms (no indels),
with European MAF &gt; 0.01.
</p>


<h3>Usage</h3>

<pre><code class="language-R">associations(
  variants,
  id,
  proxies = 1,
  r2 = 0.8,
  align_alleles = 1,
  palindromes = 1,
  maf_threshold = 0.3,
  opengwas_jwt = get_opengwas_jwt()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>variants</code></td>
<td>
<p>Array of variants e.g. <code>c("rs234", "7:105561135-105563135")</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>id</code></td>
<td>
<p>Array of GWAS studies to query. See <code>gwasinfo</code> for available studies</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>proxies</code></td>
<td>
<p><code>0</code> or (default) <code>1</code> - indicating whether to look for proxies</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>r2</code></td>
<td>
<p>Minimum proxy LD rsq value. Default=<code>0.8</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>align_alleles</code></td>
<td>
<p>Try to align tag alleles to target alleles (if <code>proxies = 1</code>).
<code>1</code> = yes (default), <code>0</code> = no</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>palindromes</code></td>
<td>
<p>Allow palindromic SNPs (if <code>proxies = 1</code>). <code>1</code> = yes (default), <code>0</code> = no</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maf_threshold</code></td>
<td>
<p>MAF threshold to try to infer palindromic SNPs. Default = <code>0.3</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>opengwas_jwt</code></td>
<td>
<p>Used to authenticate protected endpoints. Login to https://api.opengwas.io to obtain a jwt. Provide the jwt string here, or store in .Renviron under the keyname OPENGWAS_JWT.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Dataframe
</p>


</div>