<div class="container">

<table style="width: 100%;"><tr>
<td>ld_matrix_local</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Get LD matrix using local plink binary and reference dataset</h2>

<h3>Description</h3>

<p>Get LD matrix using local plink binary and reference dataset
</p>


<h3>Usage</h3>

<pre><code class="language-R">ld_matrix_local(variants, bfile, plink_bin, with_alleles = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>variants</code></td>
<td>
<p>List of variants (rsids)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bfile</code></td>
<td>
<p>Path to bed/bim/fam ld reference panel</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>plink_bin</code></td>
<td>
<p>Specify path to plink binary. Default = <code>NULL</code>.
See <a href="https://github.com/explodecomputer/plinkbinr">https://github.com/explodecomputer/plinkbinr</a> for convenient access to plink binaries</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>with_alleles</code></td>
<td>
<p>Whether to append the allele names to the SNP names.
Default: <code>TRUE</code></p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>data frame
</p>


</div>