<div class="container">

<table style="width: 100%;"><tr>
<td>lrtG</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Likelihood Ratio Tests for Identifying Disease Polymorphisms with Same Effects</h2>

<h3>Description</h3>

<p>Compute p-values and likelihoods of all possible models for a given number of disease SNP(s). 
</p>


<h3>Usage</h3>

<pre><code class="language-R"> lrtG(n.fp, genoT, genoC)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>n.fp</code></td>
<td>
<p>number of disease SNPs for tests.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genoT</code></td>
<td>

<p>matrix of control genotypes. Each row indicates a variant, and each column indicates a haplotype of an individual. Two alleles of 0 and 1 are allowed.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genoC</code></td>
<td>

<p>matrix of case genotypes. Each row indicates a variant, and each column indicates a haplotype of an individual. Two alleles of 0 and 1 are allowed.
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>matrix of likelihood ratio test results. First row indicates the index, and following n.fp rows indicate the model for each set of disease polymorphisms, and followed by p-values, -2 log(likelihood ratio) with corrections for variances, and the degree of freedom.
</p>


<h3>References</h3>

<p>L. Park, J. Kim, Rare high-impact disease variants: properties and identification, Genetics Research, 2016 Mar; 98, e6.
</p>


<h3>See Also</h3>

<p>allele.freq.G
</p>


<h3>Examples</h3>

<pre><code class="language-R">## LRT tests for a dominant variant (15th variant)
## the odds ratio: 3, control: 100, case: 100.

data(apoeG)
lrtG(1,genoT[,1:20],genoC[,1:20])

# use "lrtG(1,genoT,genoC)" for the actual test.

</code></pre>


</div>