<div class="container">

<table style="width: 100%;"><tr>
<td>lrt</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Likelihood Ratio Tests for Identifying Number of Functional Polymorphisms</h2>

<h3>Description</h3>

<p>Compute p-values and likelihoods of all possible models for a given number of functional SNP(s). 
</p>


<h3>Usage</h3>

<pre><code class="language-R"> lrt(n.fp, n, x, geno, no.con=nrow(geno))
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>n.fp</code></td>
<td>
<p>number of functional SNPs for tests.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n</code></td>
<td>
<p>array of each total number of case sample chromosomes for SNPs</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>array of each total allele number in case samples</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geno</code></td>
<td>

<p>matrix of alleles, such that each locus has a pair of adjacent columns of alleles, and the order of columns corresponds to the order of loci on a chromosome.  If there are K loci, then ncol(geno) = 2*K. Rows represent the alleles for each subject. Each allele shoud be represented as numbers (A=1,C=2,G=3,T=4).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>no.con</code></td>
<td>
<p>number of control chromosomes.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>matrix of likelihood ratio test results. First n.fp rows indicate the model for each set of disease polymorphisms, and followed by p-values, -2 log(likelihood ratio) with corrections for variances, maximum likelihood ratio estimates, and likelihood.
</p>


<h3>References</h3>

<p>L. Park, Identifying disease polymorphisms from case-control genetic association data, Genetica, 2010 138 (11-12), 1147-1159.
</p>


<h3>See Also</h3>

<p>allele.freq
hap.freq
</p>


<h3>Examples</h3>

<pre><code class="language-R">## LRT tests when SNP1 &amp; SNP6 are the functional polymorphisms.

data(apoe)

n&lt;-c(2000, 2000, 2000, 2000, 2000, 2000, 2000) #case sample size = 1000
x&lt;-c(1707, 281,1341, 435, 772, 416, 1797) #allele numbers in case samples 


Z&lt;-2 	#number of functional SNPs for tests
n.poly&lt;-ncol(apoe7)/2 	#total number of SNPs

#control sample generation( sample size = 1000 )
con.samp&lt;-sample(nrow(apoe7),1000,replace=TRUE)
con.data&lt;-array()
for (i in con.samp){
con.data&lt;-rbind(con.data,apoe7[i,])
}
con.data&lt;-con.data[2:1001,]

lrt(1,n,x,con.data)
lrt(2,n,x,con.data)
</code></pre>


</div>