<div class="container">

<table style="width: 100%;"><tr>
<td>hap.freq</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Estimation of Haplotype Frequencies with Two SNPs</h2>

<h3>Description</h3>

<p>EM computation of haplotype frequencies with two SNPs. The computation is relied on the package"haplo.stats". 
</p>


<h3>Usage</h3>

<pre><code class="language-R"> hap.freq(geno)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>geno</code></td>
<td>

<p>matrix of alleles, such that each locus has a pair of adjacent columns of alleles, and the order of columns corresponds to the order of loci on a chromosome.  If there are K loci, then ncol(geno) = 2*K. Rows represent the alleles for each subject. Each allele shoud be represented as numbers (A=1,C=2,G=3,T=4).
</p>
</td>
</tr></table>
<h3>Value</h3>

<p>matrix of haplotype frequencies consisted of two alleles from each SNP. These alleles are the same ones computed for frequency using the function "allele.freq".
</p>


<h3>See Also</h3>

<p>allele.freq
</p>


<h3>Examples</h3>

<pre><code class="language-R"> data(apoe)
 hap.freq(apoe7)
 hap.freq(apoe)
</code></pre>


</div>