<div class="container">

<table style="width: 100%;"><tr>
<td>error.rates</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Error Rates Estimation for Likelihood Ratio Tests Designed for Identifying Number of Functional Polymorphisms</h2>

<h3>Description</h3>

<p>Compute error rates for a given model. 
</p>


<h3>Usage</h3>

<pre><code class="language-R"> error.rates(H0,Z, pMc, geno, no.ca, no.con=nrow(geno), sim.no = 1000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>H0</code></td>
<td>
<p>the index number for a given model for functional SNPs</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Z</code></td>
<td>
<p>number of functional SNPs for the given model</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pMc</code></td>
<td>
<p>array of allele frequencies of case samples</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geno</code></td>
<td>

<p>matrix of alleles, such that each locus has a pair of adjacent columns of alleles, and the order of columns corresponds to the order of loci on a chromosome.  If there are K loci, then ncol(geno) = 2*K. Rows represent the alleles for each subject. Each allele shoud be represented as numbers (A=1,C=2,G=3,T=4).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>no.ca</code></td>
<td>
<p>number of case chromosomes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>no.con</code></td>
<td>
<p>number of control chromosomes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sim.no</code></td>
<td>
<p>number of simulations for error rates estimation</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>array of results consisted of Type I error rate (alpha=0.05), Type I error rate (alpha=0.01), Type II error rate (beta=0.05), Type II error rate (beta=0.01), percent when the target model has the lowest corrected -2 log likelihood ratio. 
</p>


<h3>See Also</h3>

<p>allele.freq
hap.freq
lrtB
</p>


<h3>Examples</h3>

<pre><code class="language-R">## LRT tests when SNP1 &amp; SNP6 are the functional polymorphisms.


data(apoe)

n&lt;-c(2000, 2000, 2000, 2000, 2000, 2000, 2000) #case sample size = 1000
x&lt;-c(1707, 281,1341, 435, 772, 416, 1797) #allele numbers in case samples 

Z&lt;-2 	#number of functional SNPs for tests
n.poly&lt;-ncol(apoe7)/2 	#total number of SNPs

#index number for the model in this case is 5 for SNP1 and 6. 
#apoe7 is considered to represent the true control allele and haplotype frequencies.
#Control sample size = 1000.

error.rates(5, 2, x/n, apoe7, 2000, 2000, sim.no=2)

# to obtain valid rates, use sim.no=1000.

</code></pre>


</div>