<div class="container">

<table style="width: 100%;"><tr>
<td>oligoSpecies</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Species level counts for the OligoMM12-Colitis experiment used as an example in the IgAScores package</h2>

<h3>Description</h3>

<p>Data from the colitis model described in Jackson et al. (2020, doi: <a href="https://doi.org/10.1101/2020.08.19.257501">10.1101/2020.08.19.257501</a>).
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(oligoSpecies)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>"tibble"</code>.
</p>


<h3>Details</h3>

<p>Species level counts for an experiment where mice with a defined gut microbiota (OligoMM12) were either given *Helicobacter hepaticus* and IL10R antibody or the antibody alone (the first developing colitis).
Metadata for this experiment can be found in oligoMeta. Counts were generated from ASVs from the V4 region of the 16S rRNA gene processed using DADA2 and aligned to the *"RefSeq-RDP16S_v2_May2018.fa.gz"* database.
Further details can be found in Jackson et al. (2020, doi: <a href="https://doi.org/10.1101/2020.08.19.257501">10.1101/2020.08.19.257501</a>).
</p>


<h3>References</h3>

<p>To come...
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(oligoSpecies)
</code></pre>


</div>