<div class="container">

<table style="width: 100%;"><tr>
<td>SMR_interval</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Interval Estimates for Summary Data Mendelian Randomization Analysis in the Presence of Winner's Curse</h2>

<h3>Description</h3>

<p><code>SMR_interval</code> calculates conservative box method interval, k-unit support interval, and Wald confidence interval for the causal effect. 
</p>



<h3>Usage</h3>

<pre><code class="language-R">SMR_interval(
  summary.data,
  sig.level = 5e-08,
  k = 2,
  alpha = 0.05,
  method = "box"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>summary.data</code></td>
<td>
<p>a vector (<code class="reqn">\hat b_{gx}</code>, se(<code class="reqn">\hat b_{gx}</code>), <code class="reqn">\hat b_{gy}</code>, se(<code class="reqn">\hat b_{gy}</code>)) of summary data on the exposure <code class="reqn">X</code> and the outcome <code class="reqn">Y</code>. Due to winner's curse, the association <code class="reqn">p</code>-value between the SNP and the exposure is less than <code>sig.level</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sig.level</code></td>
<td>
<p>the threshold <code class="reqn">p</code>-value used to select the instrument SNP. The default is <code>5e-8</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>k</code></td>
<td>
<p>the unit used for the k-unit support. The default value is 2.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p><code class="reqn">(1-\alpha)</code> is the conservative coverage level for the box method interval or the SMR Wald interval. the default value is 0.05</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>method to construct the interval. It is either "<code>support</code>", "<code>box</code>" or "<code>wald</code>". The default is "<code>box</code>".</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>The returned value is method-dependent.
</p>
<p>For <code>method == "box"</code>: A list containing the following components:
</p>
<p>* an interval estimate.
</p>
<p>* type of the interval: completely bounded, exclusive bounded, or bounded.
</p>
<p>For <code>method == "support"</code>: A list containing the following components:
</p>
<p>* Estimate The likelihood estimate of <code class="reqn">b</code>.
</p>
<p>* an interval estimate.
</p>
<p>For <code>method == "wald"</code>: an interval estimate.
</p>


<h3>Author(s)</h3>

<p>Kai Wang <code>&lt;kai-wang@uiowa.edu&gt;</code>
</p>


<h3>References</h3>

<p>Wang, K. (2023) Support interval for two-sample summary data-based mendelia randomization. <em>Genes</em>, 14(1):211.
</p>
<p>Wang, K. (2023) Interval estimate of causal effect in summary data based Mendelian randomization in the presence of winnerâ€™s curse. <em>Genetic Epidemiology</em>, 14(1):211.
</p>
<p>Zhu, Z. et al. (2016) Integration of summary data from GWAS and eQTL studies predicts complex trait gene targets. <em>Nature Genetics</em>, 48(5):481.
</p>


<h3>Examples</h3>

<pre><code class="language-R">summary.data = c(0.13707, 0.0235162, -0.0637257, 0.013774)
SMR_interval(summary.data)
SMR_interval(summary.data, method = "support")
SMR_interval(summary.data, method = "wald")

</code></pre>


</div>