<div class="container">

<table style="width: 100%;"><tr>
<td>tarq</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>An Accurate Normalization Method for High Throughput Sequencing Data</h2>

<h3>Description</h3>

<p>Estimates scaling factors using the trimmed average of ratios of quantiles (TARQ) method
</p>


<h3>Usage</h3>

<pre><code class="language-R">tarq(X, tau = 0.3)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>
<p>a matrix of raw counts. Rows are for taxa (genes, transcripts) and columns for samples</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tau</code></td>
<td>
<p>a numerical value in (0, 0.5). The upper <code class="reqn">\tau/2 \times 100</code>% and the lower <code class="reqn">\tau/2 \times 100</code>% of the 
ratios of quantiles are trimmed</p>
</td>
</tr>
</table>
<h3>Details</h3>


<p>Estimation of scaling factors for NGS read counts data is challenging. TARQ provides a quantile-based method for estimating 
scaling factors. It starts by ordering the raw counts 
sample by sample and constructs a reference sample from these ordered counts. To compute the scaling factor for a sample, ratios
of its quantiles to those of the reference sample are formed. Zero ratios are removed. Then extreme ratios (too large or too 
small) are trimmed before taking average over the remaining ratios.
</p>


<h3>Value</h3>

<p>a vector of scaling factors. Normalized counts can be obtained by <code>sweep(X, 2, scale.factors, FUN="/")</code>
</p>


<h3>Author(s)</h3>

<p>Kai Wang <code>&lt;kai-wang@uiowa.edu&gt;</code>
</p>


<h3>References</h3>

<p>Wang, K. (2018) An Accurate Normalization Method for Next-Generation Sequencing Data.
Submitted.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
#data(throat.otu.tab)
#data(throat.meta)
#otu.tab = t(throat.otu.tab)
#tarq(otu.tab, 0.3)

##### Use TARQ with DESeq2 
#dds &lt;- DESeqDataSetFromMatrix(countData = otu.tab,
 #                             colData = throat.meta,
 #                             design= ~ SmokingStatus)
#sizeFactors(dds) &lt;- tarq(otu.tab, 0.3)
#dds &lt;- DESeq(dds)                 
#results(dds)
#
###### Use TARQ with edgeR
#cs &lt;- colSums(otu.tab)
#scale.factors &lt;- tarq(otu.tab, 0.3)
#tmp &lt;- scale.factors/cs
#norm.factors &lt;- tmp/exp(mean(log(tmp)))
#dgList &lt;- DGEList(counts = otu.tab, genes=rownames(otu.tab), norm.factors = norm.factors)
#designMat &lt;- model.matrix(~ throat.meta$SmokingStatus)
#dgList &lt;- estimateGLMCommonDisp(dgList, design=designMat)
#fit &lt;- glmFit(dgList, designMat)
#glmLRT(fit, coef=2)

</code></pre>


</div>