<div class="container">

<table style="width: 100%;"><tr>
<td>KAT.coin</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Conditional Inference for the Kernel Association Test (KAT)</h2>

<h3>Description</h3>

<p>Computes the asymptotic and the approximate conditional p-values for the kernel association test
</p>


<h3>Usage</h3>

<pre><code class="language-R">KAT.coin(y, G, X = NULL, out_type = "D", distribution = "asymptotic", B = 1000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>y</code></td>
<td>
<p>a vector of phenotype on <code class="reqn">n</code> subjects.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>G</code></td>
<td>
<p>an <code class="reqn">n \times m</code> matrix of SNP genotypes of <code class="reqn">n</code> study subjects at <code class="reqn">m</code> loci.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>
<p>a matrix of covariates. It has <code class="reqn">n</code> rows.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>out_type</code></td>
<td>
<p>an indicator of the outcome type. <code>"C"</code> for the continuous outcome and <code>"D"</code> for the dichotomous outcome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>distribution</code></td>
<td>
<p>a character, the conditional null distribution of the test statistic can be approximated by its asymptotic distribution ("asymptotic", default) or via Monte Carlo resampling ("approximate"), as in package <code>coin</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>B</code></td>
<td>
<p>the number of permutations if <code>distribution = "approximate"</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>


<p>The asymptotic conditional null distribution is obtained using results in Strasser and Weber (1999). 
The p-value based on this distribution is computed using Davies' method.
</p>


<h3>Value</h3>

<p>A list with class "<code>htest</code>" containing the following components:
</p>
<p>* statistic the value of the kernel association test statistic.
</p>
<p>* parameter sample size and the number of SNPs.
</p>
<p>* p.value the p-value based on the asymptotic or the approximate conditional null distribution.
</p>
<p>* method a character string indicting the test performed.
</p>
<p>* data.name a character string giving the name of the data.
</p>


<h3>Author(s)</h3>

<p>Kai Wang <code>&lt;kai-wang@uiowa.edu&gt;</code>
</p>


<h3>References</h3>

<p>Strasser, H. and Weber, C. (1999) On the asymptotic theory of permutation statistics. <em>Mathematical Methods of Statistics</em>. 8(2):220-250.
</p>
<p>Wang, K. (2017) Conditional Inference for the Kernel Association Test. Bioinformatics 33 (23), 3733-3739.
</p>


<h3>Examples</h3>

<pre><code class="language-R">n=1000
y = c(rep(1, n/2), rep(0, n/2))
maf = seq(0.05, 0.5, 0.05)
g = NULL
for (j in 1:10){
   geno.freq = c(maf[j]^2, 2*maf[j]*(1-maf[j]), (1-maf[j])^2)
   g = cbind(g, sample(c(0,1,2), n, replace=TRUE, prob=geno.freq))
   }
KAT.coin(y, g, X=NULL, out_type="D", B=1000)

</code></pre>


</div>