<div class="container">

<table style="width: 100%;"><tr>
<td>IGST.weight.bootmrmrsvm</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Computation of weights for informative genes or gene set selection using SVM and MRMR feature selection technique with bootstrapping procedure
</h2>

<h3>Description</h3>

<p>The function computes the weights associated with each genes for a given dataset using SVM and MRMR feature selection technique with bootstrapping procedure.
</p>


<h3>Usage</h3>

<pre><code class="language-R">IGST.weight.bootmrmrsvm (x, y, re, v)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p> x is a n by p data frame of gene expression values where rows represent genes and columns represent samples. Each cell entry represents the expression level of a gene in a sample or subject (row names of x as gene names or gene ids).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>y</code></td>
<td>
<p> y is a p by 1 numeric vector with entries 1 or -1 representing sample labels, where, 1\/-1 represents the sample label of subjects or samples for stress or control condition(for two class problems).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>v</code></td>
<td>
<p>v is a scalar representing the weightage of a method and must be within 0 and 1. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>re</code></td>
<td>
<p>re is a scalar representing the number of bootstrap generated, re must be sufficiently large (i.e. number of times bootstrap samples are generated.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>The function returns a vector of weights associated with each genes computed from SVM and MRMR feature selection technique with bootstrapping procedure for a given dataset. </p>


<h3>Author(s)</h3>

<p>Nitesh Kumar Sharma, Dwijesh Chandra Mishra, Neeraj Budhlakoti and Md. Samir Farooqi
</p>


<h3>References</h3>

<p>Wang, J., Chen, L., Wang, Y., Zhang, J., Liang, Y., &amp; Xu, D. (2013). A computational systems biology study for understanding salt tolerance mechanism in rice. PLoS One, 8(6), e64929.
</p>
<p>Ding, C., &amp; Peng, H. (2005). Minimum redundancy feature selection from microarray gene expression data. Journal of Bioinformatics and Computational Biology, 3(02), 185-205.
</p>
<p>Mishra DC, Kumar S, Lal SB, Saha A, Chaturvedi KK, Budhlakoti N, et al.( 2018) TAGPT: A Web Server for Prediction of Trait Associated Genes using Gene Expression Data. Annals of Genetics and Genetic Disorder. 1(1): 1003.
</p>


<h3>Examples</h3>

<pre><code class="language-R">#################################
library(IGST)
data(rice_cold)
x&lt;-rice_cold[-1,]
y&lt;-rice_cold[1,]
y&lt;-as.matrix(y)
y&lt;-as.vector(y)
#s&lt;-10
#Q&lt;-0.5
v&lt;-0.5
re&lt;-10
IGST.weight.bootmrmrsvm (x, y, re, v)
</code></pre>


</div>