<div class="container">

<table style="width: 100%;"><tr>
<td>testchange</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>testchange</h2>

<h3>Description</h3>

<p>This function identifies geographic areas with significant change over time.
</p>


<h3>Usage</h3>

<pre><code class="language-R">testchange(data, time, perm = FALSE, nperm = 100, numclust = 4, topF = 500)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>a numeric matrix, each row representing a time-series
and each column representing a time point</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>time</code></td>
<td>
<p>defines the time sequence</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>perm</code></td>
<td>
<p>if perm = 'TRUE', a permutation is performed</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nperm</code></td>
<td>
<p>number of permuations</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>numclust</code></td>
<td>
<p>defines the number of clusters for the parallel processing</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>topF</code></td>
<td>
<p>number of top F values to be selected when perm = 'FALSE'</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>number of permutations of &gt;=10,000 is ideal
</p>


<h3>Value</h3>

<p>Output if perm = 'TRUE' is a list of three items:
</p>

<ul>
<li>
<p> perm.F - F values obtained from permutation tests
</p>
</li>
<li>
<p> p.values - p-values obtained from permutation tests
</p>
</li>
<li>
<p> p.adjusted - p-values adjusted by Benjamini-Hochberg method
</p>
</li>
</ul>
<p>Output if perm = 'False' is a list of three items:
</p>

<ul>
<li>
<p> obs.F - conventional F-statistic values
</p>
</li>
<li>
<p> sig.change - areas with significant change over time pattern selected by top F-statistic values
</p>
</li>
<li>
<p> sel.F - top F-statistic values selected
</p>
</li>
</ul>
<h3>References</h3>

<p>1. Song, J., Carey, M., Zhu, H., Miao, H., Ram´ırez, J. C., &amp; Wu, H. (2018). Identifying the dynamic gene regulatory network during latent HIV-1 reactivation using high-dimensional ordinary differential equations. International Journal of Computational Biology and Drug Design, 11,135-153. doi: 10.1504/IJCBDD.2018.10011910.
2. Wu, S., &amp; Wu, H. (2013). More powerful significant testing for time course gene expression data using functional principal component analysis approaches. BMC Bioinformatics, 14:6.
3. Carey, M., Wu, S., Gan, G. &amp; Wu, H. (2016). Correlation-based iterative clustering methods for time course data: The identification of temporal gene response modules for influenza infection in humans. Infectious Disease Modeling, 1, 28-39.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># This is an example not using the permutation approach

opioid_data_noNA &lt;- opioidData[complete.cases(opioidData), ] #remove NAs

mydata &lt;- as.matrix(opioid_data_noNA[,4:18])

testchange_results &lt;- testchange(data=mydata,perm=FALSE,time=seq(1,15,1))
</code></pre>


</div>