<div class="container">

<table style="width: 100%;"><tr>
<td>convert_rle_to_df</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>convert_rle_to_df</h2>

<h3>Description</h3>

<p>Converts a list of run-length encodings (RLEs) into a data frame
with 16 features after mappability profiling and nucleotide filtering.
</p>


<h3>Usage</h3>

<pre><code class="language-R">convert_rle_to_df(
  covs,
  unreliable_region_df = unreliable_regions,
  unreliable_region_enabled = TRUE,
  additional_nucleotide_info = data.frame()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>covs</code></td>
<td>
<p>A list of Coverage profile(s) in RLE format. Can be one or more
samples.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>unreliable_region_df</code></td>
<td>
<p>The unreliable regions of the virus segments.
Default is <code>unreliable_regions</code>. It includes the mappability profile from
a host genome (we only have Arabidopsis thaliana right now) and virus
references, and the regions that have CG% and A% over 60% and 45%
respectively.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>unreliable_region_enabled</code></td>
<td>
<p>Default is <code>TRUE</code>. If <code>TRUE</code>, the input will be
checked against <code>unreliable_region_df</code>. If <code>FALSE</code>, this step will be
skipped.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>additional_nucleotide_info</code></td>
<td>
<p>Additional nucleotide information for virus
segments that are not included in <code>nucleotide_info</code>. The information
provided must be a data frame that follows the format of
<code>nucleotide_info</code>. Default is an empty data frame.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A data frame object that contains the mapping result for each virus
segment that the plant sample reads are aligned to and a RLE list of
coverage information.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
df &lt;- convert_rle_to_df(example_cov)

## End(Not run)

</code></pre>


</div>