<div class="container">

<table style="width: 100%;"><tr>
<td>repOverlap</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Main function for public clonotype statistics calculations</h2>

<h3>Description</h3>

<p>The <code>repOverlap</code> function is designed to analyse the overlap between
two or more repertoires. It contains a number of methods to compare immune receptor
sequences that are shared between individuals.
</p>


<h3>Usage</h3>

<pre><code class="language-R">repOverlap(
  .data,
  .method = c("public", "overlap", "jaccard", "tversky", "cosine", "morisita",
    "inc+public", "inc+morisita"),
  .col = "aa",
  .a = 0.5,
  .b = 0.5,
  .verbose = TRUE,
  .step = 1000,
  .n.steps = 10,
  .downsample = FALSE,
  .bootstrap = NA,
  .verbose.inc = NA,
  .force.matrix = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>.data</code></td>
<td>
<p>The data to be processed. Can be data.frame,
data.table, or a list of these objects.
</p>
<p>Every object must have columns in the immunarch compatible format.
immunarch_data_format
</p>
<p>Competent users may provide advanced data representations:
DBI database connections, Apache Spark DataFrame from copy_to or a list
of these objects. They are supported with the same limitations as basic objects.
</p>
<p>Note: each connection must represent a separate repertoire.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.method</code></td>
<td>
<p>A string that specifies the method of analysis or a combination of
methods. The <code>repOverlap</code> function supports following basic methods:
"public", "overlap", "jaccard", "tversky", "cosine", "morisita".
If vector of multiple methods is given for this parameter, the first method will be used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.col</code></td>
<td>
<p>A string that specifies the column(s) to be processed. Pass one of the
following strings, separated by the plus sign: "nt" for nucleotide sequences,
"aa" for amino acid sequences, "v" for V gene segments, "j" for J gene segments. E.g.,
pass "aa+v" to compute overlaps on CDR3 amino acid sequences paired with V gene segments, i.e.,
in this case a unique clonotype is a pair of CDR3 amino acid and V gene segment.
Clonal counts of equal clonotypes will be summed up.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.a, .b</code></td>
<td>
<p>Alpha and beta parameters for Tversky Index. Default values give
the Jaccard index measure.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.verbose</code></td>
<td>
<p>if TRUE then output the progress.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.step</code></td>
<td>
<p>Either an integer or a numeric vector.
</p>
<p>In the first case, the integer defines the step of incremental overlap.
</p>
<p>In the second case, the vector encodes all repertoire sampling depths.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.n.steps</code></td>
<td>
<p>Skipped if ".step" is a numeric vector.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.downsample</code></td>
<td>
<p>If TRUE then performs downsampling to N clonotypes at each step instead of choosing the
top N clonotypes in incremental overlaps. Change nothing of you are using conventional methods.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.bootstrap</code></td>
<td>
<p>Set NA to turn off any bootstrapping, set a number to perform bootstrapping with this number of tries.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.verbose.inc</code></td>
<td>
<p>Logical. If TRUE then shows output from the computation process.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.force.matrix</code></td>
<td>
<p>Logical. If TRUE then always forces the matrix output even in case of two input repertoires.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>"public" and "shared" are synonyms that exist
for the convenience of researchers.
</p>
<p>The "overlap" coefficient is a similarity measure that measures the overlap between two finite sets.
</p>
<p>The "jaccard" index is conceptually a percentage of how many objects two sets
have in common out of how many objects they have total.
</p>
<p>The "tversky" index is an asymmetric similarity measure on sets that
compares a variant to a prototype.
</p>
<p>The "cosine" index is a measure of similarity between two non-zero vectors
of an inner product space that measures the cosine of the angle between them.
</p>
<p>The "morisita" index measures how many times it is more likely to randomly
select two sampled points from the same quadrat (the dataset is covered by a
regular grid of changing size) then it would be in the case of a random
distribution generated from a Poisson process. Duplicate objects are merged
with their counts are summed up.
</p>


<h3>Value</h3>

<p>In most cases the return value is a matrix with overlap values for each pair of repertoires.
</p>
<p>If only two repertoires were provided, return value is single numeric value.
</p>
<p>If one of the incremental method is chosen, return list of overlap matrix.
</p>


<h3>See Also</h3>

<p>inc_overlap, vis
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(immdata)

# Make data smaller for testing purposes
immdata$data &lt;- top(immdata$data, 4000)

ov &lt;- repOverlap(immdata$data, .verbose = FALSE)
vis(ov)

ov &lt;- repOverlap(immdata$data, "jaccard", .verbose = FALSE)
vis(ov, "heatmap2")
</code></pre>


</div>