<div class="container">

<table style="width: 100%;"><tr>
<td>repGermline</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Creates germlines for clonal lineages</h2>

<h3>Description</h3>

<p>This function creates germlines for clonal lineages. B cell clonal lineage
represents a set of B cells that presumably have a common origin (arising from the same VDJ
rearrangement event) and a common ancestor. Each clonal lineage has its own germline sequence
that represents the ancestral sequence for each BCR in clonal lineage. In other words,
germline sequence is a sequence of B-cells immediately after VDJ recombination, before
B-cell maturation and hypermutation process. Germline sequence is useful for assessing
the degree of mutation and maturity of the repertoire.
</p>


<h3>Usage</h3>

<pre><code class="language-R">repGermline(.data, .species, .min_nuc_outside_cdr3, .threads)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>.data</code></td>
<td>
<p>The data to be processed. Can be data.frame, data.table
or a list of these objects.
</p>
<p>It must have columns in the immunarch compatible format immunarch_data_format.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.species</code></td>
<td>
<p>Species from which the data was acquired. Available options:
"HomoSapiens" (default), "MusMusculus", "BosTaurus", "CamelusDromedarius",
"CanisLupusFamiliaris", "DanioRerio", "MacacaMulatta", "MusMusculusDomesticus",
"MusMusculusCastaneus", "MusMusculusMolossinus", "MusMusculusMusculus", "MusSpretus",
"OncorhynchusMykiss", "OrnithorhynchusAnatinus", "OryctolagusCuniculus", "RattusNorvegicus",
"SusScrofa".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.min_nuc_outside_cdr3</code></td>
<td>
<p>This parameter sets how many nucleotides should have V or J chain
outside of CDR3 to be considered good for further alignment.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.threads</code></td>
<td>
<p>Number of threads to use.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Data with added columns:
* Sequence (FR1+CDR1+FR2+CDR2+FR3+CDR3+FR4 in nucleotides; the column will be replaced if exists)
* V.allele, J.allele (chosen alleles of V and J genes),
* V.aa, J.aa (V and J sequences from original clonotype, outside CDR3, converted to amino acids)
* Germline.sequence (combined germline nucleotide sequence)
</p>


<h3>Examples</h3>

<pre><code class="language-R">
data(bcrdata)

bcrdata$data %&gt;%
  top(5) %&gt;%
  repGermline()
</code></pre>


</div>