<div class="container">

<table style="width: 100%;"><tr>
<td>seqDist</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Function for computing distance for sequences</h2>

<h3>Description</h3>

<p>Computing sequential distances between clonotypes from two repertoires:
</p>


<h3>Usage</h3>

<pre><code class="language-R">seqDist(.data, .col = 'CDR3.nt', .method = 'hamming',
 .group_by = c("V.name", "J.name"), .group_by_seqLength = TRUE, .trim_genes = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>.data</code></td>
<td>
<p>The data to be processed. Can be data.frame,
data.table, or a list of these objects.
</p>
<p>Every object must have columns in the immunarch compatible format immunarch_data_format</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.col</code></td>
<td>
<p>A string that specifies the column name to be processed. The default value is 'CDR3.nt'.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.method</code></td>
<td>
<p>Character value or user-defined function.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.group_by</code></td>
<td>
<p>Character vector of column names to group sequence by. The default value is c("V.first", "J.first"). Columns "V.first" and "J.first" containing first genes without allele suffixes are calculated automatically from "V.name" and "J.name" if absent in the data. Pass NA for no grouping options.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.group_by_seqLength</code></td>
<td>
<p>If TRUE  - adds grouping by sequence length of .col argument</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.trim_genes</code></td>
<td>
<p>If TRUE   - use only general gene values (e.g. "IGHV1-18") of .group_by columns for clustering; if FALSE - can cause very small clusters in case of high resolution genotyping</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Extra arguments for user-defined function.
</p>
<p>The default value is <code>'hamming'</code> for Hamming distance which counts the number of character substitutions that turns b into a.
If a and b have different number of characters the distance is Inf.
</p>
<p>Other possible values are:
</p>
<p><code>'lv'</code> for Levenshtein distance which counts the number of deletions, insertions and substitutions necessary to turn b into a.
</p>
<p><code>'lcs'</code> for longest common substring is defined as the longest string can be obtained by pairing characters from a and b while keeping the order of characters intact.
</p>
<p>In case of user-defined function, it should take x and y parameters as input and return dist object.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Named list of list with dist objects for given repertoires for each combination of .group_by variable(s) and/or sequence length of .col.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
data(immdata)
# Reducing data to save time on examples
immdata$data &lt;- purrr::map(immdata$data, ~ .x %&gt;% head(10))
# Computing hamming distance for the first two repertoires in \code{'immdata'}
seqDist(immdata$data[1:2])

# Here we define a custom distance function
# that will count the difference in number of characters in sequences.

f &lt;- function(x, y) {
  res &lt;- matrix(nrow = length(x), ncol = length(y))
  for (i in 1:length(x)) {
    res[i, ] &lt;- abs(nchar(x[i]) - nchar(y))
  }
  dimnames(res) &lt;- list(x, y)
  return(as.dist(res))
}

seqDist(immdata$data[1:2], .method = f, .group_by_seqLength = FALSE)
</code></pre>


</div>