<div class="container">

<table style="width: 100%;"><tr>
<td>switch_type</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Return a column's name</h2>

<h3>Description</h3>

<p>Return a column's name
</p>


<h3>Usage</h3>

<pre><code class="language-R">switch_type(type)

process_col_argument(.col)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>
<p>Character. Specifies the column to choose:
"nt" chooses the CDR3 nucleotide column,
"aa" chooses the CDR3 amino acid column,
"v" chooses the V gene segment column,
"j" chooses the J gene segment column.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.col</code></td>
<td>
<p>A string that specifies the column(s) to be processed. Select one of the
following strings, separated by the plus sign: "nt" for nucleotide sequences,
"aa" for amino acid sequences, "v" for V gene segments, "j" for J gene segments.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A column's name.
</p>


<h3>Developer Examples</h3>

<p>immunarch:::switch_type("nuc")
immunarch:::switch_type("v")
</p>


</div>