<div class="container">

<table style="width: 100%;"><tr>
<td>getKmers</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Calculate the k-mer statistics of immune repertoires</h2>

<h3>Description</h3>

<p>Calculate the k-mer statistics of immune repertoires
</p>


<h3>Usage</h3>

<pre><code class="language-R">getKmers(.data, .k, .col = c("aa", "nt"), .coding = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>.data</code></td>
<td>
<p>The data to be processed. Can be data.frame,
data.table, or a list of these objects.
</p>
<p>Every object must have columns in the immunarch compatible format.
immunarch_data_format
</p>
<p>Competent users may provide advanced data representations:
DBI database connections, Apache Spark DataFrame from copy_to or a list
of these objects. They are supported with the same limitations as basic objects.
</p>
<p>Note: each connection must represent a separate repertoire.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.k</code></td>
<td>
<p>Integer. Length of k-mers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.col</code></td>
<td>
<p>Character. Which column to use, pass "aa" (by default) for CDR3 amino acid sequence,
pass "nt" for CDR3 nucleotide sequences.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.coding</code></td>
<td>
<p>Logical. If TRUE (by default) then removes all non-coding sequences from input data first.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Data frame with two columns (k-mers and their counts).
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(immdata)
kmers &lt;- getKmers(immdata$data[[1]], 5)
kmers %&gt;% vis()
</code></pre>


</div>