<div class="container">

<table style="width: 100%;"><tr>
<td>split_to_kmers</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Analysis immune repertoire kmer statistics: sequence profiles, etc.</h2>

<h3>Description</h3>

<p>Analysis immune repertoire kmer statistics: sequence profiles, etc.
</p>


<h3>Usage</h3>

<pre><code class="language-R">split_to_kmers(.data, .k)

kmer_profile(.data, .method = c("freq", "prob", "wei", "self"), .remove.stop = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>.data</code></td>
<td>
<p>Character vector or the output from <code>getKmers</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.k</code></td>
<td>
<p>Integer. Size of k-mers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.method</code></td>
<td>
<p>Character vector of length one. If "freq" then returns a position frequency matrix (PFM) -
a matrix with occurences of each amino acid in each position.
</p>
<p>If "prob" then returns a position probability matrix (PPM) - a matrix with probabilities of occurences of
each amino acid in each position. This is a traditional representation of sequence motifs.
</p>
<p>If "wei" then returns a position weight matrix (PWM) - a matrix with log likelihoods of PPM elements.
</p>
<p>If "self" then returns a matrix with self-information of elements in PWM.
</p>
<p>For more information see https://en.wikipedia.org/wiki/Position_weight_matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.remove.stop</code></td>
<td>
<p>Logical. If TRUE (by default) remove stop codons.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p><code>split_to_kmers</code> - Data frame with two columns (k-mers and their counts).
</p>
<p><code>kmer_profile</code> - a matrix with per-position amino acid statistics.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(immdata)
kmers &lt;- getKmers(immdata$data[[1]], 5)
kmer_profile(kmers) %&gt;% vis()
</code></pre>


</div>