<div class="container">

<table style="width: 100%;"><tr>
<td>vis</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>One function to visualise them all</h2>

<h3>Description</h3>

<p>Output from every function in immunarch can be visualised with a
single function - <code>vis</code>. The <code>vis</code> automatically detects
the type of the data and draws a proper visualisation. For example, output
from the <code>repOverlap</code> function will be identified as repertoire overlap values
and respective visualisation will be chosen without any additional arguments.
See "Details" for the list of available visualisations.
</p>


<h3>Usage</h3>

<pre><code class="language-R">vis(.data, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>.data</code></td>
<td>
<p>Pass the output from any immunarch analysis tool to <code>vis()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Any other arguments, see the "Details" section for specific visualisation functions.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>List of available visualisations for different kinds of data.
</p>
<p>Basic analysis:
</p>
<p>- Exploratory analysis results (from repExplore) - see vis.immunr_exp_vol;
</p>
<p>- Clonality statistics (from repClonality) - see vis.immunr_homeo.
</p>
<p>Overlaps and public clonotypes:
</p>
<p>- Overlaps (from repOverlap) using heatmaps, circos plots, polar area plots - see vis.immunr_ov_matrix;
</p>
<p>-  Overlap clustering (from repOverlapAnalysis) - see vis.immunr_hclust;
</p>
<p>- Repertoire incremental overlaps (from repOverlap) - see vis.immunr_inc_overlap;
</p>
<p>- Public repertoire abundance (from pubRep) - vis vis.immunr_public_repertoire.
</p>
<p>Gene usage:
</p>
<p>- Gene usage statistics (from geneUsage) using bar plots, box plots - see vis.immunr_gene_usage;
</p>
<p>- Gene usage distances (from geneUsageAnalysis) using heatmaps, circos plots, polar area plots - see vis.immunr_ov_matrix;
</p>
<p>- Gene usage clustering (from geneUsageAnalysis) - see vis.immunr_hclust.
</p>
<p>Diversity estimation:
</p>
<p>- Diversity estimations (from repDiversity) - see vis.immunr_chao1.
</p>
<p>BCR analysis:
</p>
<p>- Clonal tree (from repClonalFamily) - see vis.clonal_family and vis.clonal_family_tree.
</p>
<p>Advanced analysis:
</p>
<p>- Repertoire dynamics (from trackClonotypes) - see vis.immunr_dynamics;
</p>
<p>- Sequence logo plots of amino acid distributions (from kmer_profile) - see vis_seqlogo;
</p>
<p>- Kmers distributions (from getKmers) - see vis.immunr_kmer_table;
</p>
<p>- Mutation networks (from mutationNetwork) - Work In Progress on vis.immunr_mutation_network;
</p>
<p>- CDR3 amino acid properties, e.g., biophysical (from cdrProp) - Work In Progress on vis.immunr_cdr_prop.
</p>
<p>Additionaly, we provide a wrapper functions for visualisations of common data types:
</p>
<p>- Any data frames or matrices using heatmaps - see vis_heatmap and vis_heatmap2;
</p>
<p>- Any data frames or matrices using circos plots - see vis_circos.
</p>


<h3>Value</h3>

<p>A ggplot2, pheatmap or circlize object.
</p>


<h3>See Also</h3>

<p>fixVis for precise manipulation of plots.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Load the test data
data(immdata)

# Compute and visualise:
ov &lt;- repOverlap(immdata$data)
vis(ov)

gu &lt;- geneUsage(immdata$data)
vis(gu)

dv &lt;- repDiversity(immdata$data)
vis(dv)
</code></pre>


</div>