<div class="container">

<table style="width: 100%;"><tr>
<td>spectratype</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Immune repertoire spectratyping</h2>

<h3>Description</h3>

<p>Immune repertoire spectratyping
</p>


<h3>Usage</h3>

<pre><code class="language-R">spectratype(.data, .quant = c("id", "count"), .col = "nt")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>.data</code></td>
<td>
<p>The data to be processed. Can be data.frame,
data.table, or a list of these objects.
</p>
<p>Every object must have columns in the immunarch compatible format.
immunarch_data_format
</p>
<p>Competent users may provide advanced data representations:
DBI database connections, Apache Spark DataFrame from copy_to or a list
of these objects. They are supported with the same limitations as basic objects.
</p>
<p>Note: each connection must represent a separate repertoire.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.quant</code></td>
<td>
<p>Select the column with clonal counts to evaluate. Set to "id" to count every clonotype once.
Set to "count" to take into the account number of clones per clonotype.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>.col</code></td>
<td>
<p>A string that specifies the column(s) to be processed. The output is one of the
following strings, separated by the plus sign: "nt" for nucleotide sequences,
"aa" for amino acid sequences, "v" for V gene segments, "j" for J gene segments. E.g.,
pass "aa+v" for spectratyping on CDR3 amino acid sequences paired with V gene segments, i.e.,
in this case a unique clonotype is a pair of CDR3 amino acid and V gene segment.
Clonal counts of equal clonotypes will be summed up.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Data frame with distributions of clonotypes per CDR3 length.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Load the data
data(immdata)
sp &lt;- spectratype(immdata$data[[1]], .col = "aa+v")
vis(sp)
</code></pre>


</div>