<div class="container">

<table style="width: 100%;"><tr>
<td>immunr_data_format</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Specification of the data format used by immunarch dataframes</h2>

<h3>Description</h3>

<p>- "Clones" - number of barcodes (events, UMIs) or reads;
</p>
<p>- "Proportion" - proportion of barcodes (events, UMIs) or reads;
</p>
<p>- "CDR3.nt" - CDR3 nucleotide sequence;
</p>
<p>- "CDR3.aa" - CDR3 amino acid sequence;
</p>
<p>- "V.name" - names of aligned Variable gene segments;
</p>
<p>- "D.name" - names of aligned Diversity gene segments or NA;
</p>
<p>- "J.name" - names of aligned Joining gene segments;
</p>
<p>- "V.end" - last positions of aligned V gene segments (1-based);
</p>
<p>- "D.start" - positions of D'5 end of aligned D gene segments (1-based);
</p>
<p>- "D.end" - positions of D'3 end of aligned D gene segments (1-based);
</p>
<p>- "J.start" - first positions of aligned J gene segments (1-based);
</p>
<p>- "VJ.ins" - number of inserted nucleotides (N-nucleotides) at V-J junction (-1 for receptors with VDJ recombination);
</p>
<p>- "VD.ins" - number of inserted nucleotides (N-nucleotides) at V-D junction (-1 for receptors with VJ recombination);
</p>
<p>- "DJ.ins" - number of inserted nucleotides (N-nucleotides) at D-J junction (-1 for receptors with VJ recombination);
</p>
<p>- "Sequence" - full nucleotide sequence.
</p>


</div>