<div class="container">

<table style="width: 100%;"><tr>
<td>indelmiss-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Insertion deletion analysis while accounting for possible missing data.
</h2>

<h3>Description</h3>

<p>Genome-wide gene insertion and deletion rates can be modelled in a maximum 
likelihood framework with the additional flexibility of modelling potential missing 
data using the models included within. These models simultaneously estimate insertion 
and deletion (indel) rates of gene families and proportions of "missing" data for 
(multiple) taxa of interest. The likelihood framework is utilized for parameter 
estimation. A phylogenetic tree of the taxa and gene presence/absence patterns 
(with data ordered by the tips of the tree) are required. For more details, see 
Utkarsh J. Dang, Alison M. Devault, Tatum D. Mortimer, Caitlin S. Pepperell, 
Hendrik N. Poinar, G. Brian Golding (2016). Gene insertion deletion analysis 
while accounting for possible missing data. Genetics (accepted).
</p>


<h3>Details</h3>


<table>
<tr>
<td style="text-align: left;">
Package: </td>
<td style="text-align: left;"> indelmiss</td>
</tr>
<tr>
<td style="text-align: left;">
Type: </td>
<td style="text-align: left;"> Package</td>
</tr>
<tr>
<td style="text-align: left;">
Version: </td>
<td style="text-align: left;"> 1.0.7</td>
</tr>
<tr>
<td style="text-align: left;">
Date: </td>
<td style="text-align: left;"> 2016-08-21</td>
</tr>
<tr>
<td style="text-align: left;">
License: </td>
<td style="text-align: left;"> GPL (&gt;=2)</td>
</tr>
<tr>
<td style="text-align: left;">
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Utkarsh J. Dang and G. Brian Golding
</p>
<p><a href="mailto:utkarshdang@cunet.carleton.ca">utkarshdang@cunet.carleton.ca</a>
</p>


<h3>References</h3>

  
<p>Eddelbuettel, Dirk and Romain Francois (2011). Rcpp: Seamless R and C++ Integration. Journal of Statistical Software, 40(8), 1–18.
</p>
<p>Felsenstein, Joseph. Inferring phylogenies. Vol. 2. Sunderland: Sinauer Associates, 2004.
</p>
<p>Gilbert, Paul and Ravi Varadhan (2012). numDeriv: Accurate Numerical Derivatives. R package version 2012.9-1.
</p>
<p>Hao, Weilong, and G. Brian Golding. "The fate of laterally transferred genes: life in the fast lane to adaptation or death." Genome Research 16.5 (2006): 636–643.
</p>
<p>Paradis E., Claude J. &amp; Strimmer K. 2004. APE: analyses of phylogenetics and evolution in R language. Bioinformatics 20: 289-290. R package version 3.2.
</p>
<p>Schliep K.P. 2011. phangorn: phylogenetic analysis in R. Bioinformatics, 27(4) 592-593. R package version 1.99.11.
</p>
<p>Wickham, Hadley (2012). stringr: Make it easier to work with strings. R package version 0.6.2.
</p>


</div>