<div class="container">

<table style="width: 100%;"><tr>
<td>stitch</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Paired-end read stitching.</h2>

<h3>Description</h3>

<p>This function aligns forward and reverse reads generated from a
2x amplicon sequencing platform, and produces a consensus sequence with
maximum base-call quality scores attached as "quality" attributes.
</p>


<h3>Usage</h3>

<pre><code class="language-R">stitch(x, y, up = NULL, down = NULL, mindiff = 6, minoverlap = 16)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x, y</code></td>
<td>
<p>DNAbin objects with quality attributes (see <code>readFASTQ</code> to
generate these objects from fastq text files), representing the forward and
reverse reads to be stitched. These objects can be either vectors or lists.
In the latter case, the two objects must be equal length.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>up, down</code></td>
<td>
<p>forward and reverse primer sequences (either as concatenated
character strings or "DNAbin" objects).
Either both or neither should be provided (not one or the other).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mindiff</code></td>
<td>
<p>the minimum difference in quality between two different base calls
for the higher quality call to be added to the consensus alignment. In cases where
the quality differences are less than this threshold, the ambiguity code "N" is added
to the consensus sequence. Defaults to 6.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>minoverlap</code></td>
<td>
<p>integer giving the minimum number of nucleotides that
should be shared between the forward and reverse sequence alignments.
Defaults to 16.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a "DNAbin" object or a vector of concatenated character strings,
depending on the input.
</p>


<h3>Author(s)</h3>

<p>Shaun Wilkinson
</p>


<h3>See Also</h3>

<p><code>readFASTQ</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
  ## download and extract example FASTQ file to temporary directory
  td &lt;- tempdir()
  URL &lt;- "https://www.dropbox.com/s/71ixehy8e51etdd/insect_tutorial1_files.zip?dl=1"
  dest &lt;- paste0(td, "/insect_tutorial1_files.zip")
  download.file(URL, destfile = dest, mode = "wb")
  unzip(dest, exdir = td)
  x &lt;- readFASTQ(paste0(td, "/COI_sample1_read1.fastq"), bin = FALSE)
  y &lt;- readFASTQ(paste0(td, "/COI_sample1_read2.fastq"), bin = FALSE)
  z &lt;- stitch(x, y)
  z[1]
  attr(z, "quality")[1]

</code></pre>


</div>