<div class="container">

<table style="width: 100%;"><tr>
<td>shave</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Shave ends from DNA and amino acid sequences</h2>

<h3>Description</h3>

<p>This function uses the Viterbi algorithm to semi-globally align a motif to
a DNA or AA sequence, and removes all nucleotides to the left and/or right of the
motif.
</p>


<h3>Usage</h3>

<pre><code class="language-R">shave(x, motif, direction = "both", cores = 1, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>an object of class <code>DNAbin</code> or <code>AAbin</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>motif</code></td>
<td>
<p>a <code>DNAbin</code>, <code>AAbin</code> or <code>PHMM</code> object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>direction</code></td>
<td>
<p>character string indicating
the direction of the shave. Options are "forward" (shaves everything to
the right of the motif), "backward" (shaves everything to the left of
the motif) or "both" (retains the motif region only).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cores</code></td>
<td>
<p>integer giving the number of processors for multithreading.
Defaults to 1, and reverts to 1 if <code>x</code> is not a list.
This argument may alternatively be a 'cluster' object,
in which case it is the user's responsibility to close the socket
connection at the conclusion of the operation,
for example by running <code>parallel::stopCluster(cores)</code>.
The string 'autodetect' is also accepted, in which case the maximum
number of cores to use is one less than the total number of cores available.
Note that in this case there
may be a tradeoff in terms of speed depending on the number and size
of sequences to be processed, due to the extra time required to initialize
the cluster.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>further arguments to be passed to <code>Viterbi</code>
(not including 'type').</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This functions finds the optimal semiglobal alignment (a.k.a. "glocal"
alignment or global alignment with free end gaps) between a
sequence <code>"x"</code> and a shorter sequence <code>"motif"</code>, returning
the motif region of x along with the nucleotides to the left or right
if <code>direction</code> is set to <code>"reverse"</code> or <code>"forward"</code>,
respectively.
</p>


<h3>Value</h3>

<p>an object of class <code>DNAbin</code> or <code>AAbin</code>
(depending on the input object).
</p>


<h3>Author(s)</h3>

<p>Shaun Wilkinson
</p>


<h3>See Also</h3>

<p><code>virtualPCR</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">  data(whales)
  motif = char2dna("AAGTGTAGCATCACTTATTGATCCAAATT")
  shave(whales, motif = motif, direction = "both")
</code></pre>


</div>