<div class="container">

<table style="width: 100%;"><tr>
<td>manipulate</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Further bit-level manipulation of DNA and amino acid sequences.</h2>

<h3>Description</h3>

<p>These functions provide additional methods to manipulate objects of class
<code>"DNAbin"</code> and <code>"AAbin"</code> to supplement those available in the
<code>ape</code> package.
</p>


<h3>Usage</h3>

<pre><code class="language-R">## S3 method for class 'DNAbin'
duplicated(x, incomparables = FALSE, pointers = TRUE, ...)

## S3 method for class 'DNAbin'
unique(x, incomparables = FALSE, attrs = TRUE, drop = FALSE, ...)

## S3 method for class 'DNAbin'
subset(x, subset, attrs = TRUE, drop = FALSE, ...)

## S3 method for class 'AAbin'
duplicated(x, incomparables = FALSE, pointers = TRUE, ...)

## S3 method for class 'AAbin'
unique(x, incomparables = FALSE, attrs = TRUE, drop = FALSE, ...)

## S3 method for class 'AAbin'
subset(x, subset, attrs = TRUE, drop = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>a <code>"DNAbin"</code> or <code>"AAbin"</code> object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>incomparables</code></td>
<td>
<p>placeholder, not currently functional.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pointers</code></td>
<td>
<p>logical indicating whether the re-replication index key
should be returned as a <code>"pointers"</code> attribute of the output vector
(only applicable for <code>duplicated.DNAbin</code> and <code>duplicated.AAbin</code>).
Note that this can increase the computational
time for larger sequence lists.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>further arguments to be passed between methods.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>attrs</code></td>
<td>
<p>logical indicating whether the attributes of the input object
whose length match the object length (or number of rows if it is a matrix)
should be retained and subsetted along with the object.
This is useful if the input object has species, lineage and/or taxon ID
metadata that need to be retained following the duplicate analysis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>drop</code></td>
<td>
<p>logical; indicates whether the input matrix (assuming one is
passed) should be reduced to a vector if subset to a single sequence.
Defaults to FALSE in keeping with the style of the <code>ape</code>
package functions.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>subset</code></td>
<td>
<p>logical vector giving the elements or rows to be kept.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p><code>unique</code> and <code>subset</code> return a
<code>DNAbin</code> or <code>AAbin</code> object. <code>duplicated</code> returns a logical vector.
</p>


<h3>Author(s)</h3>

<p>Shaun Wilkinson
</p>


<h3>References</h3>

<p>Paradis E, Claude J, Strimmer K, (2004) APE: analyses of phylogenetics
and evolution in R language. <em>Bioinformatics</em> <strong>20</strong>, 289-290.
</p>
<p>Paradis E (2007) A bit-level coding scheme for nucleotides. <a href="http://ape-package.ird.fr/misc/BitLevelCodingScheme.html">http://ape-package.ird.fr/misc/BitLevelCodingScheme.html</a>.
</p>
<p>Paradis E (2012) Analysis of Phylogenetics and Evolution with R
(Second Edition). Springer, New York.
</p>


<h3>See Also</h3>

<p><code>DNAbin</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">  data(whales)
  duplicates &lt;- duplicated.DNAbin(whales, point = TRUE)
  attr(duplicates, "pointers")
  ## returned indices show that the last sequence is
  ## identical to the second one.
  ## subset the reference sequence database to only include unques
  whales &lt;- subset.DNAbin(whales, subset = !duplicates)
  ## this gives the same result as
  unique.DNAbin(whales)
</code></pre>


</div>