<div class="container">

<table style="width: 100%;"><tr>
<td>trim</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Trim primer and/or index sequences.</h2>

<h3>Description</h3>

<p>This function trims the primer and/or index sequence(s) from a set of DNA sequences.
</p>


<h3>Usage</h3>

<pre><code class="language-R">trim(x, up, down = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>a "DNAbin" object or a vector of concatenated upper-case character strings,
containing the sequences to be trimmed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>up, down</code></td>
<td>
<p>"DNAbin" objects or vectors of concatenated upper-case character strings,
representing the primer sequences.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Any sequences not containing the primer(s) in either direction are discarded.
Hence this function can also be used to de-multiplex sequences and remove indices,
though it will generally be faster to do this on the sequencing platform prior to
exporting the FASTQ files.
</p>


<h3>Value</h3>

<p>a "DNAbin" object or a vector of concatenated character strings,
depending on the input.
</p>


<h3>Author(s)</h3>

<p>Shaun Wilkinson
</p>


<h3>Examples</h3>

<pre><code class="language-R">
  ## download and extract example FASTQ file to temporary directory
  td &lt;- tempdir()
  URL &lt;- "https://www.dropbox.com/s/71ixehy8e51etdd/insect_tutorial1_files.zip?dl=1"
  dest &lt;- paste0(td, "/insect_tutorial1_files.zip")
  download.file(URL, destfile = dest, mode = "wb")
  unzip(dest, exdir = td)
  x &lt;- readFASTQ(paste0(td, "/COI_sample2.fastq"))
  ## trim primers from sequences
  mlCOIintF &lt;- "GGWACWGGWTGAACWGTWTAYCCYCC"
  jgHCO2198 &lt;- "TAIACYTCIGGRTGICCRAARAAYCA"
  x &lt;- trim(x, up = mlCOIintF, down = jgHCO2198)

</code></pre>


</div>