<div class="container">

<table style="width: 100%;"><tr>
<td>dichotomizeMethy</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Dichotomize the methylation data given both tumor and normal controls.
</h2>

<h3>Description</h3>

<p>This function implements the procedure for dichotomizing methylation data described
in the paper.
</p>


<h3>Usage</h3>

<pre><code class="language-R">dichotomizeMethy(methy, methyCtr, refUseMean = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>methy</code></td>
<td>

<p>The methylation matrix for tumor samples. Each element represents the beta value which is 
bounded between 0 and 1. Rows are genes and columns are samples.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>methyCtr</code></td>
<td>

<p>Methylation matrix of normal controls. Genes should exactly the same as the tumor sample. The sample size are not
necessarily the same as tumor sample. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>refUseMean</code></td>
<td>

<p>Logical indicating whether to use mean of normal sample as reference. Default is set to FALSE which means to use median
as it is more robust.
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A binary matrix of the same dimension of input methy. 
</p>


<h3>Author(s)</h3>

<p>Pan Tong (nickytong@gmail.com), Kevin R Coombes (krc@silicovore.com)
</p>


<h3>References</h3>

<p>Tong P, Coombes KR. <em>integIRTy: a method to identify altered
genes in cancer accounting for multiple mechanisms of regulation
using item response theory</em>. Bioinformatics, 2012 Nov 15;
28(22):2861â€“9.  
</p>


<h3>See Also</h3>

<p><code>dichotomizeCN, dichotomizeExpr, dichotomize</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(OV)
binDat &lt;- dichotomizeMethy(Methy_T[1:200, ], Methy_N[1:200, ])
binDat[15:20, 1:2]
</code></pre>


</div>