<div class="container">

<table style="width: 100%;"><tr>
<td>cv.interep</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>k-folds cross-validation for interep</h2>

<h3>Description</h3>

<p>This function does k-fold cross-validation for interep and returns the optimal value of lambda.
</p>


<h3>Usage</h3>

<pre><code class="language-R">cv.interep(e, g, y, beta0, lambda1, lambda2, nfolds, corre, pmethod, maxits)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>e</code></td>
<td>
<p>matrix of environment factors.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>g</code></td>
<td>
<p>matrix of omics factors. In the case study, the omics measurements are lipidomics data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>y</code></td>
<td>
<p>the longitudinal response.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>beta0</code></td>
<td>
<p>the intial value for the coefficient vector.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lambda1</code></td>
<td>
<p>a user-supplied sequence of <code class="reqn">\lambda_{1}</code> values, which serves as a tuning parameter for individual predictors.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lambda2</code></td>
<td>
<p>a user-supplied sequence of <code class="reqn">\lambda_{2}</code> values, which serves as a tuning parameter for interactions.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nfolds</code></td>
<td>
<p>the number of folds for cross-validation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>corre</code></td>
<td>
<p>the working correlation structure that is used in the estimation algorithm. interep provides three choices for the
working correlation structure: "a" as AR-1", "i" as "independence" and "e" as "exchangeable".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pmethod</code></td>
<td>
<p>the penalization method. "mixed" refers to MCP penalty to individual main effects and group MCP penalty to interactions; "individual" means MCP penalty to all effects.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxits</code></td>
<td>
<p>the maximum number of iterations that is used in the estimation algorithm.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>When dealing with predictors with both main effects and interactions, this function returns two optimal tuning parameters,
<code class="reqn">\lambda_{1}</code> and <code class="reqn">\lambda_{2}</code>; when there are only main effects in the predictors, this function returns <code class="reqn">\lambda_{1}</code>,
which is the optimal tuning parameter for individual predictors containing main effects.
</p>


<h3>Value</h3>

<p>an object of class "cv.interep" is returned, which is a list with components:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>lam1</code></td>
<td>
<p>the optimal <code class="reqn">\lambda_{1}</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lam2</code></td>
<td>
<p>the optimal <code class="reqn">\lambda_{2}</code>.</p>
</td>
</tr>
</table>
<h3>References</h3>

<p>Zhou, F., Ren, J., Li, G., Jiang, Y., Li, X., Wang, W.and Wu, C. (2019). Penalized variable selection for Lipid–environment interactions in a longitudinal lipidomics study.
<a href="https://www.mdpi.com/2073-4425/10/12/1002"><em>Genes</em>, 10(12), 1002</a>
</p>
<p>Zhou, F., Ren, J., Liu, Y., Li, X., Wang, W.and Wu, C. (2022). Interep: An r package for high-dimensional interaction analysis of the repeated measurement data.
<em>Genes</em>, 13(3): 554 <a href="https://doi.org/10.3390/genes13030544">doi:10.3390/genes13030544</a>
</p>
<p>Zhou, F., Ren, J., Lu, X., Ma, S. and Wu, C. (2021) Gene–Environment Interaction: a Variable Selection Perspective.
<a href="https://link.springer.com/protocol/10.1007/978-1-0716-0947-7_13"><em>Epistasis: Methods and Protocols</em>, 191-223</a> 
</p>
<p>Ren, J., Zhou, F., Li, X., Chen, Q., Zhang, H., Ma, S., Jiang,Y. and Wu, C. (2020). Semi-parametric Bayesian variable selection for Gene-Environment interactions.
<em>Statistics in Medicine</em>, 39(5): 617-638 <a href="https://doi.org/10.1002/sim.8434">doi:10.1002/sim.8434</a>
</p>
<p>Wu, C., Zhou, F., Ren, J., Li, X., Jiang, Y., Ma, S. (2019). A Selective Review of Multi-Level Omics Data Integration Using Variable Selection.
<em>High-Throughput</em>, 8(1) <a href="https://doi.org/10.3390/ht8010004">doi:10.3390/ht8010004</a>
</p>
<p>Ren, J., Du, Y., Li, S., Ma, S., Jiang, Y. and Wu, C. (2019). Robust network-based regularization and variable selection for high-dimensional genomic data in cancer prognosis.
<em>Genetic epidemiology</em>, 43(3), 276-291 <a href="https://doi.org/10.1002/gepi.22194">doi:10.1002/gepi.22194</a>
</p>
<p>Ren, J., Jung, L., Du, Y., Wu, C., Jiang, Y. and Liu, J. (2019). regnet: Network-Based Regularization for Generalized Linear Models.
<a href="https://cran.r-project.org/package=regnet"><em>R package</em>, version 0.4.0</a>
</p>
<p>Wu, C., Zhang, Q., Jiang, Y. and Ma, S. (2018). Robust network-based analysis of the associations between (epi) genetic measurements.
<em>Journal of multivariate analysis</em>, 168, 119-130 <a href="https://doi.org/10.1016/j.jmva.2018.06.009">doi:10.1016/j.jmva.2018.06.009</a>
</p>
<p>Wu, C., Zhong, P.-S., and Cui, Y. (2018). Additive varying-coefficient model for nonlinear gene-environment interactions.
<em> Statistical Applications in Genetics and Molecular Biology</em>, 17(2) <a href="https://doi.org/10.1515/sagmb-2017-0008">doi:10.1515/sagmb-2017-0008</a>
</p>
<p>Wu, C., Jiang, Y., Ren, J., Cui, Y., Ma, S. (2018). Dissecting gene-environment interactions: A penalized robust approach accounting for hierarchical structures.
<em>Statistics in Medicine</em>, 37: 437–456 <a href="https://doi.org/10.1002/sim.7518">doi:10.1002/sim.7518</a>
</p>
<p>Ren, J., He, T., Li, Y., Liu, S., Du, Y., Jiang, Y. and Wu, C. (2017). Network-based regularization for high dimensional SNP data in the case–control study of Type 2 diabetes.
<em>BMC genetics</em>, 18(1), 44 <a href="https://doi.org/10.1186/s12863-017-0495-5">doi:10.1186/s12863-017-0495-5</a>
</p>
<p>Jiang, Y., Huang, Y., Du, Y., Zhao, Y., Ren, J., Ma, S., &amp; Wu, C. (2017). Identification of prognostic genes and pathways in lung adenocarcinoma using a Bayesian approach.
<em>Cancer Inform</em>, 1(7) <a href="https://doi.org/10.1177/1176935116684825">doi:10.1177/1176935116684825</a>
</p>
<p>Wu, C., and Ma, S. (2015). A selective review of robust variable selection with applications in bioinformatics.
<em>Briefings in Bioinformatics</em>, 16(5), 873–883 <a href="https://doi.org/10.1093/bib/bbu046">doi:10.1093/bib/bbu046</a>
</p>
<p>Wu, C., Shi, X., Cui, Y. and Ma, S. (2015). A penalized robust semiparametric approach for gene-environment interactions.
<em>Statistics in Medicine</em>, 34 (30): 4016–4030 <a href="https://doi.org/10.1002/sim.6609">doi:10.1002/sim.6609</a>
</p>
<p>Wu, C., Cui, Y., and Ma, S. (2014). Integrative analysis of gene–environment interactions under a multi–response partially linear varying coefficient model.
<em>Statistics in Medicine</em>, 33(28), 4988–4998 <a href="https://doi.org/10.1002/sim.6287">doi:10.1002/sim.6287</a>
</p>
<p>Wu, C. and Cui, Y. (2013). A novel method for identifying nonlinear gene–environment interactions in case–control association studies.
<em>Human Genetics</em>, 132(12):1413–1425 <a href="https://doi.org/10.1007/s00439-013-1350-z">doi:10.1007/s00439-013-1350-z</a>
</p>
<p>Wu, C. and Cui, Y. (2013). Boosting signals in gene–based association studies via efficient SNP selection.
<em>Briefings in Bioinformatics</em>, 15(2):279–291 <a href="https://doi.org/10.1093/bib/bbs087">doi:10.1093/bib/bbs087</a>
</p>
<p>Wu, C., Zhong, P.S. and Cui, Y. (2013). High dimensional variable selection for gene-environment interactions.
<a href="https://www.stt.msu.edu/Links/Research_Memoranda/RM/RM_701.pdf"><em>Technical Report</em>, Michigan State University.</a>
</p>
<p>Wu, C., Li, S., and Cui, Y. (2012). Genetic Association Studies: An Information Content Perspective.
<em>Current Genomics</em>, 13(7),  566–573 <a href="https://doi.org/10.2174/138920212803251382">doi:10.2174/138920212803251382</a>
</p>


</div>