<div class="container">

<table style="width: 100%;"><tr>
<td>traitRate</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Trait-Dependent Shifts in Molecular Rate</h2>

<h3>Description</h3>

<p>Detection of trait-dependent shifts in the rate of molecular evolution with <b>traitRate</b> (Mayrose &amp; Otto, 2011).
</p>


<h3>Usage</h3>

<pre><code class="language-R">traitRate(phy, seq, x, mainType = "Optimize_Model", 
   n, charModelParam1 = 0.5, charModelParam2 = 1, 
   gammaParam = 0.5, seqModelParam1 = 2,
   exec = "/Applications/traitRate-1.1/programs/traitRate")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>phy</code></td>
<td>
<p>a ultrametric phylogenetic tree of class <code>phylo.</code>
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seq</code></td>
<td>
<p>a multiple sequence alignment of class <code>DNAbin</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>data frame containing a binary character in the first column.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mainType</code></td>
<td>
<p>character string giving the type of analysis; two choices are possible: <code>"Optimize_Model"</code> will produce MLE of parameters and <code>"runTraitBootstrap"</code> will perform a parametric bootstrap analysis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n</code></td>
<td>
<p>numeric, the number of bootstrap replicates. Will be ignored if <code>mainType = "Optimize_Model"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>charModelParam1</code></td>
<td>
<p>numeric, giving an initial value for the rate of transitions of character state 0 to 1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>charModelParam2</code></td>
<td>
<p>numeric, giving an initial value for the rate of transitions of character state 1 to 0.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gammaParam</code></td>
<td>
<p>numeric, giving an initial value for the <b>alpha</b> parameter of the model of sequence evolution.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seqModelParam1</code></td>
<td>
<p>numeric, giving an initial value for the <b>kappa</b> parameter of the model of sequence evolution.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exec</code></td>
<td>
<p>character string giving the path to the program directory.
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Currently none, but look for the output files in the <code>'RESULTS'</code> subdirectory in the current working directory.
</p>


<h3>Note</h3>

<p>This function is under development!
</p>


<h3>Author(s)</h3>

<p>Christoph Heibl
</p>


<h3>References</h3>

<p>Mayrose, I. &amp; S.P. Otto. 2011. A likelihood method for detecting trait-dependent shifts in the rate of molecular evolution. <em>Mol. Biol. Evol.</em> <b>28</b>: 759-770
</p>


<h3>See Also</h3>

<p><code>read.tree</code> for reading phylogenetic trees, <code>read.fas</code> for reading multiple sequence alignments in FASTA format.
</p>


</div>