<div class="container">

<table style="width: 100%;"><tr>
<td>pis</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Number of Potentially-Informative Sites</h2>

<h3>Description</h3>

<p>Returns the number or positions of potentially-informative
(parsimony-informative, phylogenetically-informative) sites in DNA sequence
alignment.
</p>


<h3>Usage</h3>

<pre><code class="language-R">pis(x, what = "fraction", use.ambiguities = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>An object of class <code>DNAbin</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>what</code></td>
<td>
<p>Either of <code>"absolute"</code>, <code>"fraction"</code>, or
<code>"index"</code>, which will return the absolute number, the relative number
or the indices of the potentially-informative sites.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>use.ambiguities</code></td>
<td>
<p><em>Not yet available</em>.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Numeric (depending on <code>what</code>, the number, fraction, or indices
of potentially-informative nucleotide sites).
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(ips.16S)	

# number of potentially-informative sites:
pis(ips.16S, what = "abs")

# proportion of potentially-informative sites:
pis(ips.16S, what = "frac")

# indices of potentially-informative sites:
pis(ips.16S, what = "ind")
</code></pre>


</div>