<div class="container">

<table style="width: 100%;"><tr>
<td>read</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Reading Sequence Files</h2>

<h3>Description</h3>

<p>Read DNA and amino acid sequences from FASTA, PHILIP, and NEXUS 
formatted files.
</p>


<h3>Usage</h3>

<pre><code class="language-R">read.fas(x, text)

read.nex(x)

read.phy(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>A character string, giving the file name.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>text</code></td>
<td>
<p>A character string in FASTA format.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>An matrix (aligned sequences)  or list (unaligned sequences) of class
<code>DNAbin</code> or <code>AAbin</code>.
</p>


<h3>References</h3>

<p>Maddison, D.R., D.L. Swofford, and W.P. Maddison. 1997. NEXUS: an
extensible file format for systematic information. <em>Syst. Biol.</em> 
<b>46</b>: 590-621.
</p>


<h3>See Also</h3>

<p><code>mafft</code> and <code>prank</code> for sequence alignment,
<code>gblocks</code> and <code>aliscore</code> for quality check and
cleaning of sequence alignments, <code>cbind.DNAbin</code> for
concatenation of sequence alignments.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## bark beetle COX1 sequences
data(ips.cox1)
## create temporary file names
format &lt;- c(".fas", ".phy", ".nex")
fn &lt;- sapply(format, tempfile, 
             pattern = "ips", tmpdir = tempdir())
## write sequences files
write.fas(ips.cox1, fn[".fas"])
write.phy(ips.cox1, fn[".phy"])
write.nex(ips.cox1, fn[".nex"])
## read sequence files
fas &lt;- read.fas(fn[".fas"])
phy &lt;- read.phy(fn[".phy"])
nex &lt;- read.nex(fn[".nex"])
## remove sequence files
unlink(fn)
</code></pre>


</div>