<div class="container">

<table style="width: 100%;"><tr>
<td>code.simple.gaps</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Simple Gap/Indel Coding</h2>

<h3>Description</h3>

<p><code>code.simple.gaps</code> takes an aligned DNA sequence matrix and codes the simple gaps, i.e. gaps that do not overlap with other gaps. The gapped positions are excluded from the matrix and the coded gap characters are appended to the matrix.
</p>


<h3>Usage</h3>

<pre><code class="language-R">code.simple.gaps(x, append = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>An object of class <code>DNAbin</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>append</code></td>
<td>
<p>Logical.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>An object of class <code>DNAbin.</code>
</p>


<h3>Author(s)</h3>

<p>Christoph Heibl</p>


<h3>References</h3>

<p>Simmons, M.P. &amp; H. Ochoterena. 2000. Gaps as characters in sequence-based phylogenetic analyses. <em>Systematic Biology</em> <b>49(2)</b>: 369â€“381.	
</p>


<h3>See Also</h3>

<p><code>deleteGaps</code>, <code>deleteEmptyCells</code>, <code>trimEnds</code>
</p>


</div>