<div class="container">

<table style="width: 100%;"><tr>
<td>mrbayes.lset</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Model Settings for MrBayes</h2>

<h3>Description</h3>

<p>Set model parameters for <code>mrbayes</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">mrbayes.lset(..., partition)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>

<p>arguments in <code>tag = value</code> form, or a list of tagged values. The tags must come from the names of model parameters described in the ‘Model Parameters’ section.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>partition</code></td>
<td>

<p>a character string giving the labelling for a partion.
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a list containing a subset (including the empty and the full set) of model parameters.
</p>


<h3>Model Parameters</h3>


<dl>
<dt>nucmodel</dt>
<dd>
<p><code>"4by4"</code>, <code>"doublet"</code>, <code>"codon"</code>, or <code>"protein"</code>.
</p>
</dd>
<dt>nst</dt>
<dd>
<p><code>1</code>, <code>2</code>, <code>6</code>, or <code>"mixed"</code>.
</p>
</dd>
<dt>code</dt>
<dd>
<p><code>"universal"</code>, <code>"vertmt"</code>, <code>"mycoplasma"</code>, <code>"yeast"</code>, <code>"ciliates"</code>, or <code>"metmt"</code>. 
</p>
</dd>
<dt>ploidy</dt>
<dd>
<p><code>"haploid"</code>, <code>"diploid"</code>,  or <code>"zlinked"</code>.
</p>
</dd>
<dt>rates</dt>
<dd>
<p><code>"equal"</code>, <code>"gamma"</code>, <code>"propinv"</code>,<code>"invgamma"</code>, or <code>"adgamma"</code>.
</p>
</dd>
<dt>ngammacat</dt>
<dd>
<p><code>1-24</code> 
</p>
</dd>
<dt>nbetacat</dt>
<dd>
<p><code>1-24</code>
</p>
</dd>
<dt>omegavar</dt>
<dd>
<p><code>"equal"</code>,  <code>"ny98"</code>,  or <code>"m3"</code>.
</p>
</dd>
<dt>covarion</dt>
<dd>
<p><code>"no"</code> or <code>"yes"</code>. 
</p>
</dd>
<dt>coding</dt>
<dd>
<p><code>"all"</code>, <code>"variable"</code>, <code>"noabsencesites"</code>, or <code>"nopresencesites"</code>. 
</p>
</dd>
<dt>parsmodel</dt>
<dd>
<p><code>"no"</code> or <code>"yes"</code>.
</p>
</dd>
</dl>
<h3>Author(s)</h3>

<p>Christoph Heibl</p>


<h3>References</h3>

<p>J.P. Huelsenbeck &amp; Ronquist F. 2001. MrBayes: Bayesian inference of phylogenetic trees. <em>Bioinformatics</em> <b>17</b>: 754-755.
</p>
<p>Ronquist F. &amp; J.P. Huelsenbeck. 2003. MrBayes 3: Bayesian phylogenetic inference under mixed models. <em>Biometrics</em> <b>19</b>: 1572-1574.
</p>
<p>MrBayes website: <a href="https://mrbayes.sourceforge.net/">https://mrbayes.sourceforge.net/</a>.
</p>


<h3>See Also</h3>

<p><code>mrbayes.prset</code> to set prior distributions, <code>mrbayes.mcmc</code> to set parameters of the Markov chain Monte Carlo (MCMC), and <code>mrbayes</code> to run MrBayes locally or prepare input files for a computer cluster.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## F81
mrbayes.lset(nst = 2)

## GTR + Gamma
mrbayes.lset(nst = 6, rates = "gamma")

## GTR + Gamma + I
mrbayes.lset(nst = 6, rates = "invgamma")
</code></pre>


</div>