<div class="container">

<table style="width: 100%;"><tr>
<td>mrbayes</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Bayesian MCMC Tree Search with MrBayes</h2>

<h3>Description</h3>

<p>Provides a wrapper for Bayesian phylogenetic tree search through
MrBayes (Huelsenbeck &amp; Ronquist, 2001; Ronquist &amp; Huelsenbeck, 2003).
</p>


<h3>Usage</h3>

<pre><code class="language-R">mrbayes(
  x,
  file = "",
  lset,
  prset,
  mcmc,
  unlink,
  constraint,
  burnin = 10,
  contype = "allcompat",
  exec,
  run = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>An object of class <code>DNAbin</code> in the case of
<code>mrbayes</code> or a matrix of mode character in the case of
<code>mrbayes.mixed</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p>A character string, giving the name of the MrBayes input file.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lset</code></td>
<td>
<p>A list as returned by <code>mrbayes.prset</code> containing the
parameter setting for the prior distributions.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prset</code></td>
<td>
<p>A list as returned by <code>mrbayes.prset</code> containing the
parameter setting for the prior distributions.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mcmc</code></td>
<td>
<p>A list as returned by <code>mrbayes.mcmc</code> containing the
parameter setting for the Markov chain Monte Carlo (MCMC).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>unlink</code></td>
<td>
<p>xxx</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>constraint</code></td>
<td>
<p>xxx</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>burnin</code></td>
<td>
<p>An integer; the number of samples from the MCMC to be discarded
prior to further analysis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>contype</code></td>
<td>
<p>A character string; the type of consensus tree calculated from
the posterior distribution of trees: either <code>"halfcompat"</code>
(majority-rule consensus tree) or <code>"allcombat"</code> (strict consensus
tree).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exec</code></td>
<td>
<p>A character string giving the full path of the MrBayes program.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>run</code></td>
<td>
<p>Logical; <code>run = FALSE</code> will only print the NEXUS file,
<code>run = TRUE</code> will also start the MCMC runs, if <code>exec</code> is
correctly specified.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><code>mrbayes</code> was last updated and tested with MrBayes
<b>v3.2.2</b> under R 3.1.0 on a x86_64-apple-darwin10.8.0 (64-bit)
platform. It is intended to offer a simply parameterized building block for
larger scripts.
</p>


<h3>Value</h3>

<p>None; a NEXUS file with MrBayes block is written to a file and, if
<code>run = TRUE</code>, the MCMC runs in MrBayes are started.
</p>


<h3>References</h3>

<p>J. P. Huelsenbeck &amp; Ronquist F. 2001. MrBayes: Bayesian inference of phylogenetic trees. <em>Bioinformatics</em> <b>17</b>: 754-755.
Ronquist F. &amp; J. P. Huelsenbeck. 2003. MrBayes 3: Bayesian phylogenetic inference under mixed models. <em>Biometrics</em> <b>19</b>: 1572-1574.
MrBayes website: <a href="https://mrbayes.sourceforge.net/">https://mrbayes.sourceforge.net/</a>.
</p>


<h3>See Also</h3>

<p><code>mafft</code> and <code>prank</code> for sequence alignment;
<code>raxml</code> for maximum likelihood tree search.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(ips.cox1)
x &lt;- ips.cox1[, 100:140] # tiny alignment
mrbayes(x, file = "", mcmc = mrbayes.mcmc(ngen = 100), run = FALSE)
## Not run: 
library(phangorn)
tree &lt;- rtree(10)
Y1 &lt;- simSeq(tree, l = 20)
Y2 &lt;- simSeq(tree, l = 20, type = "USER", levels=c("0", "1"))
Y &lt;- cbind(as.character(Y1), as.character(Y2))

## End(Not run)
</code></pre>


</div>