<div class="container">

<table style="width: 100%;"><tr>
<td>deleteGaps</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Remove Gap Positions From DNA Sequences</h2>

<h3>Description</h3>

<p>Remove indel positions (or gaps) from a DNA sequence 
alignment. For faster execution, <code>deleteGaps</code> handles sequences 
in <span class="pkg">ape</span>'s bit-level coding scheme.
</p>


<h3>Usage</h3>

<pre><code class="language-R">deleteGaps(x, gap.max = nrow(x) - 4)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>An object of class <code>DNAbin</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gap.max</code></td>
<td>
<p>An integer, which gives the maximum number of
gap characters ("-") that will be tolerated at any given alignment 
position (column). Only values between <code>0</code> and <code>nrow(x) - 4</code>
make sense phylogenetically.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The default, <code>nmax = nrow(x) - 4</code>, removes all those 
positions from the alignment, which contain at least four non-gap 
characters, which is the minimum number of sequences needed to 
produce a non-trivial unrooted topology. All gaps will be excluded 
by selecting <code>nmax = 0</code> and half of all gaps with <code>nmax = 
nrow(x) / 2</code>.  
</p>
<p>In contrast, <code>del.gaps</code> removes all gap characters 
from the alignment, so most probably the result will not be a set of 
sequences of equal length and the matrix will be coerced to a list.
</p>


<h3>Value</h3>

<p>An object of class <code>DNAbin.</code>
</p>


<h3>See Also</h3>

<p><code>code.simple.gaps</code> for coding of simple gaps, 
<code>del.gaps</code> for removal of all gap symbols from an 
alignment, <code>gblocks</code> and <code>aliscore</code> for more 
sophisticated  methods of cleaning/masking alignments.
</p>


</div>