<div class="container">

<table style="width: 100%;"><tr>
<td>ips-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Interfaces to Phylogenetic Software</h2>

<h3>Description</h3>

<p>This package presents a set of functions that were formerly included in the <em>phyloch</em> package and which wrap popular phylogenetic software for sequence alignment, masking of sequence alignments, and estimation of phylogenies and ancestral character states.
</p>


<h3>Details</h3>


<table>
<tr>
<td style="text-align: left;">
 Package: </td>
<td style="text-align: left;"> ips</td>
</tr>
<tr>
<td style="text-align: left;">
 Type: </td>
<td style="text-align: left;"> Package</td>
</tr>
<tr>
<td style="text-align: left;">
Version: </td>
<td style="text-align: left;"> 0.0.12</td>
</tr>
<tr>
<td style="text-align: left;">
Date: </td>
<td style="text-align: left;"> 2019-11-14</td>
</tr>
<tr>
<td style="text-align: left;">
 License: </td>
<td style="text-align: left;"> GPL (&gt;= 2)</td>
</tr>
<tr>
<td style="text-align: left;">
</td>
</tr>
</table>
<p>There are several functions for reading and writing DNA sequences in FASTA, PHYLIP, and NEXUS format: <code>read.fas</code>, <code>read.phy</code>, <code>read.nex</code>, <code>write.fas</code>, <code>write.phy</code>, and <code>write.nex</code>.
Some functions are available for integrating BEAST with R. XML input files for BEAST can be generated with <code>rbeauti</code>. Two functions are designed to read TreeAnnotator output: <code>read.beast</code> will render an object of class <code>phylo</code> with additional node statistics appended as list elements. These additional node statistics will be lost be the subsequent use of <code>ladderize</code> or <code>rotate</code> (or similar functions that change the ordering of internal nodes).<code>read.beast.table</code> also parses the TreeAnnotator output, but returns a matrix of node statistics. 
This package itself does not implement techniques for phylogenetic analyses, but provides a series of wrappers for commonly used software packages. Sequence alignment can be done with the <code>mafft</code> and <code>prank</code>; cleaning of sequences with <code>gblocks</code> and <code>aliscore</code>. The function <code>raxml</code> and <code>mrbayes</code> are intended for phylogenetic tree search. Running <code>mrbayes</code> with argument <code>run = FALSE</code> can be used to create MrBayes-executable NEXUS files. Finally,  wrappers is provided for <code>Multistate</code> in the <code>BayesTraits</code> package (see <code>multistateML</code> and <code>multistateMCMC</code>).
Several plotting functions (<code>HPDbars</code>, <code>clade.bars</code>, <code>box.clades</code>, <code>box.tips</code>, <code>tip.color</code>, <code>edge.color</code> have been moved to the <b>viper</b> package.
</p>


<h3>Author(s)</h3>

<p>Natalie Cusimano, Christoph Heibl, Franz-Sebastian Krah,
Maintainer: Christoph Heibl (<a href="mailto:christoph.heibl@gmx.net">christoph.heibl@gmx.net</a>)
</p>


<h3>See Also</h3>

<p><code>ape</code>
</p>


</div>