<div class="container">

<table style="width: 100%;"><tr>
<td>gblocks</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Masking of Sequence Alignments with GBLOCKS</h2>

<h3>Description</h3>

<p>Provides a wrapper to Gblocks, a computer program written in
ANSI C language that eliminates poorly aligned positions and divergent
regions of an alignment of DNA or protein sequences. Gblocks selects
conserved blocks from a multiple alignment according to a set of features
of the alignment positions.
</p>


<h3>Usage</h3>

<pre><code class="language-R">gblocks(
  x,
  b1 = 0.5,
  b2 = b1,
  b3 = ncol(x),
  b4 = 2,
  b5 = "a",
  target = "alignment",
  exec
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>A matrix of DNA sequences of classes <code>DNAbin</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>b1</code></td>
<td>
<p>A real number, the <b>minimum number of sequences for a
conserved position</b> given as a fraction. Values between 0.5 and 1.0 are
allowed. <em>Larger</em> values will <em>decrease</em> the number of selected
positions, i.e. are more <em>conservative</em>. Defaults to 0.5</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>b2</code></td>
<td>
<p>A real number, the <b>minimum number of sequences for a flank
position</b> given as a fraction. Values must be equal or larger than
<code>b1</code>. <em>Larger</em> values will <em>decrease</em> the number of selected
positions, i.e. are <em>more conservative</em>. Defaults to 0.5</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>b3</code></td>
<td>
<p>An integer, the <b>maximum number of contiguous nonconserved
positions</b>; any integer is allowed. <em>Larger</em> values will
<em>increase</em> the number of selected position, i.e. are <em>less
conservative</em>. Defaults to the number of positions in the alignment.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>b4</code></td>
<td>
<p>An integer, the <b>minimum length of a block</b>, any integer equal
to or bigger than 2 is allowed. <em>Larger</em> values will <em>decrease</em>
the number of selected positions, i.e. are <em>more conservative</em>. Defaults to
2.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>b5</code></td>
<td>
<p>A character string indicating the <b>treatment of gap
positions</b>. Three choices are possible. 1. <code>"n"</code>: <em>No</em> gap
positions are allowed in the final alignment. All positions with a single
gap or more are treated as a gap position for the block selection
procedure, and they and the adjacent nonconserved positions are eliminated.
2. <code>"h"</code>: Only positions where <em>50% or more</em> of the sequences
have a gap are treated as a gap position. Thus, positions with a gap in
less than 50% of the sequences can be selected in the final alignment if
they are within an appropriate block. 3. <code>"a"</code>: <em>All</em> gap
positions can be selected. Positions with gaps are not treated differently
from other positions (default).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>target</code></td>
<td>
<p>A vector of mode <code>"character"</code> giving the output format:
<code>"alignment"</code> will return the alignment with only the selected
positions, <code>"index"</code> will return the indices of the selected position,
and <code>"score"</code> will provide a score for every position in the original
alignment (0 for excluded, 1 for included).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exec</code></td>
<td>
<p>A character string indicating the path to the GBLOCKS executable.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Explanation of the routine taken from the Online Documentation:
First, the degree of conservation of every positions of the multiple
alignment is evaluated and classified as <em>nonconserved</em>,
<em>conserved</em>, or <em>highly conserved</em>. All stretches of contiguous
nonconserved positions bigger than a certain value (<b>b3</b>) are
rejected. In such stretches, alignments are normally ambiguous and, even
when in some cases a unique alignment could be given, multiple hidden
substitutions make them inadequate for phylogenetic analysis. In the
remaining blocks, flanks are examined and positions are removed until
blocks are surrounded by highly conserved positions at both flanks. This
way, selected blocks are anchored by positions that can be aligned with
high confidence. Then, all gap positions -that can be defined in three
different ways (<b>b5</b>)- are removed. Furthermore, nonconserved
positions adjacent to a gap position are also eliminated until a conserved
position is reached, because regions adjacent to a gap are the most
difficult to align. Finally, small blocks (falling below the limit of
<b>b4</b>) remaining after gap cleaning are also removed.
</p>


<h3>Value</h3>

<p>A <code>matrix</code> of class <code>"DNAbin"</code>
</p>


<h3>Note</h3>

<p><code>gblocks</code> was last updated and tested to work with Gblocks 0.91b.
If you have problems getting the function to work with a newer version of
Gblocks, please contact the package maintainer.
</p>


<h3>References</h3>

<p>Castresana, J. 2000. Selection of conserved blocks from multiple
alignments for their use in phylogenetic analysis. <em>Molecular Biology
and Evolution</em> <b>17</b>, 540-552.
</p>
<p>Talavera, G., and J. Castresana. 2007. Improvement of phylogenies
after removing divergent and ambiguously aligned blocks from protein
sequence alignments. <em>Systematic Biology</em> <b>56</b>, 564-577.
</p>
<p><b>Gblocks website</b>: 
<a href="https://www.biologiaevolutiva.org/jcastresana/Gblocks.html">https://www.biologiaevolutiva.org/jcastresana/Gblocks.html</a>
</p>


<h3>See Also</h3>

<p><code>mafft</code> and <code>prank</code> for multiple sequence
alignment; <code>aliscore</code> for another alignment masking algorithm.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(ips.28S)
## Not run: gblocks(ips.28S)  
</code></pre>


</div>