<div class="container">

<table style="width: 100%;"><tr>
<td>extract_annotation</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Protein annotation extraction
</h2>

<h3>Description</h3>

<p>Extracts annotation columns from a long-format input
</p>


<h3>Usage</h3>

<pre><code class="language-R">extract_annotation(protein_ids, quant_table, primary_id = "PG.ProteinGroups",
                   annotation_columns = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>protein_ids</code></td>
<td>
<p>A vector of protein ids.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>quant_table</code></td>
<td>
<p>A long-format input table. The input is typically the same as input to the <code>preprocess</code> function.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>primary_id</code></td>
<td>
<p>The column containing protein ids.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>annotation_columns</code></td>
<td>
<p>A vector of columns for annotation.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A table of proteins and associated annotation extracted from the input.
</p>


<h3>Author(s)</h3>

<p>Thang V. Pham
</p>


<h3>References</h3>

<p>Pham TV, Henneman AA, Jimenez CR. iq: an R package to estimate relative protein
abundances from ion quantification in DIA-MS-based proteomics. <em>Bioinformatics</em> 2020 Apr 15;36(8):2611-2613.
</p>


<h3>See Also</h3>

<p><code>preprocess</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
data("spikeins")
extra_names &lt;- iq::extract_annotation(levels(spikeins$PG.ProteinGroups),
                                      spikeins,
                                      annotation_columns = c("PG.Genes", "PG.ProteinNames"))

</code></pre>


</div>