<div class="container">

<table style="width: 100%;"><tr>
<td>fast_read</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Reading data from an input file
</h2>

<h3>Description</h3>

<p>A highly efficient reading of a tab-separated text file for iq processing.
</p>


<h3>Usage</h3>

<pre><code class="language-R">fast_read(filename,
          sample_id = "R.Condition",
          primary_id = "PG.ProteinGroups",
          secondary_id = c("EG.ModifiedSequence", "FG.Charge", "F.FrgIon", "F.Charge"),
          intensity_col = "F.PeakArea",
          annotation_col = c("PG.Genes", "PG.ProteinNames"),
          filter_string_equal = c("F.ExcludedFromQuantification" = "False"),
          filter_string_not_equal = NULL,
          filter_double_less = c("PG.Qvalue" = "0.01", "EG.Qvalue" = "0.01"),
          filter_double_greater = NULL,
          intensity_col_sep = NULL,
          intensity_col_id = NULL,
          na_string = "0")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>filename</code></td>
<td>

<p>A long-format tab-separated text file with a primary column of protein identification, secondary columns of fragment ions, a column of sample names, a column for quantitative intensities, and extra columns for annotation.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>primary_id</code></td>
<td>
<p>Unique values in this column form the list of proteins to be quantified.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>secondary_id</code></td>
<td>
<p>A concatenation of these columns determines the fragment ions used for quantification.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sample_id</code></td>
<td>
<p>Unique values in this column form the list of samples.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>intensity_col</code></td>
<td>
<p>The column for intensities.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>annotation_col</code></td>
<td>
<p>Annotation columns</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filter_string_equal</code></td>
<td>
<p>A named vector of strings. Only rows satisfying the condition are kept.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filter_string_not_equal</code></td>
<td>
<p>A named vector of strings. Only rows satisfying the condition are kept.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filter_double_less</code></td>
<td>
<p>A named vector of strings. Only rows satisfying the condition are kept. Default PG.Qvalue &lt; 0.01 and EG.Qvalue &lt; 0.01.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filter_double_greater</code></td>
<td>
<p>A named vector of strings. Only rows satisfying the condition are kept.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>intensity_col_sep</code></td>
<td>
<p>A separator character when entries in the intensity column contain multiple values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>intensity_col_id</code></td>
<td>
<p>The column for identities of multiple quantitative values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>na_string</code></td>
<td>
<p>The value considered as NA.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>When entries in the intensity column contain multiple values, this function will replicate entries in other column and the <code>secondary_id</code> will be appended with corresponding entries in <code>intensity_col_id</code> when it is provided. Otherwise, integer values 1, 2, 3, etc... will be used.
</p>


<h3>Value</h3>

<p>A list is returned with following components
</p>
<table>
<tr style="vertical-align: top;">
<td><code>protein</code></td>
<td>
<p>A table of proteins in the first column followed by annotation columns.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sample</code></td>
<td>
<p>A vector of samples.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ion</code></td>
<td>
<p>A vector of fragment ions to be used for quantification.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>quant_table</code></td>
<td>
<p>A list of four components: protein_list (index pointing to <code>protein</code>)), sample_list (index pointing to <code>sample</code>), id (index pointing to <code>ion</code>), and quant (intensities).</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Thang V. Pham
</p>


<h3>References</h3>

<p>Pham TV, Henneman AA, Jimenez CR. iq: an R package to estimate relative protein
abundances from ion quantification in DIA-MS-based proteomics. <em>Bioinformatics</em> 2020 Apr 15;36(8):2611-2613.
</p>


</div>