<div class="container">

<table style="width: 100%;"><tr>
<td>Run_permutation</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Derive importance scores for M permuted data sets.</h2>

<h3>Description</h3>

<p>This function computes importance score for <code>M</code> permuted data sets. Sample labels of target genes are randomly permuted and iRafNet is implemented. Resulting importance scores can be used to derive an estimate of FDR.  
</p>


<h3>Usage</h3>

<pre><code class="language-R">Run_permutation(X, W, ntree, mtry,genes.name,M)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>
<p><code>(n x p)</code> Matrix containing expression levels for <code>n</code> samples and <code>p</code> genes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>W</code></td>
<td>
<p><code>(p x p)</code> Matrix containing iRafNet sampling scores. Element <code>(i,j)</code> contains score for regulatory relationship <code>(i -&gt; j)</code>. Scores must be non-negative. Larger value of sampling score corresponds to higher likelihood of gene <code>i</code> regulating gene <code>j</code>. Columns and rows of <code>W</code> must be in the same order as the columns of <code>X</code>. Sampling scores <code>W</code> are computed considering one prior data such as protein-protein interactions or gene expression from knock-out experiments.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ntree</code></td>
<td>
<p>Numeric value: number of trees.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mtry</code></td>
<td>
<p>Numeric value: number of predictors to be sampled at each node.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genes.name</code></td>
<td>
<p>Vector containing genes name. The order needs to match the rows of <code>x_j</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>M</code></td>
<td>
<p>Integer: total number of permutations.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A matrix with <code>I</code> rows and <code>M</code> columns with <code>I</code> being the total number of regulations and <code>M</code> the number of permutations. Element <code>(i,j)</code> corresponds to the importance score of interaction <code>i</code> for permuted data <code>j</code>. 
</p>


<h3>References</h3>

<p>Petralia, F., Wang, P., Yang, J., Tu, Z. (2015) Integrative random forest for gene regulatory network inference, <em>Bioinformatics</em>, <b>31</b>, i197-i205.
</p>
<p>Petralia, F., Song, W.M., Tu, Z. and Wang, P. (2016). New method for joint network analysis reveals common and different coexpression patterns among genes and proteins in breast cancer. <em>Journal of proteome research</em>, <b>15</b>(3), pp.743-754.
</p>
<p>A. Liaw and M. Wiener (2002). Classification and Regression by randomForest. <em>R News</em> <b>2</b>, 18â€“22.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
  # --- Generate data sets
  n&lt;-20                  # sample size 
  p&lt;-5                   # number of genes
  genes.name&lt;-paste("G",seq(1,p),sep="")   # genes name
  M=5;            # number of permutations
 
  data&lt;-matrix(rnorm(p*n),n,p)       # generate expression matrix
  W&lt;-abs(matrix(rnorm(p*p),p,p))          # generate score for regulatory relationships
 
  # --- Standardize variables to mean 0 and variance 1
  data &lt;- (apply(data, 2, function(x) { (x - mean(x)) / sd(x) } ))

  # --- Run iRafNet and obtain importance score of regulatory relationships
  out.iRafNet&lt;-iRafNet(data,W,mtry=round(sqrt(p-1)),ntree=1000,genes.name)

  # --- Run iRafNet for M permuted data sets
  out.perm&lt;-Run_permutation(data,W,mtry=round(sqrt(p-1)),ntree=1000,genes.name,M)

</code></pre>


</div>