<div class="container">

<table style="width: 100%;"><tr>
<td>robustness</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Robustness of ISA biclusters and PPA co-modules</h2>

<h3>Description</h3>

<p>Robustness of ISA biclusters and PPA co-modules. The more robust
biclusters/co-modules are more significant in the sense that it is
less likely to see them in random data.
</p>


<h3>Usage</h3>

<pre><code class="language-R">## S4 method for signature 'list'
robustness(normed.data, ...)
## S4 method for signature 'matrix'
isa.filter.robust(data, ...)
## S4 method for signature 'list'
ppa.filter.robust(data, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>normed.data</code></td>
<td>
<p>The normalized input data, usually calculated with
<code>isa.normalize</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>The original, not normalized input data, a matrix for
<code>isa.filter.robust</code>, a list of two matrices for
<code>ppa.filter.robust</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Additional arguments, see details below.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><code>robustness</code> can be called as
</p>
<pre>
    robustness(normed.data, row.scores, col.scores)
  </pre>
<p><code>isa.filter.robust</code> can be called as
</p>
<pre>
    isa.filter.robust(data, normed.data, isares,
                      perms = 1, row.seeds, col.seeds)
  </pre>
<p>and <code>ppa.filter.robust</code> can be called as
</p>
<pre>
    ppa.filter.robust(data, normed.data, ppares,
                      perms = 1, row1.seeds, col1.seeds,
		      row2.seeds, col2.seeds)
  </pre>
<p>These arguments are:
</p>

<dl>
<dt>normed.data</dt>
<dd>
<p>The normalized input data, usually calculated with
<code>isa.normalize</code> (for ISA) or
<code>ppa.normalize</code> (for PPA).</p>
</dd>
<dt>row.scores</dt>
<dd>
<p>The scores of the row components of the
biclusters. Usually the <code>rows</code> member of the result list, as
returned by <code>isa</code>, or <code>isa.iterate</code> or some
other ISA function.</p>
</dd>
<dt>col.scores</dt>
<dd>
<p>The scores of the columns of the biclusters, usually
the <code>columns</code> member of the result list from
<code>isa</code>.</p>
</dd>
<dt>data</dt>
<dd>
<p>The original, not normalized input data.</p>
</dd>
<dt>isares</dt>
<dd>
<p>The result of ISA, coming from <code>isa</code> or
<code>isa.iterate</code> or any other function that return the same
format.</p>
</dd>
<dt>perms</dt>
<dd>
<p>The number of permutations to perform on the input data.</p>
</dd>
<dt>row.seeds</dt>
<dd>
<p>Optionally the row seeds for the ISA run on the
scrambled data. If this and <code>col.seeds</code> are both omitted the
same number of random seeds are used as for <code>isaresult</code>.</p>
</dd>
<dt>col.seeds</dt>
<dd>
<p>Optionally the column seed to use for the ISA on the
scrambled input matrix. If both this and <code>row.seeds</code> are
omitted, then the same number of random (row) seeds will be used as
for <code>isares</code>.</p>
</dd>
<dt>ppares</dt>
<dd>
<p>The result of a PPA run, the output of the
<code>ppa.iterate</code> or the <code>ppa.unique</code>
function (or any other function that returns the same format).</p>
</dd>
<dt>row1.seeds</dt>
<dd>
<p>Optionally, the seeds based of the rows of the
first input matrix, can be given here. Otherwise random seeds are
used, the same number as it was used to find the original
co-modules.</p>
</dd>
<dt>col1.seeds</dt>
<dd>
<p>Optionally, the seeds based of the columns of the
first input matrix, can be given here. Otherwise random seeds are
used, the same number as it was used to find the original
co-modules.</p>
</dd>
<dt>row2.seeds</dt>
<dd>
<p>Optionally, the seeds based of the rows of the
second input matrix, can be given here. Otherwise random seeds are
used, the same number as it was used to find the original
co-modules.</p>
</dd>
<dt>col2.seeds</dt>
<dd>
<p>Optionally, the seeds based of the columns of the
second input matrix, can be given here. Otherwise random seeds are
used, the same number as it was used to find the original
co-modules.</p>
</dd>
</dl>
<p>Even if you generate a matrix with uniform random noise in it, if you
calculate ISA on it, you will get some biclusters, except maybe if you
use very strict threshold parameters. These biclusters contain rows
and columns that are correlated just by chance. The same is true for
PPA.
</p>
<p>To circumvent this, you can use the so-called robustness measure of
the biclusters/co-modules. The robustness of a bicluster is the
function of its rows, columns and the input data, and it is a real
number, usually positive. It is roughly equivalent to the principal
singular value of the submatrix (of the reordered input matrix)
defined by the bicluster.
</p>
<p><code>robustness</code> calculates the robustness score of a set of
biclusters/co-modules, usually coming from one or more ISA/PPA
iterations.
</p>
<p><code>isa.filter.robust</code> provides filtering based on the robustness
measure. It reshuffles the input matrix and calculates ISA on it, with
the parameters that were used to find the biclusters under
evaluation. It then calculates the robustness for the modules that
were found in the scrambled matrix (if there is any) and removes any
modules from the data set that have a lower robustness score than at
least one module in the scrambled data.
</p>
<p>You can think of <code>isa.filter.robust</code> as a permutation test, but
the input data is shuffled only once (at least by default), because of
the relatively high computational demands of the ISA.
</p>
<p><code>ppa.filter.robust</code> does essentially the same, but for PPA
co-modules.
</p>


<h3>Value</h3>

<p><code>robustness</code> returns a numeric vector, the robustness score of
each bicluster.
</p>
<p><code>isa.filter.robust</code> returns a named list, the filtered
<code>isares</code>, see the return value of <code>isa.iterate</code> for
the structure of the list.
</p>
<p><code>ppa.filter.robust</code> resturns a named list, the filtered
<code>ppares</code>, see the return value of <code>ppa.iterate</code> for
the structure of the list.
</p>


<h3>Author(s)</h3>

<p> Gabor Csardi <a href="mailto:Gabor.Csardi@unil.ch">Gabor.Csardi@unil.ch</a> </p>


<h3>References</h3>

<p>Bergmann S, Ihmels J, Barkai N: Iterative signature algorithm for the
analysis of large-scale gene expression data <em>Phys Rev E Stat
Nonlin Soft Matter Phys.</em> 2003 Mar;67(3 Pt 1):031902. Epub 2003 Mar 11.
</p>
<p>Ihmels J, Friedlander G, Bergmann S, Sarig O, Ziv Y, Barkai N:
Revealing modular organization in the yeast transcriptional network
<em>Nat Genet.</em> 2002 Aug;31(4):370-7. Epub 2002 Jul 22
</p>
<p>Ihmels J, Bergmann S, Barkai N:
Defining transcription modules using large-scale gene expression data
<em>Bioinformatics</em> 2004 Sep 1;20(13):1993-2003. Epub 2004 Mar 25.
</p>
<p>Kutalik Z, Bergmann S, Beckmann, J: A modular approach for integrative
analysis of large-scale gene-expression and drug-response data
<em>Nat Biotechnol</em> 2008 May; 26(5) 531-9.
</p>


<h3>See Also</h3>

 <p>isa2-package for a short introduction on the Iterative
Signature Algorithm. See <code>isa</code> for an easy way of running
ISA, <code>ppa</code> for an easy way of running the PPA.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## A basic ISA work flow for a single threshold combination
## In-silico data
set.seed(1)
insili &lt;- isa.in.silico()

## Random seeds
seeds &lt;- generate.seeds(length=nrow(insili[[1]]), count=100)

## Normalize input matrix
nm &lt;- isa.normalize(insili[[1]])

## Do ISA
isares &lt;- isa.iterate(nm, row.seeds=seeds, thr.row=2, thr.col=1)

## Eliminate duplicates
isares &lt;- isa.unique(nm, isares)

## Calculate robustness
rob &lt;- robustness(nm, isares$rows, isares$columns)
rob

## There are three robust ones and a lot of less robust ones
## Plot the three robust ones and three others
if (interactive()) {
  toplot1 &lt;- rev(order(rob))[1:3]
  toplot2 &lt;- sample(seq_along(rob)[-toplot1], 3)
  layout( rbind(1:3,4:6) )
  for (i in c(toplot1, toplot2)) {
    image(outer(isares$rows[,i], isares$column[,i]),
          main=round(rob[i],2))
  }
}

## Filter out not robust ones
isares2 &lt;- isa.filter.robust(insili[[1]], nm, isares)

## Probably there are only three of them left
ncol(isares2$rows)
</code></pre>


</div>