<div class="container">

<table style="width: 100%;"><tr>
<td>isa.iterate</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>The Iterative Signature Algorithm</h2>

<h3>Description</h3>

<p>Perform ISA on the (normalized) input matrix.
</p>


<h3>Usage</h3>

<pre><code class="language-R">## S4 method for signature 'list'
isa.iterate(normed.data, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>normed.data</code></td>
<td>
<p>The normalized data. A list of two matrices,
usually coming from <code>isa.normalize</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Additional arguments, see details below.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><code>isa.iterate</code> performs the ISA iteration on the specified input
seeds. It can be called as 
</p>
<pre>
    isa.iterate(normed.data, row.seeds, col.seeds,
                thr.row, thr.col = thr.row,
		direction = c("updown", "updown"),
		convergence = c("corx", "cor", "eps"),
		cor.limit = 0.99, eps = 1e-04, corx=3,
		oscillation = FALSE, maxiter = 100)
	      </pre>
<p>where the arguments are:
</p>

<dl>
<dt>normed.data</dt>
<dd>
<p>The normalized data. A list of two matrices,
usually coming from <code>isa.normalize</code>.</p>
</dd>
<dt>row.seeds</dt>
<dd>
<p>The row seed vectors to start the ISA runs
from. Every column is a seed vector. (If this argument and
<code>col.seeds</code> are both present, then both of them are used.)</p>
</dd>
<dt>col.seeds</dt>
<dd>
<p>The column seed vectors to start the ISA runs from,
every column is a seed vector. (If this argument and
<code>row.seeds</code> are both present, then both of them are used.)</p>
</dd>
<dt>thr.row</dt>
<dd>
<p>Numeric scalar or vector giving the threshold parameter
for the rows. Higher values indicate a more stringent threshold and
the result biclusters will contain less rows on average. The
threshold is measured by the number of standard deviations from the
mean, over the values of the row vector. If it is a vector then it
must contain an entry for each seed.</p>
</dd>
<dt>thr.col</dt>
<dd>
<p>Numeric scalar or vector giving the threshold parameter
for the columns. The analogue of <code>thr.row</code>.</p>
</dd>
<dt>direction</dt>
<dd>
<p>Character vector of length two, one for the rows, one
for the columns. It specifies whether we are interested in
rows/columns that are higher (‘<code>up</code>’) than average,
lower than average (‘<code>down</code>’), or both
(‘<code>updown</code>’).</p>
</dd>
<dt>convergence</dt>
<dd>
<p>Character scalar, the convergence criteria for the
ISA iteration. If it is ‘<code>cor</code>’, then convergence is
measured based on high Pearson correlation (see the <code>cor.limit</code>
argument below) of the subsequent row and
column vectors. If it is ‘<code>eps</code>’, then all entries of
the subsequent row and column vectors must be close to each other,
see the <code>eps</code> argument below.
</p>
<p>‘<code>corx</code>’ is similar to ‘<code>cor</code>’, but the
current row/column vectors are compared to the ones <code>corx</code>
steps ago, instead of the ones in the previous step. See the
<code>corx</code> argument below, that gives the size of the shift.
</p>
</dd>
<dt>cor.limit</dt>
<dd>
<p>The correlation limit for convergence if the
‘<code>cor</code>’ method is used.</p>
</dd>
<dt>eps</dt>
<dd>
<p>Limit for convergence if the ‘<code>eps</code>’ method is
used.</p>
</dd>
<dt>corx</dt>
<dd>
<p>The number of iterations to use as a shift, for checking
convergence with the ‘<code>corx</code>’ method.</p>
</dd>
<dt>oscillation</dt>
<dd>
<p>Logical scalar, whether to look for oscillating
seeds. Usually there are not too many oscillating seeds, so it is
safe to leave this on <code>FALSE</code>.</p>
</dd>
<dt>maxiter</dt>
<dd>
<p>The maximum number of iterations allowed.</p>
</dd>
</dl>
<h3>Value</h3>

<p>A named list with many components. Please see the manual page of
<code>isa</code> for a complete description.
</p>


<h3>Author(s)</h3>

<p> Gabor Csardi <a href="mailto:Gabor.Csardi@unil.ch">Gabor.Csardi@unil.ch</a> </p>


<h3>References</h3>

<p>Bergmann S, Ihmels J, Barkai N: Iterative signature algorithm for the
analysis of large-scale gene expression data <em>Phys Rev E Stat
Nonlin Soft Matter Phys.</em> 2003 Mar;67(3 Pt 1):031902. Epub 2003 Mar 11.
</p>
<p>Ihmels J, Friedlander G, Bergmann S, Sarig O, Ziv Y, Barkai N:
Revealing modular organization in the yeast transcriptional network
<em>Nat Genet.</em> 2002 Aug;31(4):370-7. Epub 2002 Jul 22
</p>
<p>Ihmels J, Bergmann S, Barkai N:
Defining transcription modules using large-scale gene expression data
<em>Bioinformatics</em> 2004 Sep 1;20(13):1993-2003. Epub 2004 Mar 25.
</p>


<h3>See Also</h3>

 <p>isa2-package for a short introduction on the Iterative
Signature Algorithm. See <code>isa</code> for an easy way of running
ISA.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## A basic ISA work flow for a single threshold combination
## In-silico data
set.seed(1)
insili &lt;- isa.in.silico()

## Random seeds
seeds &lt;- generate.seeds(length=nrow(insili[[1]]), count=100)

## Normalize input matrix
nm &lt;- isa.normalize(insili[[1]])

## Do ISA
isares &lt;- isa.iterate(nm, row.seeds=seeds, thr.row=2, thr.col=1)

## Eliminate duplicates
isares &lt;- isa.unique(nm, isares)

## Filter out not robust ones
isares &lt;- isa.filter.robust(insili[[1]], nm, isares)

## Print the sizes of the modules
cbind( colSums(isares$rows!=0), colSums(isares$columns!=0) )

## Plot the original data and the modules found
if (interactive()) {
  layout(rbind(1:2))
  image(insili[[1]], main="In silico data")
  image(outer(isares$rows[,1],isares$columns[,1])+
        outer(isares$rows[,2],isares$columns[,2])+
        outer(isares$rows[,3],isares$columns[,3]), main="ISA modules")
}
</code></pre>


</div>