<div class="container">

<table style="width: 100%;"><tr>
<td>ISOpure.step2.PPE</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Perform second step of ISOpure purification algorithm</h2>

<h3>Description</h3>

<p>Performs the second step of the ISOpure purification algorithm, taking tumor data and
normal profiles and returning the a list, ISOpureS2model, with all the updated parameters.</p>


<h3>Usage</h3>

<pre><code class="language-R">ISOpure.step2.PPE(tumordata, BB, ISOpureS1model, MIN_KAPPA, logging.level) 
</code></pre>


<h3>Arguments</h3>


<table>
<tr style="vertical-align: top;">
<td><code>tumordata</code></td>
<td>
<p>(same as for ISOpureS1) a GxD matrix representing gene expression profiles of 
heterogeneous (mixed) tumor samples, where G is the number of genes, D is the number of tumor
samples.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>BB</code></td>
<td>
<p>(same as for ISOpureS1) represents B = [b_1 ... b_(K-1)] matrix (from Genome Medicine 
paper) a Gx(K-1) matrix, where (K-1) is the number of normal profiles 
<code class="reqn">(\beta_1,...,\beta_(K-1))</code>, G is the number of genes.  These are the normal profiles 
representing normal cells that contaminate the tumor samples  (i.e. normal samples from the 
same tissue location as the tumor).  The minimum element of BB must be greater than 0 – 
i.e. every gene/transcript must be observed on some level in each normal sample.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ISOpureS1model</code></td>
<td>
<p>output model list from ISOpureS1 code</p>
</td>
</tr>
</table>
<table>
<tr style="vertical-align: top;">
<td><code>MIN_KAPPA</code></td>
<td>
<p>(optional) The minimum value allowed for the strength parameter kappa placed
over the reference cancer profile m (see Quon et al, 2013).  By default, this is set to 
1/min(BB), such that the log likelihood of the model is always finite.  However, when the 
min(BB) is very small, this forces MIN_KAPPA to be very large, and can sometimes cause the 
reference profile m to look too much like a 'normal profile' (and therefore you may observe 
the tumor samples having low % cancer content estimates).  If this is the case, you can try 
setting MIN_KAPPA=1, or some other small value.  For reference, for the data presented in 
Quon et al., 2013, MIN_KAPPA is on the order of 10^5.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>logging.level</code></td>
<td>
<p>(optional) A string that gives the logging threshold for futile.logger. The
possible options are 'TRACE', 'DEBUG', 'INFO', 'WARN', 'ERROR', 'FATAL'. Currently the 
messages in ISOpureR are only in the categories 'INFO', 'WARN', and 'FATAL', and the default 
setting is 'INFO'.  Setting a setting for the entire package will over-ride the setting for
a particular function.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>ISOpureS2model, a list with the following important fields:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>theta</code></td>
<td>
<p>a DxK matrix, giving the fractional composition of each tumor sample.  Each row 
represents a tumor sample that was part of the input, and the first K-1 columns correspond 
to the fractional composition with respect to the Source Panel contaminants.  The last 
column represents the fractional composition of the pure cancer cells.  In other words,	
each row sums to 1, and element (i,j) of the matrix denotes the fraction of tumor i 
attributable to component j (where the last column refers to cancer cells, and the first 
K-1 columns refer to different 'normal cell' components).  The 'cancer', or tumor purity, 
estimate of each tumor is simply the last column of theta.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alphapurities</code></td>
<td>
<p>(same as ISOpureS1) tumor purities (alpha_i in paper), same as the last 
column of the theta variable, pulled out for user convenience - not changed in step 2</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cc_cancerprofiles</code></td>
<td>
<p>purified cancer profiles.  This matrix is of the same dimensionality as 
tumordata, and is also on the same scale (i.e. although ISOpureS2 treats purified cancer 
profiles as parameters of a multinomial distribution, we re-scale them to be on the same 
scale as the input tumor profiles – see Genome Medicine paper). Column i of
cc_cancerprofiles corresponds to column i of tumordata.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>total_loglikelihood</code></td>
<td>
<p>log likelihood of the model</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>omega</code></td>
<td>
<p>(internal parameter, same as ISOpureS1) prior over the reference cancer profile -
not changed in step 2</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>vv</code></td>
<td>
<p>(internal parameter) hyper-parameters from Dirichlet distribution, representing both
mean and strength of a Dirichlet distribution over theta</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>kappa</code></td>
<td>
<p>(internal parameter) the strength parameter over the Dirichlet distribution over 
cc, given the reference cancer parameter, mm</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mm_weights, theta_weights, omega_weights</code></td>
<td>
<p>(internal parameters) used in the optimization
of mm, theta, and omega (instead of performing constrained optimization on these 
positively constrained variables directly, we optimize their logs in an unconstrained 
fashion.)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>log_BBtranspose, PPtranspose, log_all_rates:</code></td>
<td>
<p> (internal parameters) used in the 
calculations of loglikelihood</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>MIN_KAPPA</code></td>
<td>
<p>(internal parameter) as described in the Arguments section</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Gerald Quon, Catalina Anghel, Francis Nguyen</p>


<h3>References</h3>

<p>G Quon, S Haider, AG Deshwar, A Cui, PC Boutros, QD Morris. <em>Computational purification of 
individual tumor gene expression profiles</em>. Genome Medicine (2013) 5:29, 
<a href="http://genomemedicine.com/content/5/3/29">http://genomemedicine.com/content/5/3/29</a>.
</p>
<p>G Quon, QD Morris. <em>ISOLATE: a computational strategy for identifying the primary origin of
cancers using high-thoroughput sequencing</em>. Bioinformatics 2009, 25:2882-2889 
<a href="http://bioinformatics.oxfordjournals.org/content/25/21/2882">http://bioinformatics.oxfordjournals.org/content/25/21/2882</a>.
</p>


</div>